---
title: "ofav_GOSeq"
author: "jillashey"
date: "11/14/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# Load libraries
library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("gridExtra")
library("clusterProfiler")
library("DataCombine")
library("VennDiagram")
```

This code is using the ofav data. This code is also using the GO terms generated by Diamond (B2G), Swiss Prot (B2G), and IPS [here](https://github.com/JillAshey/FunctionalAnnotation/tree/main/final_Annotations) in Nov. 2021. 

## Read in all expressed ofav genes (poverA = 0.85,5)
```{r}
# Obtain names of all expressed Ofav genes (poverA = 0.85,5), and all differentially expressed genes (p<0.05)
gcounts_filt_ofav <- read.csv("Output/DESeq2/ofav/ofav_counts_filtered.csv", header = T)
dim(gcounts_filt_ofav) # 18815 rows x 16
colnames(gcounts_filt_ofav)[1] <-"gene_id" # make colnames a true column called gene_id
length(unique(gcounts_filt_ofav$gene_id)) # 18815 total unique gene ids 
for ( col in 1:ncol(gcounts_filt_ofav)){
  colnames(gcounts_filt_ofav)[col] <-  gsub("X", "", colnames(gcounts_filt_ofav)[col])
}
head(gcounts_filt_ofav)
# Isolate LOC term
gcounts_filt_ofav$gene_id <- gsub("\\|.*", "", gcounts_filt_ofav$gene_id)
gcounts_filt_ofav$gene_id <- gsub(".*-", "", gcounts_filt_ofav$gene_id)
```

## Read in ofav DEGs
```{r}
DEG_ofav <- read.csv("Output/DESeq2/ofav/ofav_unique.sig.list.csv", header = TRUE)
dim(DEG_ofav) # 16 x 16
colnames(DEG_ofav)[1] <-"gene_id" # make colnames a true column called gene_id
for ( col in 1:ncol(DEG_ofav)){
  colnames(DEG_ofav)[col] <-  gsub("X", "", colnames(DEG_ofav)[col])
}
head(DEG_ofav)
length(unique(DEG_ofav$gene_id)) # 16 total unique gene ids 
# Isolate LOC term
DEG_ofav$gene_id <- gsub("\\|.*", "", DEG_ofav$gene_id)
DEG_ofav$gene_id <- gsub(".*-", "", DEG_ofav$gene_id)
```

```{r}
# Read in length data (calculated directly from transcripts) and merge with filtered counts 
length_Ofav <- read.csv("Output/DESeq2/ofav/length.mRNA_Ofav.csv")[,2:3]
length_Ofav <- length_Ofav[!duplicated(length_Ofav[c(2)]),] # remove duplicate gene ids

length_merge <- merge(length_Ofav, gcounts_filt_ofav, by = "gene_id")
```

# Build GOSEQ vector 
GOseq requires a vector of all genes, all differentially expressed genes, and gene lengths

## Make vectors
```{r}
#DEG<- filter(length_Acerv, gene_id %in% DEG_acerv_sub$gene_id) - for some reason, R keeps giving me error saying object 'gene_id' not found, so used another way below
DEG <- length_merge[length_merge$gene_id %in% DEG_ofav$gene_id, ]
dim(DEG) # 16 x 18
DEG_names <- as.vector(DEG$gene_id)
gene_vector <- as.integer(length_merge$gene_id%in%DEG_names)
names(gene_vector) <- length_merge$gene_id

# Make ID vector 
IDvector <- length_merge$gene_id

# Make length vector
lengthVector <- length_merge$length
lengthVector <- as.numeric(lengthVector)
```

## Calculate Probability Weighting Function
```{r}
DEG.pwf<-nullp(gene_vector, IDvector, bias.data=lengthVector) #weight vector by length of gene

# plot looks weird...does this mean that there is not much length bias in the dataset?
```

# GOSeq w/ my annotations from Nov. 2021 (GO terms generated by Diamond (B2G), Swiss Prot (B2G), and IPS)

## Load GO terms 
```{r}
annot_GO <- read.csv("~/Desktop/PutnamLab/Repositories/FunctionalAnnotation/FunctionalAnnotation/final_Annotations/Ofav_GOterms_20211114.csv", header = TRUE)[,2:3]
# GO terms already split into single GO term per row/sequence 
colnames(annot_GO) <- c("gene_id", "GO.ID")
annot_GO$gene_id <- gsub("-RA", "", annot_GO$gene_id)
annot_GO[annot_GO == "NA"] <- NA
annot_GO$GO.ID<- as.character(annot_GO$GO.ID)
annot_GO$GO.ID <- replace_na(annot_GO$GO.ID, "unknown")
annot_GO$GO.ID <- as.factor(annot_GO$GO.ID)
annot_GO$gene_id <- as.factor(annot_GO$gene_id)
annot_GO <- unique(annot_GO)

dim(annot_GO) # 152417 x 2
length(unique(annot_GO$GO.ID)) # 12131 unique GO.IDs
head(annot_GO)
```

```{r}
# Merge original gff with annot GO df to get LOC term
ofav <- read.csv("~/Desktop/GFFs/ofav/GCF_002042975.1_ofav_dov_v1_genomic.gff", header=FALSE, sep="\t", skip=6) # read in gff from NCBI
colnames(ofav) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene") # name cols
ofav <- na.omit(ofav)
ofav <- subset(ofav, id == "CDS")
ofav$prot_id <- gsub(".*cds-","", ofav$gene) # remove everything before -cds
ofav$prot_id <- gsub(";.*","", ofav$prot_id) # remove everything after ;
ofav$gene_id <- gsub(".*LOC","", ofav$gene) # remove everything before LOC
ofav$gene_id <- gsub(";.*","", ofav$gene_id) # remove everything after ;
ofav$gene_id <- gsub("iso.*","", ofav$gene_id) # remove everything after iso
ofav$gene_id <- paste0("LOC", ofav$gene_id)

# select cols with XP- and LOC info
ofav_sub <- ofav[,c("prot_id", "gene_id")]
ofav_sub<- unique(ofav_sub)
colnames(annot_GO) <- c("prot_id", "GO.ID") # temperoraily change annot_GO col names to match ofav_sub

# Merge ofav_sub w/ annot GO df and remove prot_id
merge <- merge(annot_GO, ofav_sub, by = "prot_id")
merge <- select(merge, c(GO.ID, gene_id))
merge <- na.omit(merge)
merge <- unique(merge)

# Okay lets see if this merge file works w/ GOSeq
```

# Perform GOseq

## Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
```{r}
### Perform GOseq
# Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
# should I include this: test.cats=c("GO:CC", "GO:BP", "GO:MF") ?
GO.wall<-goseq(DEG.pwf, ID_vector, gene2cat=merge, method="Wallenius", use_genes_without_cat=TRUE)
# Using manually entered categories.
# Calculating the p-values...
# 'select()' returned 1:1 mapping between keys and columns
write.csv(GO.wall, file = "Output/GOSeq/ofav/ofav_GO_ALL_20211117.csv")

# Check GO.wall df
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

# Find significantly enriched GO terms 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")
enriched.GO.05 <- enriched.GO.05[order(-enriched.GO.05$numDEInCat),]
enriched.GO.05$term <- as.factor(enriched.GO.05$term)
head(enriched.GO.05)

# Subset enriched GO terms by ontology (BP, CC, MF) and save as csv. 
MF <- subset(enriched.GO.05, ontology=="MF")
MF <- MF[order(-MF$numDEInCat),]
#CC <- subset(enriched.GO.05, ontology=="CC") --- no sig enriched CC terms 
#CC <- CC[order(-CC$numDEInCat),] 
BP <- subset(enriched.GO.05, ontology=="BP")
BP <- BP[order(-BP$numDEInCat),]
write.csv(MF, file = "Output/GOSeq/ofav/ofav_MF_Sig_Enriched_GO.05_20211117.csv")
#write.csv(CC, file = "Output/GOSeq/ofav/ofav_CC_Sig_Enriched_GO.05_20211117.csv") 
write.csv(BP, file = "Output/GOSeq/ofav/ofav_BP_Sig_Enriched_GO.05_20211117.csv")
write.csv(enriched.GO.05, file = "Output/GOSeq/ofav/ofav_Sig_Enriched_GO.05_ALL_20211117.csv")
```

## Merge DEG gene ids and GO info
```{r}
# Merge gene ids and sig enriched GO terms 
colnames(merge) <- c("category", "gene_id")
merge_cat <- merge(enriched.GO.05, merge, by.x = "category") # contains gene ids and GO info 
#merge <- unique(merge)

# Merge DEG with merge 
merge_DEG <- merge(merge_cat, DEG, by = "gene_id") # contains GO info and gene counts for DEGs
```

## Add treatment info to df and save 
```{r}
# Read in DEG treatment info
ofav_treatment <- read.csv("Output/DESeq2/ofav/ofav_DEGs_AllTreatments_20211104.csv", header = TRUE)
ofav_treatment <- select(ofav_treatment, c("baseMean", "log2FoldChange", "lfcSE", "stat", "pvalue", "padj", "Treatment_Compare", "gene_id"))

# Merge treatment info with GO info 
merge_all <- merge(merge_DEG, ofav_treatment, by = "gene_id")
# yay now I have a df with all the info - GO info, gene ids, DESeq2 stats, gene counts
dim(merge_all) # 51 x 31
length(unique(merge_all$gene_id)) # 7 unique genes that are differentially expressed and have GO term info
length(unique(merge_all$category)) # 20 unique GO terms 
write.csv(merge_all, file = "Output/GOSeq/ofav/ofav_ByTreatment_GO.terms_20211117.csv")

# Subset enriched GO terms by ontology (BP, CC, MF) and save as csv. 
MF_ByTreatment <- subset(merge_all, ontology=="MF")
MF_ByTreatment <- MF_ByTreatment[order(-MF_ByTreatment$numDEInCat),]
#CC_ByTreatment <- subset(merge_all, ontology=="CC")
#CC_ByTreatment <- CC_ByTreatment[order(-CC_ByTreatment$numDEInCat),] # no sig enriched under CC
BP_ByTreatment <- subset(merge_all, ontology=="BP")
BP_ByTreatment <- BP_ByTreatment[order(-BP$numDEInCat),]
write.csv(MF_ByTreatment, file = "Output/GOSeq/ofav/ofav_MF_Sig_Enriched_ByTreatment_GO.05_20211117.csv")
#write.csv(CC_ByTreatment, file = "Output/GOSeq/ofav/mcav_CC_Sig_Enriched_ByTreatment_GO.05_20211117.csv")
write.csv(BP_ByTreatment, file = "Output/GOSeq/ofav/ofav_BP_Sig_Enriched_ByTreatment_GO.05_20211117.csv")
```

## Plot terms by ontology numbers 
```{r}
MFplot <- MF %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none"
  ) +
  xlab("") +
  ylab("") +
  ggtitle("Molecular Function") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
MFplot

# no CC plot 

BPplot <- BP %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none") +
  xlab("") +
  ylab("") +
  ggtitle("Biological Process") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
BPplot # lots of BP, hard to see. Could subset df and plot by BP dfs

# Save MF, CC, and BP plot in grid layout
GOplot <- grid.arrange(MFplot, BPplot, ncol=3, clip="off")
ggsave("Output/Figs/ofav/ofav_GOplot_05_FullAnnot_20211117.pdf", GOplot, width = 40, height = 40, units = c("in"))
```

# Combining MF and BP into one plot and order by pvalue
```{r}
GOplot2 <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,45) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot2
ggsave("Output/Figs/ofav/ofav_GOplot2_05_FullAnnot_20211117.pdf", GOplot2, width = 28, height = 28, units = c("in"))

# Combining into one plot and order by pvalue 
GOplot_pvalue <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=over_represented_pvalue) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=over_represented_pvalue), color="grey") +
  geom_text(aes(label = numDEInCat), hjust = -1, vjust = 0.5, size = 3) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,0.05) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("p-value") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot_pvalue
ggsave("Output/Figs/ofav/ofav_GOplot_pvalue_05_FullAnnot_20211117.pdf", GOplot_pvalue, width = 28, height = 28, units = c("in"))
```











