---
title: "ofav_GOSeq"
author: "jillashey"
date: "11/14/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# Load libraries
library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("gridExtra")
library("clusterProfiler")
library("DataCombine")
library("VennDiagram")
```

This code is using the ofav data. This code is also using the GO terms generated by Diamond (B2G), Swiss Prot (B2G), and IPS [here](https://github.com/JillAshey/FunctionalAnnotation/tree/main/final_Annotations) in Nov. 2021. 

## Read in all expressed plob genes (poverA = 0.85,5)
```{r}
# Obtain names of all expressed Mcav genes (poverA = 0.85,5), and all differentially expressed genes (p<0.05)
gcounts_filt_ofav <- read.csv("Output/DESeq2/ofav/ofav_counts_filtered.csv", header = T)
dim(gcounts_filt_ofav) # 18815 rows x 13
colnames(gcounts_filt_ofav)[1] <-"gene_id" # make colnames a true column called gene_id
length(unique(gcounts_filt_ofav$gene_id)) # 18815 total unique gene ids 
for ( col in 1:ncol(gcounts_filt_ofav)){
  colnames(gcounts_filt_ofav)[col] <-  gsub("X", "", colnames(gcounts_filt_ofav)[col])
}
head(gcounts_filt_ofav)
# Isolate LOC term
gcounts_filt_ofav$gene_id <- gsub("\\|.*", "", gcounts_filt_ofav$gene_id)
gcounts_filt_ofav$gene_id <- gsub(".*-", "", gcounts_filt_ofav$gene_id)
```

## Read in plob DEGs
```{r}
DEG_ofav <- read.csv("Output/DESeq2/ofav/ofav_unique.sig.list.csv", header = TRUE)
dim(DEG_ofav) # 62 x 16
colnames(DEG_ofav)[1] <-"gene_id" # make colnames a true column called gene_id
for ( col in 1:ncol(DEG_ofav)){
  colnames(DEG_ofav)[col] <-  gsub("X", "", colnames(DEG_ofav)[col])
}
head(DEG_ofav)
length(unique(DEG_ofav$gene_id)) # 16 total unique gene ids 
# Isolate LOC term
DEG_ofav$gene_id <- gsub("\\|.*", "", DEG_ofav$gene_id)
DEG_ofav$gene_id <- gsub(".*-", "", DEG_ofav$gene_id)
```

```{r}
# Read in length data (calculated directly from transcripts) and merge with filtered counts 
length_Ofav <- read.csv("Output/DESeq2/ofav/length.mRNA_Ofav.csv")
length_merge <- merge(length_Ofav, gcounts_filt_ofav, by = "gene_id")
```

# Build GOSEQ vector 
GOseq requires a vector of all genes, all differentially expressed genes, and gene lengths

## Make vectors
```{r}
#DEG<- filter(length_Acerv, gene_id %in% DEG_acerv_sub$gene_id) - for some reason, R keeps giving me error saying object 'gene_id' not found, so used another way below
DEG <- length_merge[length_merge$gene_id %in% DEG_ofav$gene_id, ]
dim(DEG) # 16 x 18
DEG_names <- as.vector(DEG$gene_id)
gene_vector <- as.integer(length_merge$gene_id%in%DEG_names)
names(gene_vector) <- length_merge$gene_id

# Make ID vector 
IDvector <- length_merge$gene_id

# Make length vector
lengthVector <- length_merge$length
```

















## Calculate Probability Weighting Function
```{r}
DEG.pwf<-nullp(gene_vector, IDvector, bias.data=lengthVector) #weight vector by length of gene

# plot looks weird...does this mean that there is not much length bias in the dataset?
```


















