---
title: "Acerv DESeq2"
author: "jillashey"
date: "11/21/2021"
output: html_document
---

#### Code for Francois sedimentation data. Acerv only samples analyzed here aligned against Acerv. STAR was read aligner with gff annotation file from Baums lab

## Load packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("ggrepel")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("gridExtra")
library("clusterProfiler")
library("DataCombine")
library("VennDiagram")
```

## Load gene count matrix
```{r}
# Load gene count matrix
acerv_counts <- read.csv("Output/DESeq2/acerv/acerv_gene_count_matrix.csv", header = TRUE, row.names = "gene_id")
dim(acerv_counts) # 33715 x 15
for ( col in 1:ncol(acerv_counts)){
  colnames(acerv_counts)[col] <-  sub(".fastq.trim.fq.Aligned.sortedByCoord.out.bam.merge.gtf", "", colnames(acerv_counts)[col]) # removing excess from col names
}
for ( col in 1:ncol(acerv_counts)){
  colnames(acerv_counts)[col] <-  gsub("X", "", colnames(acerv_counts)[col]) # remvoing X from col names
}
acerv_counts$ID <- rownames(acerv_counts)
acerv_counts$ID <- gsub("TU", "model", acerv_counts$ID)
rownames(acerv_counts) <- acerv_counts$ID
acerv_counts <- select(acerv_counts, -ID)
```

## Load metadata file 
```{r}
metadata_acerv <- read.csv("Data/acervmetadata_filtered.csv")
rownames(metadata_acerv) <- metadata_acerv$SampleID
```

## Subset count data for only ofav samples based on SampleID and make sure rows of metadata = cols of count data
```{r}
acerv_ID <- metadata_acerv$SampleID
count_acerv <- select(acerv_counts, all_of(acerv_ID))
all(rownames(metadata_acerv) %in% colnames(acerv_counts)) # must come out TRUE
```

## Filter reads by proportion of samples containing cutoff value
```{r}
filt <- filterfun(pOverA(0.85, 5)) # set filter values for P over A = 0.85 and 5 (ie 85% of samples )
tfil <- genefilter(acerv_counts, filt) # create filter for counts data 
keep <- acerv_counts[tfil,] # identify genes to keep based on filter
gn.keep <- rownames(keep)
acerv_counts_filt <- as.matrix(acerv_counts[which(rownames(acerv_counts) %in% gn.keep),]) 
#write.csv(acerv_counts_filt, "~/Desktop/acerv_counts_filtered.csv")
storage.mode(acerv_counts_filt) <- "integer" # stores count data as integer 
# Check to make sure rownames in metadata == colnames in counts data 
all(rownames(acerv_metadata) %in% colnames(acerv_counts_filt)) # must come out TRUE
# Set Treatment as a factor
acerv_metadata$Treatment <- factor(acerv_metadata$Treatment, levels = c("control", "Treatment1", "Treatment2", "Treatment3", "Treatment4"))
# create matrix that can be read in DESeq
data <- DESeqDataSetFromMatrix(countData = acerv_counts_filt, colData = acerv_metadata, design = ~ Treatment)

```

## Set Treatment as a factor
```{r}
acerv_metadata$Treatment <- factor(acerv_metadata$Treatment, levels = c("control", "Treatment1", "Treatment2", "Treatment3", "Treatment4"))
data <- DESeqDataSetFromMatrix(countData = acerv_counts_filt, colData = acerv_metadata, design = ~ Treatment)
```

## Expression visualization
First we are going to log-transform the data using a variance stabilizing transforamtion (vst). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects. To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4. Otherwise, there could be artefacts in our results.
```{r}
SF.data <- estimateSizeFactors(data) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than 4 to use vst
SF.data
print(sizeFactors(SF.data)) #view size factors
# size factors all less than 4, can use VST
vst <- vst(data, blind = FALSE) 
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
#    function: y = a/x + b, and a local regression fit was automatically substituted.
#    specify fitType='local' or 'mean' to avoid this message next time.
## Should I specify a fitType?
head(assay(vst), 3) # view data
sampleDists <- dist(t(assay(vst))) # calculate distance matrix
sampleDistMatrix <- as.matrix(sampleDists) # create distance matrix
rownames(sampleDistMatrix) <- colnames(vst) # assign row names
colnames(sampleDistMatrix) <- NULL # assign col names 
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255) # assign colors 
pheatmap(sampleDistMatrix, # plot matrix
         clustering_distance_rows = sampleDists, # cluster rows
         clustering_distance_cols = sampleDists, # cluster cols
         col=colors) # set colors
plotPCA(vst, intgroup = c("Treatment")) # plot PCA of samples with all data 

# pretty clear control outlier in PCA. going to proceed w/ outlier included, but will come back later to rerun w/o outlier 
```

```{r}
DEG.int <- DESeq(data) # run differential expression test by treatment (?) using wald test 
# estimating size factors
# estimating dispersions
# gene-wise dispersion estimates
# mean-dispersion relationship
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
# function: y = a/x + b, and a local regression fit was automatically substituted.
# specify fitType='local' or 'mean' to avoid this message next time.
# final dispersion estimates
# fitting model and testing
DEG.int.res <- results(DEG.int) # why does it say 'Wald test p-value: Treatment Treatment4 vs control' for DEG.int.res? Is it only looking at treatment 4 and control? In DESeq object created above, it says that design is Treatment
resultsNames(DEG.int) 
#[1] "Intercept"                       "Treatment_Treatment1_vs_control"
#[3] "Treatment_Treatment2_vs_control" "Treatment_Treatment3_vs_control"
#[5] "Treatment_Treatment4_vs_control"
```

## Differential expression comparisons by treatment 
### Compare control and T1
```{r}
DEG_control_vs_T1 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment1"))
DEG_control_vs_T1

DEG_control_vs_T1 <- as.data.frame(DEG_control_vs_T1) # make full results into a df
DEG_control_vs_T1["Treatment_Compare"] <- "CvsT1" # add treatment comparison col
DEG_control_vs_T1.sig.num <- sum(DEG_control_vs_T1$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_control_vs_T1.sig.num # 30 DEGs
DEG_control_vs_T1.sig <- subset(DEG_control_vs_T1, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T1.sig["Treatment_Compare"] <- "CvsT1" # add treatment comparison col
DEG_control_vs_T1.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T1.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T1.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T1.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library. Specify fit type???
# error from vst:
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
#    function: y = a/x + b, and a local regression fit was automatically substituted.
#    specify fitType='local' or 'mean' to avoid this message next time.
# Estimated rdf < 1.0; not estimating variance
DEG_control_vs_T1.sig.list <- as.data.frame(counts(DEG_control_vs_T1.sig.list))
DEG_control_vs_T1.sig.list_full <- cbind(DEG_control_vs_T1.sig, DEG_control_vs_T1.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T1.sig.list_full, "Output/DESeq2/acerv/acerv_C_vs_T1_DEG.csv")
```

### Compare control and T2
```{r}
DEG_control_vs_T2 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment2"))
DEG_control_vs_T2

DEG_control_vs_T2 <- as.data.frame(DEG_control_vs_T2) # make full results into a df
DEG_control_vs_T2["Treatment_Compare"] <- "CvsT2" # add treatment comparison col
DEG_control_vs_T2.sig.num <- sum(DEG_control_vs_T2$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_control_vs_T2.sig.num # 35 DEGs
DEG_control_vs_T2.sig <- subset(DEG_control_vs_T2, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T2.sig["Treatment_Compare"] <- "CvsT2" # add treatment comparison col
DEG_control_vs_T2.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T2.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T2.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T2.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library. Specify fit type???
# error from vst:
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
#    function: y = a/x + b, and a local regression fit was automatically substituted.
#    specify fitType='local' or 'mean' to avoid this message next time.
# Estimated rdf < 1.0; not estimating variance
DEG_control_vs_T2.sig.list <- as.data.frame(counts(DEG_control_vs_T2.sig.list))
DEG_control_vs_T2.sig.list_full <- cbind(DEG_control_vs_T2.sig, DEG_control_vs_T2.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T2.sig.list_full, "Output/DESeq2/acerv/acerv_C_vs_T2_DEG.csv")
```

### Compare control and T3
```{r}
DEG_control_vs_T3 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment3"))
DEG_control_vs_T3

DEG_control_vs_T3 <- as.data.frame(DEG_control_vs_T3) # make full results into a df
DEG_control_vs_T3["Treatment_Compare"] <- "CvsT3" # add treatment comparison col
DEG_control_vs_T3.sig.num <- sum(DEG_control_vs_T3$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_control_vs_T3.sig.num # 20 DEGs
DEG_control_vs_T3.sig <- subset(DEG_control_vs_T3, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T3.sig["Treatment_Compare"] <- "CvsT3" # add treatment comparison col
DEG_control_vs_T3.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T3.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T3.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T3.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library. Specify fit type???
# error from vst:
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
#    function: y = a/x + b, and a local regression fit was automatically substituted.
#    specify fitType='local' or 'mean' to avoid this message next time.
# Estimated rdf < 1.0; not estimating variance
DEG_control_vs_T3.sig.list <- as.data.frame(counts(DEG_control_vs_T3.sig.list))
DEG_control_vs_T3.sig.list_full <- cbind(DEG_control_vs_T3.sig, DEG_control_vs_T3.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T3.sig.list_full, "Output/DESeq2/acerv/acerv_C_vs_T3_DEG.csv")
```

### Compare control and T4
```{r}
DEG_control_vs_T4 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment4"))
DEG_control_vs_T4

DEG_control_vs_T4 <- as.data.frame(DEG_control_vs_T4) # make full results into a df
DEG_control_vs_T4["Treatment_Compare"] <- "CvsT4" # add treatment comparison col
DEG_control_vs_T4.sig.num <- sum(DEG_control_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_control_vs_T4.sig.num # 3 DEGs
DEG_control_vs_T4.sig <- subset(DEG_control_vs_T4, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T4.sig["Treatment_Compare"] <- "CvsT4" # add treatment comparison col
DEG_control_vs_T4.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T4.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T4.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T4.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library. Specify fit type???
# error from vst:
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
#    function: y = a/x + b, and a local regression fit was automatically substituted.
#    specify fitType='local' or 'mean' to avoid this message next time.
# Estimated rdf < 1.0; not estimating variance
DEG_control_vs_T4.sig.list <- as.data.frame(counts(DEG_control_vs_T4.sig.list))
DEG_control_vs_T4.sig.list_full <- cbind(DEG_control_vs_T4.sig, DEG_control_vs_T4.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T4.sig.list_full, "Output/DESeq2/acerv/acerv_C_vs_T4_DEG.csv")
```

### Compare T1 and T2
```{r}
DEG_T1_vs_T2 <- results(DEG.int, contrast = c("Treatment", "Treatment1", "Treatment2"))
DEG_T1_vs_T2

DEG_T1_vs_T2 <- as.data.frame(DEG_T1_vs_T2) # make full results into a df
DEG_T1_vs_T2["Treatment_Compare"] <- "T1vsT2" # add treatment comparison col
DEG_T1_vs_T2.sig.num <- sum(DEG_T1_vs_T2$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_T1_vs_T2.sig.num # 0 DEGs
```

### Compare T1 and T3
```{r}
DEG_T1_vs_T3 <- results(DEG.int, contrast = c("Treatment", "Treatment1", "Treatment3"))
DEG_T1_vs_T3

DEG_T1_vs_T3 <- as.data.frame(DEG_T1_vs_T3) # make full results into a df
DEG_T1_vs_T3["Treatment_Compare"] <- "T1vsT3" # add treatment comparison col
DEG_T1_vs_T3.sig.num <- sum(DEG_T1_vs_T3$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_T1_vs_T3.sig.num # 0 DEGs
```

### Compare T1 and T4
```{r}
DEG_T1_vs_T4 <- results(DEG.int, contrast = c("Treatment", "Treatment1", "Treatment4"))
DEG_T1_vs_T4

DEG_T1_vs_T4 <- as.data.frame(DEG_T1_vs_T4) # make full results into a df
DEG_T1_vs_T4["Treatment_Compare"] <- "T1vsT4" # add treatment comparison col
DEG_T1_vs_T4.sig.num <- sum(DEG_T1_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_T1_vs_T4.sig.num # 1 DEGs

DEG_T1_vs_T4.sig <- subset(DEG_T1_vs_T4, padj <0.05) # identify and subset significant pvalues
DEG_T1_vs_T4.sig["Treatment_Compare"] <- "T1vsT4" # add treatment comparison col
DEG_T1_vs_T4.sig.list <- data[which(rownames(data) %in% rownames(DEG_T1_vs_T4.sig)),] # subset list of significant genes from original count data 
DEG_T1_vs_T4.vst.sig <- varianceStabilizingTransformation(DEG_T1_vs_T4.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library. Specify fit type???
# error from vst:
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
#    function: y = a/x + b, and a local regression fit was automatically substituted.
#    specify fitType='local' or 'mean' to avoid this message next time.
# Estimated rdf < 1.0; not estimating variance
DEG_T1_vs_T4.sig.list <- as.data.frame(counts(DEG_T1_vs_T4.sig.list))
DEG_T1_vs_T4.sig.list_full <- cbind(DEG_T1_vs_T4.sig, DEG_T1_vs_T4.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_T1_vs_T4.sig.list_full, "Output/DESeq2/acerv/acerv_T1_vs_T4_DEG.csv")
```

### Compare T2 and T3
```{r}
DEG_T2_vs_T3 <- results(DEG.int, contrast = c("Treatment", "Treatment2", "Treatment3"))
DEG_T2_vs_T3

DEG_T2_vs_T3 <- as.data.frame(DEG_T2_vs_T3) # make full results into a df
DEG_T2_vs_T3["Treatment_Compare"] <- "T2vsT3" # add treatment comparison col
DEG_T2_vs_T3.sig.num <- sum(DEG_T2_vs_T3$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_T2_vs_T3.sig.num # 0 DEGs
```

### Compare T2 and T4
```{r}
DEG_T2_vs_T4 <- results(DEG.int, contrast = c("Treatment", "Treatment2", "Treatment4"))
DEG_T2_vs_T4

DEG_T2_vs_T4 <- as.data.frame(DEG_T2_vs_T4) # make full results into a df
DEG_T2_vs_T4["Treatment_Compare"] <- "T2vsT3" # add treatment comparison col
DEG_T2_vs_T4.sig.num <- sum(DEG_T2_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_T2_vs_T4.sig.num # 0 DEGs
```

### Compare T3 and T4
```{r}
DEG_T3_vs_T4 <- results(DEG.int, contrast = c("Treatment", "Treatment3", "Treatment4"))
DEG_T3_vs_T4

DEG_T3_vs_T4 <- as.data.frame(DEG_T3_vs_T4) # make full results into a df
DEG_T3_vs_T4["Treatment_Compare"] <- "T3vsT4" # add treatment comparison col
DEG_T3_vs_T4.sig.num <- sum(DEG_T3_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_T3_vs_T4.sig.num # 0 DEGs
```

## Make full list of differentially expressed genes and treatments 
```{r}
# Create gene_id from rownames
DEG_control_vs_T1.sig.list_full$gene_id <- rownames(DEG_control_vs_T1.sig.list_full)
rownames(DEG_control_vs_T1.sig.list_full) <- NULL
DEG_control_vs_T2.sig.list_full$gene_id <- rownames(DEG_control_vs_T2.sig.list_full)
rownames(DEG_control_vs_T2.sig.list_full) <- NULL
DEG_control_vs_T3.sig.list_full$gene_id <- rownames(DEG_control_vs_T3.sig.list_full)
rownames(DEG_control_vs_T3.sig.list_full) <- NULL
DEG_control_vs_T4.sig.list_full$gene_id <- rownames(DEG_control_vs_T4.sig.list_full)
rownames(DEG_control_vs_T4.sig.list_full) <- NULL
DEG_T1_vs_T4.sig.list_full$gene_id <- rownames(DEG_T1_vs_T4.sig.list_full)
rownames(DEG_T1_vs_T4.sig.list_full) <- NULL

# Find similar genes among treatments 
int1 <- intersect(DEG_control_vs_T1.sig.list_full$gene_id, DEG_control_vs_T2.sig.list_full$gene_id)
length(unique(int1)) # 15 genes shared between control_vs_T1 and control_vs_T2'
int2 <- intersect(DEG_control_vs_T1.sig.list_full$gene_id, DEG_control_vs_T3.sig.list_full$gene_id)
length(unique(int2)) # 0 genes shared between control_vs_T1 and control_vs_T3
int3 <- intersect(DEG_control_vs_T1.sig.list_full$gene_id, DEG_control_vs_T4.sig.list_full$gene_id)
length(unique(int3)) # 0 genes shared between control_vs_T1 and control_vs_T4


# bind all DEGs
DEGs.all <- rbind(DEG_control_vs_T1.sig.list_full,
                  DEG_control_vs_T2.sig.list_full,
                  DEG_control_vs_T3.sig.list_full,
                  DEG_control_vs_T4.sig.list_full,
                  DEG_T1_vs_T4.sig.list_full) # make sure column names are the same in all dfs before binding!
dim(DEGs.all) # 89 x 23
length(unique(DEGs.all$gene_id)) # 35 unique genes between all treatments 
#write.csv(DEGs.all, "Output/DESeq2/acerv/acerv_DEGs_AllTreatments_20211104.csv")

# Bind all genes
gene.all <- rbind(DEG_control_vs_T1,
                  DEG_control_vs_T2,
                  DEG_control_vs_T3,
                  DEG_control_vs_T4,
                  DEG_T1_vs_T4)
#write.csv(gene.all, "Output/DESeq2/acerv/acerv_AllGenes_AllTreatments_20211104.csv")
```

## 
```{r}
unique.sig.list <- data[which(rownames(data) %in% DEGs.all$DEGs), ] # subset list of sig transcripts from original count data
#write.csv(counts(unique.sig.list), file = "~/Desktop/pdam_unique.sig.list_20210326.csv")
SFtest <- estimateSizeFactors(unique.sig.list)
print(sizeFactors(SFtest))
unique.vst.sig <- varianceStabilizingTransformation(unique.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
# function: y = a/x + b, and a local regression fit was automatically substituted.
# specify fitType='local' or 'mean' to avoid this message next time.
```






