---
title: "acerv_DESeq2"
author: "jillashey"
date: "5/9/2022"
output: html_document
---

#### Code for Francois sedimentation data. Acerv only samples analyzed here aligned against Acerv STAR was read aligner and Acerv genomic info can be found [here](https://urldefense.proofpoint.com/v2/url?u=https-3A__usegalaxy.org_u_skitch_h_acervicornis-2Dgenome&d=DwMFaQ&c=dWz0sRZOjEnYSN4E4J0dug&r=6vvIGXnw7HQ0elnOAqPM4w&m=Rjen9vl1lZrdyO2SNwJ6j7w56Prx5pycPCwj0iwf4Yk&s=NRR8w0F5_lhn8Dw6U_oVk2bGIF50D94N7mPwx_XYZGQ&e=)

## Load packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(root.dir = '/Users/jillashey/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/', echo = TRUE)
getwd()
knitr::opts_knit$set(root.dir = '/tmp')

library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("ggrepel")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("gridExtra")
library("clusterProfiler")
library("DataCombine")
library("VennDiagram")
```



## Load gene count matrix
```{r}
getwd()

acerv_counts <- read.csv("../../Output/DESeq2/acerv/acerv_gene_count_matrix.csv", header = TRUE, row.names = "gene_id")
dim(acerv_counts) # 33715 x 15
for ( col in 1:ncol(acerv_counts)){
  colnames(acerv_counts)[col] <-  sub(".fastq.trim.fq.Aligned.sortedByCoord.out.bam.merge.gtf", "", colnames(acerv_counts)[col]) # removing excess from col names
}
for ( col in 1:ncol(acerv_counts)){
  colnames(acerv_counts)[col] <-  gsub("X", "", colnames(acerv_counts)[col]) # remvoing X from col names
}
# the data has some gene names with 'TU' and some gene names with 'model'. I thought the TU and model were the same genes, just with one word switched 
# out for the other in certain files. Shoot i need go back to all acerv data and redo. 
# okay so the names with model in them get removed downstream because their counts are all 0 because they are not genes? need to check w/ star and stringtie
# so i'll remove the rows with model now so it doesn't confound anything downstream. 
acerv_counts <- acerv_counts[!grepl("model", rownames(acerv_counts)),]
dim(acerv_counts) # 18958 x 15
```

## Load metadata file 
```{r}
getwd()

acerv_metadata <- read.csv("../../Data/acerv_metadata_filtered.csv", header = T)

# Make row names to be SampleID 
rownames(acerv_metadata) <- acerv_metadata$SampleID
```

## Remove outlier 
```{r}
## Try removing weird outliers in T1 and T2 and rerun
# Identified 24_T12_Ac_FM and 45_T41_Ac_SC_1 as outliers in previous script

# Subset acerv counts without the two outliers 
acerv_counts_sub <- select(acerv_counts, -c("24_T12_Ac_FM", "45_T41_Ac_SC_1"))

# Subset metadata to remove the two outliers 
rownames.remove <- c("24_T12_Ac_FM", "45_T41_Ac_SC_1")
acerv_metadata_sub <- acerv_metadata[!(rownames(acerv_metadata) %in% rownames.remove),]
```

## Subset count data for only acerv samples based on SampleID and make sure rows of metadata = cols of count data
```{r}
all(rownames(acerv_metadata_sub) %in% colnames(acerv_counts_sub)) # must come out TRUE
```

## Filter reads by proportion of samples containing cutoff value
```{r}
# Filter reads by proportion of samples containing cutoff value
filt <- filterfun(pOverA(0.85, 5)) # set filter values for P over A = 0.85 and 5 (ie keep if 85% of samples have a count >5)
tfil <- genefilter(acerv_counts_sub, filt) # create filter for counts data 
keep <- acerv_counts_sub[tfil,] # identify genes to keep based on filter
# the gene names with model in them would get filtered out here because they do not have any counts 
gn.keep <- rownames(keep)
# Based on filt info, keep only the genes that pass in acerv_counts_filt
acerv_counts_sub_filt <- as.matrix(acerv_counts_sub[which(rownames(acerv_counts_sub) %in% gn.keep),]) 
#write.csv(acerv_counts_sub_filt, "~/Desktop/acerv_counts_sub_filt_20210327.csv")
storage.mode(acerv_counts_sub_filt) <- "integer" # stores count data as integer 
# Check to make sure rownames in metadata == colnames in counts data 
all(rownames(acerv_metadata_sub) %in% colnames(acerv_counts_sub_filt)) # must come out TRUE
```

## Set Treatment as a factor
```{r}
acerv_metadata_sub$Treatment <- factor(acerv_metadata_sub$Treatment, levels = c("control", "Treatment1", "Treatment2", "Treatment3", "Treatment4"))
# create matrix that can be read by DESeq
data <- DESeqDataSetFromMatrix(countData = acerv_counts_sub_filt, colData = acerv_metadata_sub, design = ~ Treatment)
```

## Expression visualization
First we are going to log-transform the data using a variance stabilizing transforamtion (vst). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects. To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4. Otherwise, there could be artefacts in our results.
```{r}
# Estimate size factors to determine if we can use vst to transform our data. Size factors should be less than 4 to use vst
SF.data <- estimateSizeFactors(data) 
SF.data
print(sizeFactors(SF.data)) #view size factors
# size factors all less than 4, can use VST

# apply a variance stabilizing transforamtion to minimize effects of small counts and normalize wrt library size
vst <- vst(data, blind = FALSE) 
head(assay(vst), 3) # view data

# Using vst object, plot heat-map of sample-to-sample distances
sampleDists <- dist(t(assay(vst))) # calculate distance matrix
sampleDistMatrix <- as.matrix(sampleDists) # create distance matrix
rownames(sampleDistMatrix) <- colnames(vst) # assign row names
colnames(sampleDistMatrix) <- NULL # assign col names 
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255) # assign colors 
acerv_heatmap <- pheatmap(sampleDistMatrix, # plot distance matrix
         clustering_distance_rows = sampleDists, # cluster rows
         clustering_distance_cols = sampleDists, # cluster cols
         col=colors) # set colors
acerv_heatmap

# Using vst object, make PCA plot of samples 
vst_PCAdata <- plotPCA(vst, intgroup = c("Treatment"), returnData=TRUE) # create PCA loadings 
percentVar <- round(100*attr(vst_PCAdata, "percentVar")) #plot PCA of samples with all data
acerv_PCAplot <- ggplot(vst_PCAdata, aes(PC1, PC2, color=Treatment)) + 
   geom_point(size=3) +
   geom_text(aes(label=name),hjust=0, vjust=0) +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  scale_color_manual(values = c(control="gray", Treatment1="darkslategray2", Treatment2="darkslategray3", Treatment3="darkslategray4", Treatment4="darkslategray")) +
   coord_fixed() +
   ggtitle("A. cervicornis - all genes") +
   theme_bw() + #Set background color
   theme(panel.border = element_blank(), # Set border
         #panel.grid.major = element_blank(), #Set major gridlines
         #panel.grid.minor = element_blank(), #Set minor gridlines
         axis.line = element_line(colour = "black"), #Set axes color
         plot.background=element_blank()) #Set the plot background
acerv_PCAplot 

# save plots 
acerv_heatmap_PCA <- grid.arrange(acerv_PCAplot, acerv_heatmap[[4]], nrow=1, ncol = 2, clip="off")
#ggsave("Output/Figs/acerv/acerv_heatmap_PCA_AllGenes_20220520.jpeg", acerv_heatmap_PCA, width = 20, height = 25, units = "cm")
#ggsave("Output/Figs/acerv/acerv_heatmap_PCA_AllGenes_20220520.pdf", acerv_heatmap_PCA, width = 20, height = 25, units = "cm")
```

## Run DESeq2
```{r}
# Differential gene expression analysis 
DEG.int <- DESeq(data) # run differential expression test by treatment (?) using wald test 
# estimating size factors
# estimating dispersions
# gene-wise dispersion estimates
# mean-dispersion relationship
# final dispersion estimates
# fitting model and testing
DEG.int.res <- results(DEG.int) # save DE results ; why does it say 'Wald test p-value: Treatment Treatment4 vs control' for DEG.int.res? Is it only looking at treatment 4 and control? In DESeq object created above, it says that design is Treatment
resultsNames(DEG.int) # view DE results 
#[1] "Intercept"                       "Treatment_Treatment1_vs_control" "Treatment_Treatment2_vs_control"
#[4] "Treatment_Treatment3_vs_control" "Treatment_Treatment4_vs_control"
# NAs in padj: https://hbctraining.github.io/DGE_workshop/lessons/05_DGE_DESeq2_analysis2.html 
# http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#why-are-some-p-values-set-to-na
```

## Differential expression comparisons by treatment 
### Compare control and T1
```{r}
DEG_control_vs_T1 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment1")) # results of DESeq2 comparing C and T1
DEG_control_vs_T1 
DEG_control_vs_T1 <- as.data.frame(DEG_control_vs_T1) # make results into a df
DEG_control_vs_T1["Treatment_Compare"] <- "CvsT1" # adding treatment comparison column
#write.csv(DEG_control_vs_T1, file = "~/Desktop/acerv_sub_control_vs_T1_all_genes_20210219.csv") # maybe include gene counts too?
DEG_control_vs_T1.sig.num <- sum(DEG_control_vs_T1$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_control_vs_T1.sig.num
# 70 DEGs
# not sure why this # of DEGs differs from the one that includes the outliers (DEG=19). Trying to figure out why this is...Removed a sample from the pile? idk
DEG_control_vs_T1.sig <- subset(DEG_control_vs_T1, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T1.sig["Treatment_Compare"] <- "CvsT1" # adding treatment comparison column
DEG_control_vs_T1.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T1.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T1.sig.list <- as.data.frame(counts(DEG_control_vs_T1.sig.list)) # make list of sig gene counts into a df
DEG_control_vs_T1.sig.list_full <- cbind(DEG_control_vs_T1.sig, DEG_control_vs_T1.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T1.sig.list_full, file = "~/Desktop/acerv_sub_control_vs_T1_DEG_full_20210219.csv") # write out csv
DEG_control_vs_T1.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T1.sig)),] # turn back into formal class DESeqTransform or else vst will not run
DEG_control_vs_T1.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T1.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
## before running vst above, turn DEG_control_vs_T1.sig.list back into formal class DESeqTransform by running line DEG_control_vs_T1.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T1.sig)),]
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
# function: y = a/x + b, and a local regression fit was automatically substituted.
# specify fitType='local' or 'mean' to avoid this message next time.
```

### Compare control and T2
```{r}
DEG_control_vs_T2 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment2")) # results of DESeq2 comparing C and T2
DEG_control_vs_T2 
DEG_control_vs_T2 <- as.data.frame(DEG_control_vs_T2) # make results into a df
DEG_control_vs_T2["Treatment_Compare"] <- "CvsT2" # adding treatment comparison column
#write.csv(DEG_control_vs_T2, file = "~/Desktop/acerv_sub_control_vs_T2_all_genes_20210219.csv") # maybe include gene counts too?
DEG_control_vs_T2.sig.num <- sum(DEG_control_vs_T2$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_control_vs_T2.sig.num
# 141 DEGs
# not sure why this # of DEGs differs from the one that includes the outliers (DEG=35). Trying to figure out why this is...
# Removed a sample from the pile? idk
DEG_control_vs_T2.sig <- subset(DEG_control_vs_T2, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T2.sig["Treatment_Compare"] <- "CvsT2" # adding treatment comparison column
DEG_control_vs_T2.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T2.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T2.sig.list <- as.data.frame(counts(DEG_control_vs_T2.sig.list)) # make list of sig gene counts into a df
DEG_control_vs_T2.sig.list_full <- cbind(DEG_control_vs_T2.sig, DEG_control_vs_T2.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T2.sig.list_full, file = "~/Desktop/acerv_sub_control_vs_T2_DEG_full_20210219.csv") # write out csv
DEG_control_vs_T2.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T2.sig)),] # turn back into formal class DESeqTransform or else vst will not run
DEG_control_vs_T2.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T2.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
```

### Compare control and T3
```{r}
DEG_control_vs_T3 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment3")) # results of DESeq2 comparing C and T2
DEG_control_vs_T3 
DEG_control_vs_T3 <- as.data.frame(DEG_control_vs_T3) # make results into a df
DEG_control_vs_T3["Treatment_Compare"] <- "CvsT3" # adding treatment comparison column
#write.csv(DEG_control_vs_T3, file = "~/Desktop/acerv_sub_control_vs_T3_all_genes_20210219.csv") # maybe include gene counts too?
DEG_control_vs_T3.sig.num <- sum(DEG_control_vs_T3$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_control_vs_T3.sig.num
# 56 DEGs
# not sure why this # of DEGs differs from the one that includes the outliers (DEG=20). Trying to figure out why this is...
# Removed a sample from the pile? idk
DEG_control_vs_T3.sig <- subset(DEG_control_vs_T3, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T3.sig["Treatment_Compare"] <- "CvsT3" # adding treatment comparison column
DEG_control_vs_T3.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T3.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T3.sig.list <- as.data.frame(counts(DEG_control_vs_T3.sig.list)) # make list of sig gene counts into a df
DEG_control_vs_T3.sig.list_full <- cbind(DEG_control_vs_T3.sig, DEG_control_vs_T3.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T3.sig.list_full, file = "~/Desktop/acerv_sub_control_vs_T3_DEG_full_20210219.csv") # write out csv
DEG_control_vs_T3.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T3.sig)),] # turn back into formal class DESeqTransform or else vst will not run
DEG_control_vs_T3.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T3.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
```

### Compare control and T4
```{r}
DEG_control_vs_T4 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment4")) # results of DESeq2 comparing C and T2
DEG_control_vs_T4
DEG_control_vs_T4 <- as.data.frame(DEG_control_vs_T4) # make results into a df
DEG_control_vs_T4["Treatment_Compare"] <- "CvsT4" # adding treatment comparison column
#write.csv(DEG_control_vs_T4, file = "~/Desktop/acerv_sub_control_vs_T4_all_genes_20210219.csv") # maybe include gene counts too?
DEG_control_vs_T4.sig.num <- sum(DEG_control_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_control_vs_T4.sig.num
# 74 DEGs
# not sure why this # of DEGs differs from the one that includes the outliers (DEG=3). Trying to figure out why this is... Removed a sample from the pile? idk
DEG_control_vs_T4.sig <- subset(DEG_control_vs_T4, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T4.sig["Treatment_Compare"] <- "CvsT4" # adding treatment comparison column
DEG_control_vs_T4.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T4.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T4.sig.list <- as.data.frame(counts(DEG_control_vs_T4.sig.list)) # make list of sig gene counts into a df
DEG_control_vs_T4.sig.list_full <- cbind(DEG_control_vs_T4.sig, DEG_control_vs_T4.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T4.sig.list_full, file = "~/Desktop/acerv_sub_control_vs_T4_DEG_full_20210219.csv") # write out csv
DEG_control_vs_T4.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T4.sig)),] # turn back into formal class DESeqTransform or else vst will not run
DEG_control_vs_T4.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T4.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
# function: y = a/x + b, and a local regression fit was automatically substituted.
# specify fitType='local' or 'mean' to avoid this message next time.
```

### Compare T1 and T2
```{r}
DEG_T1_vs_T2 <- results(DEG.int, contrast = c("Treatment", "Treatment1", "Treatment2")) # results of DESeq2 comparing C and T2
DEG_T1_vs_T2 
DEG_T1_vs_T2 <- as.data.frame(DEG_T1_vs_T2) # make results into a df
DEG_T1_vs_T2["Treatment_Compare"] <- "T1vsT2" # adding treatment comparison column
#write.csv(DEG_T1_vs_T2, file = "~/Desktop/acerv_sub_control_vs_T2_all_genes_20210219.csv") # maybe include gene counts too?
DEG_T1_vs_T2.sig.num <- sum(DEG_T1_vs_T2$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T1_vs_T2.sig.num
# 51 DEGs
# not sure why this # of DEGs differs from the one that includes the outliers (DEG=0). Trying to figure out why this is...Removed a sample from the pile? idk
DEG_T1_vs_T2.sig <- subset(DEG_T1_vs_T2, padj <0.05) # identify and subset significant pvalues
DEG_T1_vs_T2.sig["Treatment_Compare"] <- "T1vsT2" # adding treatment comparison column
DEG_T1_vs_T2.sig.list <- data[which(rownames(data) %in% rownames(DEG_T1_vs_T2.sig)),] # subset list of significant genes from original count data 
DEG_T1_vs_T2.sig.list <- as.data.frame(counts(DEG_T1_vs_T2.sig.list)) # make list of sig gene counts into a df
DEG_T1_vs_T2.sig.list_full <- cbind(DEG_T1_vs_T2.sig, DEG_T1_vs_T2.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_T1_vs_T2.sig.list_full, file = "~/Desktop/acerv_sub_T1_vs_T2_DEG_full_20210219.csv") # write out csv
DEG_T1_vs_T2.sig.list <- data[which(rownames(data) %in% rownames(DEG_T1_vs_T2.sig)),] # turn back into formal class DESeqTransform or else vst will not run
DEG_T1_vs_T2.vst.sig <- varianceStabilizingTransformation(DEG_T1_vs_T2.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
# function: y = a/x + b, and a local regression fit was automatically substituted.
# specify fitType='local' or 'mean' to avoid this message next time.
```

### Compare T1 and T3
```{r}
DEG_T1_vs_T3 <- results(DEG.int, contrast = c("Treatment", "Treatment1", "Treatment3")) # results of DESeq2 comparing C and T2
DEG_T1_vs_T3
DEG_T1_vs_T3 <- as.data.frame(DEG_T1_vs_T3) # make results into a df
DEG_T1_vs_T3["Treatment_Compare"] <- "T1vsT3" # adding treatment comparison column
#write.csv(DEG_T1_vs_T3, file = "~/Desktop/acerv_sub_T1_vs_T3_all_genes_20210219.csv") # maybe include gene counts too?
DEG_T1_vs_T3.sig.num <- sum(DEG_T1_vs_T3$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T1_vs_T3.sig.num
# 0 DEGs
```

### Compare T1 and T4
```{r}
DEG_T1_vs_T4 <- results(DEG.int, contrast = c("Treatment", "Treatment1", "Treatment4")) # results of DESeq2 comparing C and T2
DEG_T1_vs_T4
DEG_T1_vs_T4 <- as.data.frame(DEG_T1_vs_T4) # make results into a df
DEG_T1_vs_T4["Treatment_Compare"] <- "T1vsT4" # adding treatment comparison column
#write.csv(DEG_T1_vs_T4, file = "~/Desktop/acerv_sub_T1_vs_T4_all_genes_20210219.csv") # maybe include gene counts too?
DEG_T1_vs_T4.sig.num <- sum(DEG_T1_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T1_vs_T4.sig.num
# 0 DEGs
```

### Compare T2 and T3
```{r}
DEG_T2_vs_T3 <- results(DEG.int, contrast = c("Treatment", "Treatment2", "Treatment3")) # results of DESeq2 comparing C and T2
DEG_T2_vs_T3
DEG_T2_vs_T3 <- as.data.frame(DEG_T2_vs_T3) # make results into a df
DEG_T2_vs_T3["Treatment_Compare"] <- "T12sT3" # adding treatment comparison column
#write.csv(DEG_T2_vs_T3, file = "~/Desktop/acerv_sub_T2_vs_T3_all_genes_20210219.csv") # maybe include gene counts too?
DEG_T2_vs_T3.sig.num <- sum(DEG_T2_vs_T3$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T2_vs_T3.sig.num
# 3 DEGs
DEG_T2_vs_T3.sig <- subset(DEG_T2_vs_T3, padj <0.05) # identify and subset significant pvalues
DEG_T2_vs_T3.sig["Treatment_Compare"] <- "T2vsT3" # adding treatment comparison column
DEG_T2_vs_T3.sig.list <- data[which(rownames(data) %in% rownames(DEG_T2_vs_T3.sig)),] # subset list of significant genes from original count data 
DEG_T2_vs_T3.sig.list <- as.data.frame(counts(DEG_T2_vs_T3.sig.list)) # make list of sig gene counts into a df
DEG_T2_vs_T3.sig.list_full <- cbind(DEG_T2_vs_T3.sig, DEG_T2_vs_T3.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_T2_vs_T3.sig.list_full, file = "~/Desktop/acerv_sub_T2_vs_T3_DEG_full_20210219.csv") # write out csv
DEG_T1_vs_T2.sig.list <- data[which(rownames(data) %in% rownames(DEG_T1_vs_T2.sig)),] # turn back into formal class DESeqTransform or else vst will not run
DEG_T1_vs_T2.vst.sig <- varianceStabilizingTransformation(DEG_T1_vs_T2.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
# function: y = a/x + b, and a local regression fit was automatically substituted.
# specify fitType='local' or 'mean' to avoid this message next time.
```

### Compare T2 and T4
```{r}
DEG_T2_vs_T4 <- results(DEG.int, contrast = c("Treatment", "Treatment2", "Treatment4")) # results of DESeq2 comparing C and T2
DEG_T2_vs_T4
DEG_T2_vs_T4 <- as.data.frame(DEG_T2_vs_T4) # make results into a df
DEG_T2_vs_T4["Treatment_Compare"] <- "T2vsT4" # adding treatment comparison column
#write.csv(DEG_T2_vs_T4, file = "~/Desktop/acerv_sub_T2_vs_T4_all_genes_20210219.csv") # maybe include gene counts too?
DEG_T2_vs_T4.sig.num <- sum(DEG_T2_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T2_vs_T4.sig.num
# 0 DEGs
```

### Compare T3 and T4
```{r}
# Compare T3 vs T4
DEG_T3_vs_T4 <- results(DEG.int, contrast = c("Treatment", "Treatment3", "Treatment4")) # results of DESeq2 comparing C and T2
DEG_T3_vs_T4
DEG_T3_vs_T4 <- as.data.frame(DEG_T3_vs_T4) # make results into a df
DEG_T3_vs_T4["Treatment_Compare"] <- "T3vsT4" # adding treatment comparison column
#write.csv(DEG_T3_vs_T4, file = "~/Desktop/acerv_sub_T3_vs_T4_all_genes_20210219.csv") # maybe include gene counts too?
DEG_T3_vs_T4.sig.num <- sum(DEG_T3_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T3_vs_T4.sig.num
# 0 DEGs
```

## Make full list of differentially expressed genes and treatments 
```{r}
DEG_control_vs_T1.sig.list_full$gene_id <- rownames(DEG_control_vs_T1.sig.list_full)
rownames(DEG_control_vs_T1.sig.list_full) <- NULL
DEG_control_vs_T2.sig.list_full$gene_id <- rownames(DEG_control_vs_T2.sig.list_full)
rownames(DEG_control_vs_T2.sig.list_full) <- NULL
DEG_control_vs_T3.sig.list_full$gene_id <- rownames(DEG_control_vs_T3.sig.list_full)
rownames(DEG_control_vs_T3.sig.list_full) <- NULL
DEG_control_vs_T4.sig.list_full$gene_id <- rownames(DEG_control_vs_T4.sig.list_full)
rownames(DEG_control_vs_T4.sig.list_full) <- NULL
DEG_T1_vs_T2.sig.list_full$gene_id <- rownames(DEG_T1_vs_T2.sig.list_full)
rownames(DEG_T1_vs_T2.sig.list_full) <- NULL
DEG_T2_vs_T3.sig.list_full$gene_id <- rownames(DEG_T2_vs_T3.sig.list_full)
rownames(DEG_T2_vs_T3.sig.list_full) <- NULL

DEGs.all <- rbind(DEG_control_vs_T1.sig.list_full, 
                  DEG_control_vs_T2.sig.list_full,
                  DEG_control_vs_T3.sig.list_full, 
                  DEG_control_vs_T4.sig.list_full,
                  DEG_T1_vs_T2.sig.list_full,
                  DEG_T2_vs_T3.sig.list_full)
```

## Identify unique genes from DEGs from treatments 
```{r}
dim(DEGs.all) # 395 x 21
length(unique(DEGs.all$gene_id)) # 215 unique genes between all treatments 
#write.csv(DEGs.all, file = "~/Desktop/acerv_sub_DEGs.all_treatment_20210219.csv")
```

## Look at size factors and run vst normalization
```{r}
unique.sig.list <- data[which(rownames(data) %in% DEGs.all$gene_id), ] # subset list of sig transcripts from original count data
#write.csv(counts(unique.sig.list), file = "~/Desktop/acerv_sub_unique.sig.list_20210219.csv")
SFtest <- estimateSizeFactors(unique.sig.list)
print(sizeFactors(SFtest))
unique.vst.sig <- varianceStabilizingTransformation(unique.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
```


# PCA plot of DEGs
```{r}
getwd()

acerv_sub_DEG_PCA <- plotPCA(unique.vst.sig, intgroup = c("Treatment"), returnData=TRUE)
percentVar_pca_acerv_sub <- round(100*attr(acerv_sub_DEG_PCA, "percentVar")) #plot PCA of samples with all data

acerv_sub_DEG_PCA_plot <- ggplot(acerv_sub_DEG_PCA, aes(PC1, PC2, color = Treatment)) +
  geom_point(size=6) +
  xlab(paste0("PC1: ",percentVar_pca_acerv_sub[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_pca_acerv_sub[2],"% variance")) +
  scale_color_manual(values = c(control="gray", Treatment1="darkslategray2", Treatment2="darkslategray3", Treatment3="darkslategray4", Treatment4="darkslategray")) +
  #coord_fixed() +
  ggtitle(label = "A. cervicornis") +
  theme_bw() + #Set background color
  theme(legend.text = element_text(size=13), 
        #legend.position="none",
        plot.background = element_blank(),
        #legend.title = element_text(size=18, face="bold"), 
        legend.title=element_blank(),
        axis.text = element_text(size=13), 
        axis.title = element_text(size=15,  face="bold"), 
        axis.title.y = element_text(vjust=-1.5),
        plot.title = element_text(size = 20, face = "italic", hjust = 0.5))
acerv_sub_DEG_PCA_plot # PCA plot is of differentially expressed genes only
PC.info <- acerv_sub_DEG_PCA_plot$data
ggsave("../../Output/Figs/acerv/acerv_DEGs_PCA_20230906.jpeg", acerv_sub_DEG_PCA_plot, width = 20, height = 15, units = "cm")
ggsave("../../Output/Figs/acerv/acerv_DEGs_PCA_20230906.pdf", acerv_sub_DEG_PCA_plot, width = 20, height = 15, units = "cm")
```




