# Title: Sediment stress - acerv GOseq
# Author: Jill Ashey
# date: 1/25/21

### Gene Ontology Analysis of Differentially Expressed Genes in Acerv
# Using updated GO terms (terms generated by Blast2GO, Interproscan, and Uniprot)

# This script is using subsetted acerv data (removed samples 24 and 45)


# Load packages 
library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("forcats")
library("gridExtra")



### Code below uses my full annotation file 

####### Subsetted samples GO analysis

# Obtain names of all expressed Acerv genes (poverA = 0.85,5), and all differentially expressed planuala genes (p<0.05)
gcounts_filt_acerv_sub <- read.csv("Output/DESeq2/acerv/acerv_counts_sub_filt_20210327.csv", header = TRUE)
dim(gcounts_filt_acerv_sub) # 9054 rows x 14
for ( col in 1:ncol(gcounts_filt_acerv_sub)){
  colnames(gcounts_filt_acerv_sub)[col] <-  gsub("X", "", colnames(gcounts_filt_acerv_sub)[col])
}
colnames(gcounts_filt_acerv_sub)[1] <-"gene_id"
head(gcounts_filt_acerv_sub)
gcounts_filt_acerv_sub$gene_id <- gsub("TU", "model", gcounts_filt_acerv_sub$gene_id) # sub model for TU to match gene names in annot and interpro files

# Load DEGs
DEG_acerv_sub <- read.csv("~/Desktop/acerv_sub_unique.sig.list_20210219.csv", header = TRUE)
# use the file or acerv_sub_DEGs.all_treatment_20210219.csv?
dim(DEG_acerv_sub) # 215 x 14
for ( col in 1:ncol(DEG_acerv_sub)){
  colnames(DEG_acerv_sub)[col] <-  gsub("X", "", colnames(DEG_acerv_sub)[col])
}
colnames(DEG_acerv_sub)[1] <-"gene_id"
head(DEG_acerv_sub)
DEG_acerv_sub$gene_id <- gsub("TU", "model", DEG_acerv_sub$gene_id) # sub model for TU to match gene names in annot and interpro files

# Read in length data (calculated directly from transcripts) and merge with filtered counts 
length_Acerv <- read_csv("Desktop/length.mRNA_Acerv.csv")
length_merge <- merge(length_Acerv, gcounts_filt_acerv_sub, by = "gene_id")




#### Build GOSEQ vector 
#GOseq requires a vector of all genes, all differentially expressed genes, and gene lengths
# Make DEG/gene vector 
DEG <- filter(length_merge, gene_id%in%DEG_acerv_sub$gene_id) #make vector of differentially expressed genes
dim(DEG) # 215 rows 
DEG_names <- as.vector(DEG$gene_id)
gene_vector=as.integer(length_merge$gene_id%in%DEG_names)
names(gene_vector)=length_merge$gene_id

# Make ID vector 
IDvector <- length_merge$gene_id

# Make length vector
lengthVector <- length_merge$length

# Calculate Probability Weighting Function
DEG.pwf<-nullp(gene_vector, IDvector, bias.data=lengthVector) #weight vector by length of gene
# plot looks weird...does this mean that there is not much length bias in the dataset?







#### Prepare GO term dataframe 
annot <- read.csv("Desktop/PutnamLab/Repositories/FunctionalAnnotation/FunctionalAnnotation/final_Annotations/acerv_FullAnnot.csv", header = TRUE)
GO.annot <- select(annot, gene_id, GO.IDs)
GO.annot$GO.IDs <- gsub(",", ";", GO.annot$GO.IDs)
splitted <- strsplit(as.character(GO.annot$GO.ID), ";") #split into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.annot$gene_id, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row
colnames(GO.terms) <- c("gene_id", "GO.ID")
GO.terms$GO.ID <- gsub("F:", "", GO.terms$GO.ID)
GO.terms$GO.ID <- gsub("P:", "", GO.terms$GO.ID)
GO.terms$GO.ID <- gsub("C:", "", GO.terms$GO.ID)
GO.terms$GO.ID <- gsub(" ", "", GO.terms$GO.ID)
GO.terms[GO.terms == "NA"] <- NA

head(GO.terms)
tail(GO.terms)
GO.terms$GO.ID<- as.character(GO.terms$GO.ID)
GO.terms$GO.ID <- replace_na(GO.terms$GO.ID, "unknown")
GO.terms$GO.ID <- as.factor(GO.terms$GO.ID)
GO.terms$gene_id <- as.factor(GO.terms$gene_id)
GO.terms <- unique(GO.terms)

dim(GO.terms) # 44817 x 2
head(GO.terms, 10)
tail(GO.terms, 10)



##Find enriched GO terms, "selection-unbiased testing for category enrichment amongst significantly expressed genes for RNA-seq data"
GO.wall_FullAnnot<-goseq(DEG.pwf, IDvector, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)
# Using manually entered categories.
# Calculating the p-values...
# 'select()' returned 1:1 mapping between keys and columns
write.csv(GO.wall_FullAnnot, file = "~/Desktop/acerv_sub_GO_ALL_FullAnnot_20210222.csv")

# Find significantly enriched GO terms 
acerv.GO.05 <- GO.wall_FullAnnot$category[GO.wall_FullAnnot$over_represented_pvalue<.05]
acerv.GO.05 <- as.data.frame(acerv.GO.05)
colnames(acerv.GO.05) <- c("category")
acerv.GO.05 <- merge(acerv.GO.05, GO.wall_FullAnnot, by="category")
acerv.GO.05 <- acerv.GO.05[order(acerv.GO.05$ontology, acerv.GO.05$over_represented_pvalue, -acerv.GO.05$numDEInCat),]
#Subset enriched GO terms by category and save as csv
MF <- subset(acerv.GO.05, ontology=="MF")
MF <- MF[order(-MF$numDEInCat),]
#CC <- subset(acerv.GO.05, ontology=="CC") - no CC significantly enriched terms 
#CC <- CC[order(-CC$numDEInCat),] 
BP <- subset(acerv.GO.05, ontology=="BP")
BP <- BP[order(-BP$numDEInCat),]
write.csv(MF, file = "~/Desktop/acerv_sub_MF_Sig_Enriched_GO.05_FullAnnot_20210222.csv")
#write.csv(CC, file = "~/Desktop/acerv_CC_Sig_Enriched_GO.05_FullAnnot_20210219.csv") 
write.csv(BP, file = "~/Desktop/acerv_sub_BP_Sig_Enriched_GO.05_FullAnnot_20210222.csv")
write.csv(acerv.GO.05, file = "~/Desktop/acerv_sub_Sig_Enriched_GO.05_ALL_FullAnnot_2021022.csv")

# Merge gene ids and sig enriched GO terms 
colnames(GO.terms) <- c("gene_id", "category")
merge <- merge(acerv.GO.05, GO.terms, by.x = "category") # contains gene ids and GO info 

# Merge DEG with merge 
merge_again <- merge(merge, DEG, by = "gene_id") # contains GO info and gene counts for DEGs

# Read in DEG treatment info
acerv_treatment <- read.csv("~/Desktop/acerv_sub_DEGs.all_treatment_20210219.csv", header = TRUE)
acerv_treatment <- select(acerv_treatment, c("baseMean", "log2FoldChange", "lfcSE", "stat", "pvalue", "padj", "Treatment_Compare", "gene_id"))
acerv_treatment$gene_id <- gsub("TU", "model", acerv_treatment$gene_id)

# Merge treatment info with merge_again 
merge_all <- merge(merge_again, acerv_treatment, by = "gene_id")
# yay now I have a df with all the info - GO info, gene ids, DESeq2 stats, gene counts
dim(merge_all) # 24 x 29
length(unique(merge_all$gene_id)) # 13 unique genes that are differentially expressed and have GO term info
length(unique(merge_all$category)) # 15 unique GO terms 
write.csv(merge_all, file = "~/Desktop/acerv_sub_ByTreatment_GO.terms_FullAnnot_20210222.csv")
MF_ByTreatment <- subset(merge_all, ontology=="MF")
MF_ByTreatment <- MF_ByTreatment[order(-MF_ByTreatment$numDEInCat),]
#CC_ByTreatment <- subset(merge_all, ontology=="CC")
#CC_ByTreatment <- CC_ByTreatment[order(-CC_ByTreatment$numDEInCat),] # no sig enriched under CC
BP_ByTreatment <- subset(merge_all, ontology=="BP")
BP_ByTreatment <- BP_ByTreatment[order(-BP$numDEInCat),]
write.csv(MF_ByTreatment, file = "~/Desktop/acerv_sub_MF_Sig_Enriched_ByTreatment_GO.05_FullAnnot_20210222.csv")
#write.csv(CC_ByTreatment, file = "~/Desktop/acerv_CC_Sig_Enriched_ByTreatment_GO.05_20210208.csv")
write.csv(BP_ByTreatment, file = "~/Desktop/acerv_sub_BP_Sig_Enriched_ByTreatment_GO.05_FullAnnot_20210222.csv")

# Plot terms by number of differentially expressed functions
MFplot <- MF %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none"
  ) +
  xlab("") +
  ylab("") +
  ggtitle("Molecular Function") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
MFplot

# CCplot <- CC %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
#   ggplot( aes(x=term, y=numDEInCat) ) +
#   geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
#   geom_point(size=3, color="#69b3a2") +
#   coord_flip() +
#   theme(
#     panel.grid.minor.y = element_blank(),
#     panel.grid.major.y = element_blank(),
#     legend.position="none"
#   ) +
#   xlab("") +
#   ylab("") +
#   ggtitle("Cellular Component") + #add a main title
#   theme(plot.title = element_text(face = 'bold',
#                                   size = 12,
#                                   hjust = 0)) +
#   theme_bw() + #Set background color
#   theme(panel.border = element_blank(), # Set border
#         panel.grid.major = element_blank(), #Set major gridlines
#         panel.grid.minor = element_blank(), #Set minor gridlines
#         axis.line = element_line(colour = "black"), #Set axes color
#         plot.background=element_blank())#Set the plot background
# CCplot -- no sig enriched CC terms 

BPplot <- BP %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none") +
  xlab("") +
  ylab("") +
  ggtitle("Biological Process") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
BPplot

# Save MF, CC, and BP plot in grid layout
GOplot <- grid.arrange(MFplot, BPplot, ncol=3, clip="off")
ggsave("~/Desktop/acerv_sub_GOplot_05_FullAnnot_20210222.pdf", GOplot, width = 21, height = 21, units = c("in"))


# Combining MF, CC, and BP into one plot and order by pvalue
GOplot2 <- acerv.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,45) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot2
ggsave("~/Desktop/acerv_sub_GOplot2_05_FullAnnot_20210222.pdf", GOplot2, width = 28, height = 28, units = c("in"))

# Combining into one plot and order by pvalue 
GOplot_pvalue <- acerv.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=over_represented_pvalue) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=over_represented_pvalue), color="grey") +
  geom_text(aes(label = numDEInCat), hjust = -1, vjust = 0.5, size = 3) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,0.05) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("p-value") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot_pvalue
ggsave("~/Desktop/acerv_sub_GOplot_pvalue_05_FullAnnot_20210222.pdf", GOplot_pvalue, width = 28, height = 28, units = c("in"))

















































############ Code below is using GO terms from my annotations (IPS, b2g, uniprot + old), gff reference file, and gtf stringtie file 
### OLD CODE AS OF 20210219

# Obtain names of all expressed Acerv genes (poverA = 0.85,5), and all differentially expressed planuala genes (p<0.05)
# gcounts_filt_acerv_sub <- read.csv("~/Desktop/acerv_counts_sub_filt_20210219.csv", header = TRUE)
# dim(gcounts_filt_acerv_sub) # 9054 rows x 14
# for ( col in 1:ncol(gcounts_filt_acerv_sub)){
#   colnames(gcounts_filt_acerv_sub)[col] <-  gsub("X", "", colnames(gcounts_filt_acerv_sub)[col])
# }
# colnames(gcounts_filt_acerv_sub)[1] <-"gene_id"
# head(gcounts_filt_acerv_sub)
# gcounts_filt_acerv_sub$gene_id <- gsub("TU", "model", gcounts_filt_acerv_sub$gene_id) # sub model for TU to match gene names in annot and interpro files
# # or use the acerv gene count matrix?
# 
# # Load DEGs
# DEG_acerv_sub <- read.csv("~/Desktop/acerv_sub_unique.sig.list_20210219.csv", header = TRUE)
# # use the file or acerv_sub_DEGs.all_treatment_20210219.csv?
# dim(DEG_acerv_sub) # 215 x 22
# for ( col in 1:ncol(DEG_acerv_sub)){
#   colnames(DEG_acerv_sub)[col] <-  gsub("X", "", colnames(DEG_acerv_sub)[col])
# }
# colnames(DEG_acerv_sub)[1] <-"gene_id"
# head(DEG_acerv_sub)
# DEG_acerv_sub$gene_id <- gsub("TU", "model", DEG_acerv_sub$gene_id) # sub model for TU to match gene names in annot and interpro files
# 
# #Import merged annotated gtf file that was made from stringTie
# map <- read.csv("~/Desktop/GFFs/Acerv.merged.annotated.gtf", header=FALSE, sep="\t")
# map <- subset(map, V3=="transcript") # select only transcripts
# dim(map) # 48478 x 9
# map <- separate(map, V9, into = c("transcript_id", "gene_id", "gene_name", "xloc", "cmp_ref", "class_code", "tss_id"), sep = ";")
# # columns were not evenly separated, so some things are in different rows, but I don't think I'll be using those cols anyway
# # gene_id and gene_id_created were originally transcript_id and gene_id respectively. gene_id_created had the MSTG names included, which are the 
# # novel 'genes' created by stringtie
# map <- select(map, c(V1, V4, V5, transcript_id, gene_id)) # might need to include gene name and/or xloc at some point
# colnames(map) <- c("scaffold", "start", "stop", "transcript_id", "gene_id")
# map$transcript_id <- gsub("transcript_id", "", map$transcript_id)
# map$gene_id <- gsub("gene_id", "", map$gene_id)
# map$gene_id <- gsub(" ","",map$gene_id) #remove blank spaces 
# map <- map %>% mutate(map, length = stop - start)
# colnames(map) <- c("scaffold", "start", "stop", "gene_id", "gene_id_original", "length")
# 
# #map$gene_id <- gsub("model", "TU", map$gene_id)
# # in the annotation file, TU designations "Parent" or overall gene name and model designates the gene parts (ie exon, CDS, etc)
# map <- map %>% mutate_all(na_if, " ") # Remove any duplicates
# map <- select(map, c(transcript_id, gene_id))
# map <- unique(map) # might have to subset the unique values by either transcript_id or gene_id
# dim(map) #48478 x 2
# # transcript id has model in the name, while gene id has TU in name 
# 
# # Import reference annotation file 
# ref <- read.table("~/Desktop/GFFs/Acerv.GFFannotations.fixed_transcript.gff3", sep = "\t", header = FALSE)
# ref <- subset(ref, V3 == "mRNA")
# colnames(ref) <- c("scaffold", "gene.predict", "id", "start", "stop", "pos1", "pos2", "pos3", "attr", "gene_id")
# ref <- select(ref, c(scaffold, start, stop, gene_id))
# #ref$transcript_id <- gsub(".*;", "", ref$transcript_id)
# #ref$transcript_id <- gsub("Name=", "", ref$transcript_id)
# ref <- ref %>% mutate(ref, length = stop - start)
# #ref$gene_id <- gsub("TU", "model", ref$gene_id) # sub model for TU to match gene names in annot and interpro files
# dim(ref) # 15243 x 5
# 
# # Build a dataframe that links the gene IDs of expressed genes (poverA = 0.85,5), the gene ids of those genes (from the gene map), and the gene lengths (from the annotation file)
# # Going to try to use transcript_id here instead of gene_id in map df. I'll need to rename cols in the map file so it can be properly merged with counts df
# 
# test <- merge(ref, gcounts_filt_acerv_sub, by = "gene_id")
# test_again <- merge(test, map, by = "gene_id")
# 
# 
# 
# 
# gcounts_filt_acerv_sub$gene_id <- as.factor(gcounts_filt_acerv_sub$gene_id)
# map$gene_id <- as.factor(map$gene_id)
# 
# 
# 
# map$gene_id <- gsub(" ", "", map$gene_id)
# acerv_filt_sub.map <- merge(gcounts_filt_acerv_sub, map, by = "gene_id")
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# dim(acerv_filt_sub.map) # should be same # of rows as gcounts_filt_acerv (9054)
# #acerv_filt_sub.map$gene_id <- gsub("model", "TU", acerv_filt_sub.map$gene_id)
# 
# # Find gene positions in ref corresponding to expressed genes 
# acerv_genes_sub.map_ref <- merge(acerv_filt_sub.map, ref, by = "gene_id", all.x = TRUE)
# #acerv_genes_sub.map_ref <- select(acerv_genes_sub.map_ref, -transcript_id)
# # why do some 
# 
# #### Build GOSEQ vector 
# #GOseq requires a vector of all genes and all differentially expressed genes. 
# # Make gene vector 
# DEG <- filter(acerv_filt_sub.map, gene_id%in%DEG_acerv_sub$gene_id) #make vector of differentially expressed genes
# dim(DEG) # should be same # of rows as DEG_acerv (215)
# DEG_names <- as.vector(DEG$gene_id)
# 
# # Make vector of all expressed genes (poverA = 0.85,5) with non-differentially expressed genes as 0 and differentially expressed genes as 1
# gene_vector=as.integer(acerv_filt_sub.map$gene_id%in%DEG_names)
# names(gene_vector)=acerv_filt_sub.map$gene_id
# head(gene_vector) # should be 9054
# 
# #Make ID vector
# ID_vector <- acerv_filt_sub.map$gene_id
# head(ID_vector) # should be 9054
# 
# #Make length vector
# length_vector <- acerv_filt_sub.map$length
# tail(length_vector) # should be 9054
# 
# #Calculate Probability Weighting Function
# DEG.pwf<-nullp(gene_vector, ID_vector, bias.data=length_vector) #weight vector by length of gene
# sum(is.na(DEG.pwf$pwf)) # 5037 out of 9054 genes do not have lengths 
# #DEG.pwf <- na.omit(DEG.pwf)
# # NAs in nullp: https://rdrr.io/bioc/goseq/man/nullp.html
# # plot sloping downward, but I want it to be sloping upward. not sure why, need to check 
# # downward slope means that the proportion of short differentially expressed genes is high, while the proportion of long DEGs is low. 
# 
# 
# 
# 
# # Some DEG gene ids are not in the gff file, but they are in the fasta file. not sure how they can be in one but not the other. 
# # Therefore, i may not be able to use the DEGs that are not in the gff file because they do not have a length attached to them
# # and the whole point of GOseq is to weight the enrichment by the length of genes (assumption that longer genes will have more reads)
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# #annot <- read.csv("Desktop/PutnamLab/Repositories/FunctionalAnnotation/FunctionalAnnotation/final_Annotations/acerv_FuncAnn_UniP_B2G.csv", header = TRUE)
# annot <- read.csv("Desktop/PutnamLab/Repositories/FunctionalAnnotation/FunctionalAnnotation/final_Annotations/acerv_FullAnnot.csv", header = TRUE)
# 
# ## edit fullAnnot file in acerv_compile to make sure all annotations (uniprot, IPS, b2g) are included in file 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# #annot$gene_id <- gsub("model", "TU", annot$gene_id) # need to change to match gene count gene_ids - because annotations were made with protein/transcript seqs, they have the 'model' designation while the gene counts have the 'TU' designation.
# 
# #### Prepare GO term dataframe 
# GO.annot <- select(annot, gene_id, GO.IDs)
# GO.annot$GO.IDs <- gsub(",", ";", GO.annot$GO.IDs)
# splitted <- strsplit(as.character(GO.annot$GO.ID), ";") #split into multiple GO ids
# GO.terms <- data.frame(v1 = rep.int(GO.annot$gene_id, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row
# colnames(GO.terms) <- c("gene_id", "GO.ID")
# GO.terms$GO.ID <- gsub("F:", "", GO.terms$GO.ID)
# GO.terms$GO.ID <- gsub("P:", "", GO.terms$GO.ID)
# GO.terms$GO.ID <- gsub("C:", "", GO.terms$GO.ID)
# GO.terms$GO.ID <- gsub(" ", "", GO.terms$GO.ID)
# GO.terms[GO.terms == "NA"] <- NA
# 
# head(GO.terms)
# tail(GO.terms)
# GO.terms$GO.ID<- as.character(GO.terms$GO.ID)
# GO.terms$GO.ID <- replace_na(GO.terms$GO.ID, "unknown")
# GO.terms$GO.ID <- as.factor(GO.terms$GO.ID)
# GO.terms$gene_id <- as.factor(GO.terms$gene_id)
# GO.terms <- unique(GO.terms)
# 
# dim(GO.terms) 
# head(GO.terms, 10)
# tail(GO.terms, 10)
# 
# 
# ### Perform GOseq
# # Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
# GO.wall<-goseq(DEG.pwf, ID_vector, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)
# # so i need to remove the NAs from the DEG.pwf df
# # Using manually entered categories.
# # Calculating the p-values...
# # 'select()' returned 1:1 mapping between keys and columns
# write.csv(GO.wall, file = "~/Desktop/acerv_sub_GO_ALL_20210206.csv")
# 
# 
# 
# #### Explore enriched GO terms
# class(GO.wall)
# dim(GO.wall) # 1771 x 7
# head(GO.wall)
# tail(GO.wall)
# #Find overrepresented (expressed more?) enriched GO terms that are statistically significant at cutoff (p<0.05)
# enriched.GO.05 <- GO.wall$category[GO.wall$over_represented_pvalue<.05]
# enriched.GO.05 <- data.frame(enriched.GO.05)
# dim(enriched.GO.05) # 14 x 1
# colnames(enriched.GO.05) <- c("category")
# enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")
# enriched.GO.05 <- enriched.GO.05[order(-enriched.GO.05$numDEInCat),]
# enriched.GO.05$term <- as.factor(enriched.GO.05$term)
# head(enriched.GO.05)
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# 
# ### Prepare GO term dataframe 
# # Import GO term
# annot_GO <- read.csv("~/Desktop/acerv_GO_20210219.csv", header=TRUE)
# annot_GO <- select(annot_GO, -X)
# colnames(annot_GO) <- c("gene_id", "GO.ID")
# 
# #annot_GO$gene_id <- gsub("model", "TU", annot_GO$gene_id)
# annot_GO$GO.ID <- gsub(",", ";", annot_GO$GO.ID)
# 
# # Split GO terms so there is only 1 GO term per row
# GO.terms <- separate_rows(annot_GO, GO.ID, sep = ";")
# colnames(GO.terms) <- c("gene_id", "GO.ID")
# GO.terms <- GO.terms[!grepl("NA", GO.terms$GO.ID),]
# GO.terms <- unique(GO.terms)
# 
# # Merge GO terms with all expressed acerv genes 
# GO.terms <- merge(acerv_filt_sub.map, GO.terms, by.x="gene_id")
# unique(GO.terms)
# dim(GO.terms) # 14099 x 19
# 
# # Clean up GO.terms 
# GO.terms <- select(GO.terms, gene_id, GO.ID)
# GO.terms$GO.ID <- replace_na(GO.terms$GO.ID, "unknown")
# head(GO.terms, 10)
# tail(GO.terms, 10)
# GO.terms <- unique(GO.terms)
# 
# ### Perform GOseq
# # Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
# GO.wall2<-goseq(DEG.pwf, ID_vector, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)
# # so i need to remove the NAs from the DEG.pwf df
# # Using manually entered categories.
# # Calculating the p-values...
# # 'select()' returned 1:1 mapping between keys and columns
# write.csv(GO.wall, file = "~/Desktop/acerv_sub_GO_ALL_20210206.csv")
# 
# #### Explore enriched GO terms
# class(GO.wall)
# dim(GO.wall) # 1771 x 7
# head(GO.wall)
# tail(GO.wall)
# #Find overrepresented (expressed more?) enriched GO terms that are statistically significant at cutoff (p<0.05)
# enriched.GO.05 <- GO.wall$category[GO.wall$over_represented_pvalue<.05]
# enriched.GO.05 <- data.frame(enriched.GO.05)
# dim(enriched.GO.05) # 14 x 1
# colnames(enriched.GO.05) <- c("category")
# enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")
# enriched.GO.05 <- enriched.GO.05[order(-enriched.GO.05$numDEInCat),]
# enriched.GO.05$term <- as.factor(enriched.GO.05$term)
# head(enriched.GO.05)
# #Subset enriched GO terms by category and save as csv
# MF <- subset(enriched.GO.05, ontology=="MF")
# MF <- MF[order(-MF$numDEInCat),]
# CC <- subset(enriched.GO.05, ontology=="CC") 
# CC <- CC[order(-CC$numDEInCat),] 
# BP <- subset(enriched.GO.05, ontology=="BP")
# BP <- BP[order(-BP$numDEInCat),]
# write.csv(MF, file = "~/Desktop/acerv_sub_MF_Sig_Enriched_GO.05_20210206.csv")
# write.csv(CC, file = "~/Desktop/acerv_CC_Sig_Enriched_GO.05_20210206.csv") 
# write.csv(BP, file = "~/Desktop/acerv_sub_BP_Sig_Enriched_GO.05_20210206.csv")
# write.csv(enriched.GO.05, file = "~/Desktop/acerv_sub_Sig_Enriched_GO.05_ALL_20210206.csv")
# 
# # Merge enriched GO terms with gene ids (from GO.terms df)
# colnames(GO.terms) <- c("gene_id","category")
# GO.terms_gene_ids <- merge(enriched.GO.05, GO.terms, by = "category", all.x = TRUE) # this combines gene ids with GO info from goseq analysis by GO term
# GO.terms_gene_ids <- unique(GO.terms_gene_ids)
# GO.terms_gene_ids.acerv_genes_sub.map_unique.ref <- merge(GO.terms_gene_ids, acerv_genes_sub.map_ref, by = "gene_id") # this combines GO info with counts and length data by gene id 
# GO.terms_gene_ids.acerv_genes_sub.map_unique.ref <- select(GO.terms_gene_ids.acerv_genes_sub.map_unique.ref, c(gene_id, scaffold, start, stop, length, category, over_represented_pvalue, under_represented_pvalue, numDEInCat, numInCat, term, ontology ))
# GO.terms_gene_ids.acerv_genes_sub.map_unique.ref <- na.omit(GO.terms_gene_ids.acerv_genes_sub.map_unique.ref)
# write.csv(GO.terms_gene_ids.acerv_genes_sub.map_unique.ref, file = "~/Desktop/acerv_sub_GOterms_gene_ids_20210207.csv")
# 
# # Merge GO.terms_gene_ids.acerv_genes.map_unique.ref and treatment comparison list 
# DEG_treatment <- read.csv("~/Desktop/acerv_sub_DEGs.all_treatment_20210208.csv", header = TRUE)
# #colnames(DEG_treatment)[1] <-"gene_id"
# ByTreatment_GO.terms <- merge(DEG_treatment, GO.terms_gene_ids.acerv_genes_sub.map_unique.ref, by = "gene_id", all.x = T) # this combines GO info, counts, and lengths with DEG treatment info by gene id 
# ByTreatment_GO.terms <- na.omit(ByTreatment_GO.terms)
# dim(ByTreatment_GO.terms) # 86 x 33
# length(unique(ByTreatment_GO.terms$gene_id)) # 39 unique gene ids 
# #ByTreatment_GO.terms <- filter(ByTreatment_GO.terms, over_represented_pvalue > 0)
# ############### THIS IS WHERE I NEED TO GET THE GENE IDS TO BLAST
# # now I have a df with gene ids, treatment comparisons, GO terms, and term info
# # in order for a gene id to make it onto this list, it has to:
# #1) be differentially expressed 
# #2) have length data from gff file 
# #3) have GO term(s) assigned to it 
# MF_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="MF")
# MF_ByTreatment <- MF_ByTreatment[order(-MF_ByTreatment$numDEInCat),]
# CC_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="CC")
# CC_ByTreatment <- CC_ByTreatment[order(-CC_ByTreatment$numDEInCat),] # no sig enriched under CC
# BP_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="BP")
# BP_ByTreatment <- BP_ByTreatment[order(-BP$numDEInCat),]
# write.csv(MF_ByTreatment, file = "~/Desktop/acerv_sub_MF_Sig_Enriched_ByTreatment_GO.05_20210208.csv")
# write.csv(CC_ByTreatment, file = "~/Desktop/acerv_CC_Sig_Enriched_ByTreatment_GO.05_20210208.csv")
# write.csv(BP_ByTreatment, file = "~/Desktop/acerv_sub_BP_Sig_Enriched_ByTreatment_GO.05_20210208.csv")
# write.csv(ByTreatment_GO.terms, file = "~/Desktop/acerv_sub_ByTreatment_GO.terms_20210208.csv")
# 
# # Plot terms by number of differentially expressed functions
# MFplot <- MF %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
#   ggplot( aes(x=term, y=numDEInCat) ) +
#   geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
#   geom_point(size=3, color="#69b3a2") +
#   coord_flip() +
#   theme(
#     panel.grid.minor.y = element_blank(),
#     panel.grid.major.y = element_blank(),
#     legend.position="none"
#   ) +
#   xlab("") +
#   ylab("") +
#   ggtitle("Molecular Function") + #add a main title
#   theme(plot.title = element_text(face = 'bold', 
#                                   size = 12, 
#                                   hjust = 0)) +
#   theme_bw() + #Set background color 
#   theme(panel.border = element_blank(), # Set border
#         panel.grid.major = element_blank(), #Set major gridlines
#         panel.grid.minor = element_blank(), #Set minor gridlines
#         axis.line = element_line(colour = "black"), #Set axes color
#         plot.background=element_blank())#Set the plot background
# MFplot
# 
# CCplot <- CC %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
#   ggplot( aes(x=term, y=numDEInCat) ) +
#   geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
#   geom_point(size=3, color="#69b3a2") +
#   coord_flip() +
#   theme(
#     panel.grid.minor.y = element_blank(),
#     panel.grid.major.y = element_blank(),
#     legend.position="none"
#   ) +
#   xlab("") +
#   ylab("") +
#   ggtitle("Cellular Component") + #add a main title
#   theme(plot.title = element_text(face = 'bold',
#                                   size = 12,
#                                   hjust = 0)) +
#   theme_bw() + #Set background color
#   theme(panel.border = element_blank(), # Set border
#         panel.grid.major = element_blank(), #Set major gridlines
#         panel.grid.minor = element_blank(), #Set minor gridlines
#         axis.line = element_line(colour = "black"), #Set axes color
#         plot.background=element_blank())#Set the plot background
# CCplot
# 
# BPplot <- BP %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
#   ggplot( aes(x=term, y=numDEInCat) ) +
#   geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
#   geom_point(size=3, color="#69b3a2") +
#   coord_flip() +
#   theme(
#     panel.grid.minor.y = element_blank(),
#     panel.grid.major.y = element_blank(),
#     legend.position="none") +
#   xlab("") +
#   ylab("") +
#   ggtitle("Biological Process") + #add a main title
#   theme(plot.title = element_text(face = 'bold', 
#                                   size = 12, 
#                                   hjust = 0)) +
#   theme_bw() + #Set background color 
#   theme(panel.border = element_blank(), # Set border
#         panel.grid.major = element_blank(), #Set major gridlines
#         panel.grid.minor = element_blank(), #Set minor gridlines
#         axis.line = element_line(colour = "black"), #Set axes color
#         plot.background=element_blank())#Set the plot background
# BPplot
# 
# # Save MF, CC, and BP plot in grid layout
# GOplot <- grid.arrange(MFplot, BPplot, ncol=3, clip="off")
# ggsave("~/Desktop/acerv_sub_GOplot_05_20210207.pdf", GOplot, width = 21, height = 21, units = c("in"))
# 
# # Combining MF, CC, and BP into one plot and order by pvalue
# GOplot2 <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
#   mutate(term = fct_reorder(term, ontology)) %>%
#   ggplot( aes(x=term, y=numDEInCat) ) +
#   geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
#   geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
#   geom_point(size=3, aes(colour = ontology)) +
#   coord_flip() +
#   ylim(0,25) +
#   theme(
#     panel.grid.minor.y = element_blank(),
#     panel.grid.major.y = element_blank(),
#     legend.position="bottom"
#   ) +
#   xlab("") +
#   ylab("") +
#   theme_bw() + #Set background color 
#   theme(panel.border = element_blank(), # Set border
#         panel.grid.major = element_blank(), #Set major gridlines
#         panel.grid.minor = element_blank(), #Set minor gridlines
#         axis.line = element_line(colour = "black"), #Set axes color
#         plot.background=element_blank()) #Set the plot background #set title attributes
# GOplot2
# ggsave("~/Desktop/acerv_sub_GOplot2_05_20210207.pdf", GOplot2, width = 28, height = 28, units = c("in"))
# 
# # Combining into one plot and order by pvalue 
# GOplot_pvalue <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
#   mutate(term = fct_reorder(term, ontology)) %>%
#   ggplot( aes(x=term, y=over_represented_pvalue) ) +
#   geom_segment( aes(x=term ,xend=term, y=0, yend=over_represented_pvalue), color="grey") +
#   geom_text(aes(label = numDEInCat), hjust = -1, vjust = 0.5, size = 3) +
#   geom_point(size=3, aes(colour = ontology)) +
#   coord_flip() +
#   ylim(0,0.05) +
#   theme(
#     panel.grid.minor.y = element_blank(),
#     panel.grid.major.y = element_blank(),
#     legend.position="bottom"
#   ) +
#   xlab("") +
#   ylab("p-value") +
#   theme_bw() + #Set background color 
#   theme(panel.border = element_blank(), # Set border
#         panel.grid.major = element_blank(), #Set major gridlines
#         panel.grid.minor = element_blank(), #Set minor gridlines
#         axis.line = element_line(colour = "black"), #Set axes color
#         plot.background=element_blank()) #Set the plot background #set title attributes
# GOplot_pvalue
# ggsave("~/Desktop/acerv_sub_GOplot_pvalue_05_20210207.pdf", GOplot_pvalue, width = 28, height = 28, units = c("in"))
# 
