---
title: "acerv_GOSeq"
author: "jillashey"
date: "11/4/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# Load libraries
library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("gridExtra")
library("clusterProfiler")
library("DataCombine")
library("VennDiagram")
```

This code is using the subsetted acerv data. This code is also using the GO terms generated by Diamond (B2G), Swiss Prot (B2G), and IPS [here](https://github.com/JillAshey/FunctionalAnnotation/tree/main/final_Annotations) in Nov. 2021. 

## Read in all expressed pdam genes (poverA = 0.85,5)
```{r}
# Obtain names of all expressed Acerv genes (poverA = 0.85,5), and all differentially expressed planuala genes (p<0.05)
gcounts_filt_acerv_sub <- read.csv("Output/DESeq2/acerv/acerv_counts_sub_filt_20210327.csv", header = TRUE)
dim(gcounts_filt_acerv_sub) # 9054 rows x 14
colnames(gcounts_filt_acerv_sub)[1] <-"gene_id" # make colnames a true column called gene_id
head(gcounts_filt_acerv_sub)
length(unique(gcounts_filt_acerv_sub$gene_id)) # 9054 total unique gene ids 
gcounts_filt_acerv_sub$gene_id <- gsub("TU", "model", gcounts_filt_acerv_sub$gene_id) # sub model for TU to match gene ids in annot files
```

## Read in acerv sub DEGs
```{r}
DEG_acerv_sub <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = TRUE)
dim(DEG_acerv_sub) # 215 x 14

colnames(DEG_acerv_sub)[1] <-"gene_id" # make colnames a true column called gene_id
head(DEG_acerv_sub)
length(unique(DEG_acerv_sub$gene_id)) # 215 total unique gene ids 
DEG_acerv_sub$gene_id <- gsub("TU", "model", DEG_acerv_sub$gene_id) # sub model for TU to match gene ids in annot files
```

## Read in length data (calculated directly from transcripts) and merge with filtered counts 
```{r}
length_Acerv <- read.csv("Output/DESeq2/acerv/length.mRNA_Acerv.csv", header = TRUE)
length_merge <- merge(gcounts_filt_acerv_sub, length_Acerv, by = "gene_id")
dim(length_merge) # 9052 x 15 -- 2 genes lost because they are actually tRNA transcripts that somehow snuck in? idk, I checked them against the mRNA sequences and .gff file and I feel confident moving forward w/o them. They also are not DEGs
colnames(length_merge)[1] <-"gene_id" # make colnames a true column called gene_id
```

# Build GOSEQ vector 
GOseq requires a vector of all genes, all differentially expressed genes, and gene lengths

## Make vectors
```{r}
#DEG<- filter(length_Acerv, gene_id %in% DEG_acerv_sub$gene_id) - for some reason, R keeps giving me error saying object 'gene_id' not found, so used another way below
DEG <- length_merge[length_merge$gene_id %in% DEG_acerv_sub$gene_id, ]
dim(DEG) # 215 x 15
DEG_names <- as.vector(DEG$gene_id)
gene_vector <- as.integer(length_merge$gene_id%in%DEG_names)
names(gene_vector) <- length_merge$gene_id

# Make ID vector 
IDvector <- length_merge$gene_id

# Make length vector
lengthVector <- length_merge$length
```

## Calculate Probability Weighting Function
```{r}
DEG.pwf<-nullp(gene_vector, IDvector, bias.data=lengthVector) #weight vector by length of gene
# plot looks weird...does this mean that there is not much length bias in the dataset?
```

# GOSeq w/ my annotations from Nov. 2021 (GO terms generated by Diamond (B2G), Swiss Prot (B2G), and IPS)

## Load GO terms 
```{r}
annot_GO <- read.csv("~/Desktop/PutnamLab/Repositories/FunctionalAnnotation/FunctionalAnnotation/final_Annotations/Acerv_GOterms_20211104.csv", header = TRUE)[,2:3]
# GO terms already split into single GO term per row/sequence 
colnames(annot_GO) <- c("gene_id", "GO.ID")
annot_GO[annot_GO == "NA"] <- NA
annot_GO$GO.ID<- as.character(annot_GO$GO.ID)
annot_GO$GO.ID <- replace_na(annot_GO$GO.ID, "unknown")
annot_GO$GO.ID <- as.factor(annot_GO$GO.ID)
annot_GO$gene_id <- as.factor(annot_GO$gene_id)
annot_GO <- unique(annot_GO)

dim(annot_GO) # 237178 x 2
length(unique(annot_GO$GO.ID)) # 16372 unique GO.IDs
head(annot_GO)
```

# Perform GOseq

## Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
```{r}
### Perform GOseq
# Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
# should I include this: test.cats=c("GO:CC", "GO:BP", "GO:MF") ?
GO.wall<-goseq(DEG.pwf, ID_vector, gene2cat=annot_GO, method="Wallenius", use_genes_without_cat=TRUE)
# Using manually entered categories.
# Calculating the p-values...
# 'select()' returned 1:1 mapping between keys and columns
write.csv(GO.wall, file = "Output/GOSeq/acerv/acerv_sub_GO_ALL_20211104.csv")

# Check GO.wall df
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

# Find significantly enriched GO terms 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")
enriched.GO.05 <- enriched.GO.05[order(-enriched.GO.05$numDEInCat),]
enriched.GO.05$term <- as.factor(enriched.GO.05$term)
head(enriched.GO.05)

# Subset enriched GO terms by ontology (BP, CC, MF) and save as csv. 
MF <- subset(enriched.GO.05, ontology=="MF")
MF <- MF[order(-MF$numDEInCat),]
CC <- subset(enriched.GO.05, ontology=="CC") 
CC <- CC[order(-CC$numDEInCat),] 
BP <- subset(enriched.GO.05, ontology=="BP")
BP <- BP[order(-BP$numDEInCat),]
write.csv(MF, file = "Output/GOSeq/acerv/acerv_sub_MF_Sig_Enriched_GO.05_20211104.csv")
write.csv(CC, file = "Output/GOSeq/acerv/acerv_CC_Sig_Enriched_GO.05_20211104.csv") 
write.csv(BP, file = "Output/GOSeq/acerv/acerv_sub_BP_Sig_Enriched_GO.05_20211104.csv")
write.csv(enriched.GO.05, file = "Output/GOSeq/acerv/acerv_sub_Sig_Enriched_GO.05_ALL_20211104.csv")
```

## Merge DEG gene ids and GO info
```{r}
# Merge gene ids and sig enriched GO terms 
colnames(annot_GO) <- c("gene_id", "category")
merge <- merge(enriched.GO.05, annot_GO, by.x = "category") # contains gene ids and GO info 
#merge <- unique(merge)

# Merge DEG with merge 
merge_again <- merge(merge, DEG, by = "gene_id") # contains GO info and gene counts for DEGs

```

## Add treatment info to df and save 
```{r}
# Read in DEG treatment info
acerv_treatment <- read.csv("Output/DESeq2/acerv/acerv_sub_DEGs.all_treatment_20210219.csv", header = TRUE)
acerv_treatment <- select(acerv_treatment, c("baseMean", "log2FoldChange", "lfcSE", "stat", "pvalue", "padj", "Treatment_Compare", "gene_id"))
acerv_treatment$gene_id <- gsub("TU", "model", acerv_treatment$gene_id)

# Merge treatment info with GO info 
merge_all <- merge(merge_again, acerv_treatment, by = "gene_id")
# yay now I have a df with all the info - GO info, gene ids, DESeq2 stats, gene counts
dim(merge_all) # 994 x 29
length(unique(merge_all$gene_id)) # 83 unique genes that are differentially expressed and have GO term info
length(unique(merge_all$category)) # 366 unique GO terms 
write.csv(merge_all, file = "Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv")

# Subset enriched GO terms by ontology (BP, CC, MF) and save as csv. 
MF_ByTreatment <- subset(merge_all, ontology=="MF")
MF_ByTreatment <- MF_ByTreatment[order(-MF_ByTreatment$numDEInCat),]
CC_ByTreatment <- subset(merge_all, ontology=="CC")
CC_ByTreatment <- CC_ByTreatment[order(-CC_ByTreatment$numDEInCat),] # no sig enriched under CC
BP_ByTreatment <- subset(merge_all, ontology=="BP")
BP_ByTreatment <- BP_ByTreatment[order(-BP$numDEInCat),]
write.csv(MF_ByTreatment, file = "Output/GOSeq/acerv/acerv_sub_MF_Sig_Enriched_ByTreatment_GO.05_20211104.csv")
write.csv(CC_ByTreatment, file = "Output/GOSeq/acerv/acerv_sub_CC_Sig_Enriched_ByTreatment_GO.05_20211104.csv")
write.csv(BP_ByTreatment, file = "Output/GOSeq/acerv/acerv_sub_BP_Sig_Enriched_ByTreatment_GO.05_20211104.csv")
```

## Plot terms by ontology numbers 
```{r}
MFplot <- MF %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none"
  ) +
  xlab("") +
  ylab("") +
  ggtitle("Molecular Function") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
MFplot

CCplot <- CC %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none"
  ) +
  xlab("") +
  ylab("") +
  ggtitle("Cellular Component") + #add a main title
  theme(plot.title = element_text(face = 'bold',
                                  size = 12,
                                  hjust = 0)) +
  theme_bw() + #Set background color
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
CCplot 

BPplot <- BP %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none") +
  xlab("") +
  ylab("") +
  ggtitle("Biological Process") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
BPplot # lots of BP, hard to see. Could subset df and plot by BP dfs

# Save MF, CC, and BP plot in grid layout
GOplot <- grid.arrange(MFplot, BPplot, ncol=3, clip="off")
ggsave("Output/Figs/acerv/acerv_sub_GOplot_05_FullAnnot_20211104.pdf", GOplot, width = 21, height = 21, units = c("in"))
```

# Combining MF, CC, and BP into one plot and order by pvalue
```{r}
GOplot2 <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,45) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot2
ggsave("Output/Figs/acerv/acerv_sub_GOplot2_05_FullAnnot_20211104.pdf", GOplot2, width = 28, height = 28, units = c("in"))

# Combining into one plot and order by pvalue 
GOplot_pvalue <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=over_represented_pvalue) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=over_represented_pvalue), color="grey") +
  geom_text(aes(label = numDEInCat), hjust = -1, vjust = 0.5, size = 3) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,0.05) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("p-value") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot_pvalue
ggsave("Output/Figs/acerv/acerv_sub_GOplot_pvalue_05_FullAnnot_20211104.pdf", GOplot_pvalue, width = 28, height = 28, units = c("in"))
```











