# Title: Sediment stress - acerv GOseq
# Author: Jill Ashey
# date: 1/25/21

### Gene Ontology Analysis of Differentially Expressed Genes in Acerv
# Using updated GO terms (terms generated by Blast2GO, Interproscan, and Uniprot)

# This script is using subsetted acerv data (removed samples 24 and 45)


# Load packages 
library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("forcats")
library("gridExtra")

####### Subsetted samples GO analysis

# Obtain names of all expressed Acerv genes (poverA = 0.85,5), and all differentially expressed planuala genes (p<0.05)
gcounts_filt_acerv_sub <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/acerv/acerv_counts_sub_filt.csv", header = TRUE)
dim(gcounts_filt_acerv_sub) # 9054 rows x 14
for ( col in 1:ncol(gcounts_filt_acerv_sub)){
  colnames(gcounts_filt_acerv_sub)[col] <-  gsub("X", "", colnames(gcounts_filt_acerv_sub)[col])
}
colnames(gcounts_filt_acerv_sub)[1] <-"gene_id"
head(gcounts_filt_acerv_sub)
#gcounts_filt_acerv_sub$gene_id <- gsub("TU", "model", gcounts_filt_acerv_sub$gene_id) # sub model for TU to match gene names in annot and interpro files

# Load DEGs
DEG_acerv_sub <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/acerv/acerv_sub_unique.sig.list.csv", header = TRUE)
dim(DEG_acerv_sub) # 217 x 14
for ( col in 1:ncol(DEG_acerv_sub)){
  colnames(DEG_acerv_sub)[col] <-  gsub("X", "", colnames(DEG_acerv_sub)[col])
}
colnames(DEG_acerv_sub)[1] <-"gene_id"
head(DEG_acerv_sub)
#DEG_acerv_sub$gene_id <- gsub("TU", "model", DEG_acerv_sub$gene_id) # sub model for TU to match gene names in annot and interpro files

#Import merged annotated gtf file
map <- read.csv("~/Desktop/GFFs/Acerv.merged.annotated.gtf", header=FALSE, sep="\t")
map <- subset(map, V3=="transcript") # select only transcripts
dim(map) # 48478 x 9
map <- separate(map, V9, into = c("transcript_id", "gene_id", "gene_name", "xloc", "cmp_ref", "class_code", "tss_id"), sep = ";")
# columns were not evenly separated, so some things are in different rows, but I don't think I'll be using those cols anyway
# gene_id and gene_id_created were originally transcript_id and gene_id respectively. gene_id_created had the MSTG names included, which are the 
# novel 'genes' created by stringtie
map <- select(map, c(V1, V4, V5, transcript_id, gene_id)) # might need to include gene name and/or xloc at some point
colnames(map) <- c("scaffold", "start", "stop", "transcript_id", "gene_id")
map$transcript_id <- gsub("transcript_id", "", map$transcript_id)
map$gene_id <- gsub("gene_id", "", map$gene_id)
map$gene_id <- gsub(" ","",map$gene_id) #remove blank spaces 
#map$gene_id <- gsub("model", "TU", map$gene_id)
map <- map %>% mutate_all(na_if, " ") # Remove any duplicates
map <- select(map, c(transcript_id, gene_id))
map <- unique(map) # might have to subset the unique values by either transcript_id or gene_id
dim(map) #48478 x 2
# transcript id has model in the name, while gene id has TU in name 

# Import reference annotation file 
ref <- read.table("~/Desktop/GFFs/Acerv.GFFannotations.fixed_transcript.gff3", sep = "\t", header = FALSE)
ref <- subset(ref, V3 == "gene")
colnames(ref) <- c("scaffold", "gene.predict", "id", "start", "stop", "pos1", "pos2", "pos3", "attr", "gene_id")
ref <- select(ref, c(scaffold, start, stop, gene_id))
#ref$transcript_id <- gsub(".*;", "", ref$transcript_id)
#ref$transcript_id <- gsub("Name=", "", ref$transcript_id)
ref <- ref %>% mutate(ref, length = stop - start)
#ref$gene_id <- gsub("TU", "model", ref$gene_id) # sub model for TU to match gene names in annot and interpro files
dim(ref) # 15243 x 5

# Build a dataframe that links the gene IDs of expressed genes (poverA = 0.85,5), the gene ids of those genes (from the gene map), and the gene lengths (from the annotation file)
# Going to try to use transcript_id here instead of gene_id in map df. I'll need to rename cols in the map file so it can be properly merged with counts df
acerv_filt_sub.map <- merge(gcounts_filt_acerv_sub, map, by = "gene_id", all.x = TRUE)
dim(acerv_filt_sub.map) # should be same # of rows as gcounts_filt_acerv (9054)
#acerv_filt_sub.map$gene_id <- gsub("model", "TU", acerv_filt_sub.map$gene_id)

# Find gene positions in ref corresponding to expressed genes 
acerv_genes_sub.map_ref <- merge(acerv_filt_sub.map, ref, by = "gene_id", all.x = TRUE)

#### Build GOSEQ vector 
#GOseq requires a vector of all genes and all differentially expressed genes. 
#Make gene vector
DEG <- filter(acerv_genes_sub.map_ref, gene_id%in%DEG_acerv_sub$gene_id) #make vector of differentially expressed genes
dim(DEG) # should be same # of rows as DEG_acerv (217)
DEG_names <- as.vector(DEG$gene_id)

# Make vector of all expressed genes (poverA = 0.85,5) with non-differentially expressed genes as 0 and differentially expressed genes as 1
gene_vector=as.integer(acerv_genes_sub.map_ref$gene_id%in%DEG_names)
names(gene_vector)=acerv_genes_sub.map_ref$gene_id
head(gene_vector) # should be 9054

#Make ID vector
ID_vector <- acerv_genes_sub.map_ref$gene_id
head(ID_vector) # should be 9054

#Make length vector
length_vector <- acerv_genes_sub.map_ref$length
head(length_vector) # should be 9054

#Calculate Probability Weighting Function
DEG.pwf<-nullp(gene_vector, ID_vector, bias.data=length_vector) #weight vector by length of gene
# plot sloping downward, but I want it to be sloping upward. not sure why, need to check 

### Prepare GO term dataframe 
# Import GO term
annot_GO <- read.csv("~/Desktop/acerv_GO_20210124.csv", header=TRUE)
annot_GO <- select(annot_GO, -X)
annot_GO$gene_id <- gsub("model", "TU", annot_GO$gene_id)
annot_GO$GO.ID <- gsub(",", ";", annot_GO$GO.ID)

# Split GO terms so there is only 1 GO term per row
GO.terms <- separate_rows(annot_GO, GO.ID, sep = ";")
colnames(GO.terms) <- c("gene_id", "GO.ID")
GO.terms <- GO.terms[!grepl("NA", GO.terms$GO.ID),]
GO.terms <- unique(GO.terms)

# Merge GO terms with all expressed acerv genes 
GO.terms <- merge(acerv_genes_sub.map_ref, GO.terms, by.x="gene_id")
unique(GO.terms)
dim(GO.terms) # 14099 x 20

# Clean up GO.terms 
GO.terms <- select(GO.terms, gene_id, GO.ID)
GO.terms$GO.ID <- replace_na(GO.terms$GO.ID, "unknown")
head(GO.terms, 10)
tail(GO.terms, 10)

### Perform GOseq
# Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
GO.wall<-goseq(DEG.pwf, ID_vector, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)
# Using manually entered categories.
# Calculating the p-values...
# 'select()' returned 1:1 mapping between keys and columns
# Warning message:
#   In goseq(DEG.pwf, ID_vector, gene2cat = GO.terms, method = "Wallenius",  :
#              Missing length data for 56% of genes.  Accuarcy of GO test will be reduced.
# need to look into this error 
write.csv(GO.wall, file = "~/Desktop/acerv_sub_GO_ALL_20210125.csv")

#### Explore enriched GO terms
class(GO.wall)
dim(GO.wall) # 1771 x 7
head(GO.wall)
tail(GO.wall)
#Find overrepresented (expressed more?) enriched GO terms that are statistically significant at cutoff (p<0.05)
enriched.GO.05 <- GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05 <- data.frame(enriched.GO.05)
dim(enriched.GO.05) # 14 x 1
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")
enriched.GO.05 <- enriched.GO.05[order(-enriched.GO.05$numDEInCat),]
enriched.GO.05$term <- as.factor(enriched.GO.05$term)
head(enriched.GO.05)
#Subset enriched GO terms by category and save as csv
MF <- subset(enriched.GO.05, ontology=="MF")
MF <- MF[order(-MF$numDEInCat),]
CC <- subset(enriched.GO.05, ontology=="CC") 
CC <- CC[order(-CC$numDEInCat),] # no significantly enriched GO terms under CC
BP <- subset(enriched.GO.05, ontology=="BP")
BP <- BP[order(-BP$numDEInCat),]
write.csv(MF, file = "~/Desktop/acerv_sub_MF_Sig_Enriched_GO.05_20210125.csv")
#write.csv(CC, file = "~/Desktop/acerv_CC_Sig_Enriched_GO.05_20210124.csv") 
write.csv(BP, file = "~/Desktop/acerv_sub_BP_Sig_Enriched_GO.05_20210125.csv")
write.csv(enriched.GO.05, file = "~/Desktop/acerv_sub_Sig_Enriched_GO.05_ALL_20210125.csv")

# Merge enriched GO terms with gene ids (from GO.terms df)
colnames(GO.terms) <- c("gene_id","category")
GO.terms_gene_ids <- merge(enriched.GO.05, GO.terms, by = "category", all.x = TRUE)
GO.terms_gene_ids <- unique(GO.terms_gene_ids)
GO.terms_gene_ids.acerv_genes_sub.map_unique.ref <- merge(acerv_genes_sub.map_ref, GO.terms_gene_ids, by = "gene_id", all.x = TRUE)
GO.terms_gene_ids.acerv_genes_sub.map_unique.ref <- select(GO.terms_gene_ids.acerv_genes_sub.map_unique.ref, c(gene_id, scaffold, start, stop, length, category, over_represented_pvalue, under_represented_pvalue, numDEInCat, numInCat, term, ontology ))
#write.csv(GO.terms_gene_ids.acerv_genes.map_unique.ref, file = "~/Desktop/acerv_GOterms_gene_ids.csv")

# Merge GO.terms_gene_ids.acerv_genes.map_unique.ref and treatment comparison list 
DEG_treatment <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/acerv/acerv_sub_DEGs.all_treatment.csv", header = TRUE)
colnames(DEG_treatment)[1] <-"gene_id"
ByTreatment_GO.terms <- merge(GO.terms_gene_ids.acerv_genes_sub.map_unique.ref, DEG_treatment, by = "gene_id", all.x = TRUE)
ByTreatment_GO.terms <- filter(ByTreatment_GO.terms, over_represented_pvalue > 0)
# now I have a df with DEGs gene names, treatment comparisons, GO terms, and term info
MF_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="MF")
MF_ByTreatment <- MF_ByTreatment[order(-MF_ByTreatment$numDEInCat),]
CC_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="CC")
CC_ByTreatment <- CC_ByTreatment[order(-CC_ByTreatment$numDEInCat),] # no sig enriched under CC
BP_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="BP")
BP_ByTreatment <- BP_ByTreatment[order(-BP$numDEInCat),]
write.csv(MF_ByTreatment, file = "~/Desktop/acerv_sub_MF_Sig_Enriched_ByTreatment_GO.05_20210125.csv")
#write.csv(CC_ByTreatment, file = "~/Desktop/acerv_CC_Sig_Enriched_ByTreatment_GO.05_20210124.csv")
write.csv(BP_ByTreatment, file = "~/Desktop/acerv_sub_BP_Sig_Enriched_ByTreatment_GO.05_20210125.csv")
write.csv(ByTreatment_GO.terms, file = "~/Desktop/acerv_sub_ByTreatment_GO.terms_20210125.csv")

# Plot terms by number of differentially expressed functions
MFplot <- MF %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none"
  ) +
  xlab("") +
  ylab("") +
  ggtitle("Molecular Function") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
MFplot

# CCplot <- CC %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
#   ggplot( aes(x=term, y=numDEInCat) ) +
#   geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
#   geom_point(size=3, color="#69b3a2") +
#   coord_flip() +
#   theme(
#     panel.grid.minor.y = element_blank(),
#     panel.grid.major.y = element_blank(),
#     legend.position="none"
#   ) +
#   xlab("") +
#   ylab("") +
#   ggtitle("Cellular Component") + #add a main title
#   theme(plot.title = element_text(face = 'bold', 
#                                   size = 12, 
#                                   hjust = 0)) +
#   theme_bw() + #Set background color 
#   theme(panel.border = element_blank(), # Set border
#         panel.grid.major = element_blank(), #Set major gridlines
#         panel.grid.minor = element_blank(), #Set minor gridlines
#         axis.line = element_line(colour = "black"), #Set axes color
#         plot.background=element_blank())#Set the plot background
# CCplot

BPplot <- BP %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none") +
  xlab("") +
  ylab("") +
  ggtitle("Biological Process") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
BPplot

# Save MF, CC, and BP plot in grid layout
GOplot <- grid.arrange(MFplot, BPplot, ncol=3, clip="off")
ggsave("~/Desktop/acerv_sub_GOplot_05_20210125.pdf", GOplot, width = 21, height = 21, units = c("in"))

# Combining MF, CC, and BP into one plot and order by pvalue
GOplot2 <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,25) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot2
ggsave("~/Desktop/acerv_sub_GOplot2_05_20210125.pdf", GOplot2, width = 28, height = 28, units = c("in"))

# Combining into one plot and order by pvalue 
GOplot_pvalue <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=over_represented_pvalue) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=over_represented_pvalue), color="grey") +
  geom_text(aes(label = numDEInCat), hjust = -1, vjust = 0.5, size = 3) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,0.05) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("p-value") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot_pvalue
ggsave("~/Desktop/acerv_sub_GOplot_pvalue_05_20210125.pdf", GOplot_pvalue, width = 28, height = 28, units = c("in"))

