---
title: "mcap DESeq2"
author: "jillashey"
date: "3/15/2022"
output: html_document
---

#### Code for Francois sedimentation data. Mcap only samples analyzed here aligned against Mcap V2. STAR was read aligner.

## Load packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("ggrepel")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("gridExtra")
library("clusterProfiler")
library("DataCombine")
library("VennDiagram")
```

## Load gene count matrix 
```{r}
countdata <- read.csv("Output/DESeq2/mcap/gene_count_mcap_matrix.csv")
dim(countdata) # 55086 x 12

# Correct colnames 
for ( col in 1:ncol(countdata)){
  colnames(countdata)[col] <-  gsub("X", "", colnames(countdata)[col])
}
for ( col in 1:ncol(countdata)){
  colnames(countdata)[col] <-  gsub(".fastq.trim.fq.Aligned.sortedByCoord.out.bam.merge.gtf", "", colnames(countdata)[col])
}
rownames(countdata) <- countdata$gene_id
```

## Load metadata
```{r}
metadata <- read.csv("Data/mcap_metadata_raw_filtered.csv")

# We only used single stranded reads, so remove the sampleIDs labeled with _1.fastq.gz
metadata <- metadata[!grepl("_1.fastq.gz", metadata$SampleID),]

# Remove .fastq.gz from sample ID name
metadata$SampleID <- gsub(".fastq.gz", "", metadata$SampleID)

# remove cols with no info 
metadata <- metadata[, colSums(is.na(metadata)) == 0]

# Make sampleID the row names 
rownames(metadata) <- metadata$SampleID
```

## Subset count data for only pacuta samples with metadata based on SampleID and make sure rows of metadata = cols of count data
```{r}
mcap_ID <- metadata$SampleID
count_mcap <- select(countdata, all_of(mcap_ID))
all(rownames(metadata) %in% colnames(count_mcap)) # must come out TRUE
```

## Filter reads by proportion of samples containing cutoff value
```{r}
filt <- filterfun(pOverA(0.85, 5)) # set filter values for P over A; I used 0.85 and 5
tfil <- genefilter(count_mcap, filt) # create filter for counts data 
keep <- count_mcap[tfil,] # identify genes to keep based on filter
gn.keep <- rownames(keep)
mcap_counts_filt <- as.matrix(count_mcap[which(rownames(count_mcap) %in% gn.keep),]) 
storage.mode(mcap_counts_filt) <- "integer" # stores count data as integer 
#write.csv(mcap_counts_filt, "Output/DESeq2/mcap/mcap_counts_filt_20220315.csv")
# Checking to make sure rownames in metadata == colnames in counts data 
all(rownames(metadata) %in% colnames(mcap_counts_filt)) # must come out TRUE
```

## Set Treatment as a factor
```{r}
metadata$Treatment <- factor(metadata$Treatment, levels = c("control", "mid", "high"))
data <- DESeqDataSetFromMatrix(countData = mcap_counts_filt, colData = metadata, design = ~ Treatment)
```

## Expression visualization
First we are going to log-transform the data using a variance stabilizing transforamtion (vst). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects. To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4. Otherwise, there could be artefacts in our results.
```{r}
SF.data <- estimateSizeFactors(data) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than 4 to use vst
SF.data
print(sizeFactors(SF.data)) #view size factors
# size factors all less than 4, can use VST
vst <- vst(data, blind = FALSE) 
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
#    function: y = a/x + b, and a local regression fit was automatically substituted.
#    specify fitType='local' or 'mean' to avoid this message next time.
## Should I specify a fitType?
head(assay(vst), 3) # view data
sampleDists <- dist(t(assay(vst))) # calculate distance matrix
sampleDistMatrix <- as.matrix(sampleDists) # create distance matrix
rownames(sampleDistMatrix) <- colnames(vst) # assign row names
colnames(sampleDistMatrix) <- NULL # assign col names 
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255) # assign colors 
pheatmap(sampleDistMatrix, # plot matrix
         clustering_distance_rows = sampleDists, # cluster rows
         clustering_distance_cols = sampleDists, # cluster cols
         col=colors) # set colors
plotPCA(vst, intgroup = c("Treatment")) # plot PCA of samples with all data 
```

## Run DESeq2
```{r}
DEG.int <- DESeq(data) # run differential expression test by treatment using wald test 
# estimating size factors
# estimating dispersions
# gene-wise dispersion estimates
# mean-dispersion relationship
# final dispersion estimates
# fitting model and testing
DEG.int.res <- results(DEG.int) # save DE results 
resultsNames(DEG.int) # view DE results 
# [1] "Intercept"                 "Treatment_mid_vs_control"  "Treatment_high_vs_control"
```

## Differential expression comparisons by treatment 
### Compare control and mid
```{r}
DEG_control_vs_mid <- results(DEG.int, contrast = c("Treatment", "control", "mid"))
DEG_control_vs_mid

DEG_control_vs_mid <- as.data.frame(DEG_control_vs_mid) # make full results into a df
DEG_control_vs_mid["Treatment_Compare"] <- "CvsMid" # add treatment comparison col
DEG_control_vs_mid.sig.num <- sum(DEG_control_vs_mid$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_control_vs_mid.sig.num # 114 DEGs
DEG_control_vs_mid.sig <- subset(DEG_control_vs_mid, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_mid.sig["Treatment_Compare"] <- "CvsMid" # add treatment comparison col
DEG_control_vs_mid.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_mid.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_mid.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_mid.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize
DEG_control_vs_mid.sig.list <- as.data.frame(counts(DEG_control_vs_mid.sig.list))
DEG_control_vs_mid.sig.list_full <- cbind(DEG_control_vs_mid.sig, DEG_control_vs_mid.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_mid.sig.list_full, "Output/DESeq2/mcap/mcap_C_vs_Mid_DEG_20220315.csv")
```

### Compare control and high
```{r}
DEG_control_vs_high <- results(DEG.int, contrast = c("Treatment", "control", "high"))
DEG_control_vs_high

DEG_control_vs_high <- as.data.frame(DEG_control_vs_high) # make full results into a df
DEG_control_vs_high["Treatment_Compare"] <- "CvsHigh" # add treatment comparison col
DEG_control_vs_high.sig.num <- sum(DEG_control_vs_high$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_control_vs_high.sig.num # 116 DEGs
DEG_control_vs_high.sig <- subset(DEG_control_vs_high, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_high.sig["Treatment_Compare"] <- "CvsHigh" # add treatment comparison col
DEG_control_vs_high.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_high.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_high.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_high.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize
DEG_control_vs_high.sig.list <- as.data.frame(counts(DEG_control_vs_high.sig.list))
DEG_control_vs_high.sig.list_full <- cbind(DEG_control_vs_high.sig, DEG_control_vs_high.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_high.sig.list_full, "Output/DESeq2/mcap/mcap_C_vs_High_DEG_20220315.csv")
```

### Compare mid and high
```{r}
DEG_mid_vs_high <- results(DEG.int, contrast = c("Treatment", "mid", "high"))
DEG_mid_vs_high

DEG_mid_vs_high <- as.data.frame(DEG_mid_vs_high) # make full results into a df
DEG_mid_vs_high["Treatment_Compare"] <- "MidvsHigh" # add treatment comparison col
DEG_mid_vs_high.sig.num <- sum(DEG_mid_vs_high$padj <0.05, na.rm = T) # identify # of significant pvalues with 10%FDR (padj<0.1) -  jk using 0.05
DEG_mid_vs_high.sig.num # 0 DEGs 
```

## Make full list of differentially expressed genes and treatments 
```{r}
# Create gene_id from rownames
DEG_control_vs_mid.sig.list_full$gene_id <- rownames(DEG_control_vs_mid.sig.list_full)
rownames(DEG_control_vs_mid.sig.list_full) <- NULL
DEG_control_vs_high.sig.list_full$gene_id <- rownames(DEG_control_vs_high.sig.list_full)
rownames(DEG_control_vs_high.sig.list_full) <- NULL

# bind all DEGs
DEGs.all <- rbind(DEG_control_vs_mid.sig.list_full,
                  DEG_control_vs_high.sig.list_full) # make sure column names are the same in all dfs before binding!
dim(DEGs.all) # 230 x 19
length(unique(DEGs.all$gene_id)) # 157 unique DEGs between all treatments 
#write.csv(DEGs.all, "Output/DESeq2/mcap/mcap_DEGs_AllTreatments_20220315.csv")

# Bind all genes
gene.all <- rbind(DEG_control_vs_mid,
                  DEG_control_vs_high,
                  DEG_mid_vs_high)
#write.csv(gene.all, "Output/DESeq2/mcap/mcap_AllGenes_AllTreatments_20220315.csv")
```

## Identify unique genes from DEGs in CvsMid and CvsHigh
```{r}
DEGs.all_mcap <- DEGs.all$gene_id
DEGs.all_mcap <- unique(DEGs.all_mcap)
DEGs.all_mcap <- as.data.frame(DEGs.all_mcap) 
dim(DEGs.all_mcap) # 157 unique DEGs among treatment comparisons 
```

## 
```{r}
unique.sig.list <- data[which(rownames(data) %in% DEGs.all_mcap$DEGs), ] # subset list of sig transcripts from original count data
#write.csv(counts(unique.sig.list), file = "Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv")
SFtest <- estimateSizeFactors(unique.sig.list)
print(sizeFactors(SFtest))
unique.vst.sig <- varianceStabilizingTransformation(unique.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
```

## PCA plot of diff-expressed genes 
```{r}
mcap_DEGPCAdata <- plotPCA(unique.vst.sig, intgroup = c("Treatment"), returnData=TRUE)
percentVar_pca_mcap <- round(100*attr(mcap_DEGPCAdata, "percentVar")) #plot PCA of samples with all data
mcap_DEGPCAplot <- ggplot(mcap_DEGPCAdata, aes(PC1, PC2, color=Treatment)) +
  geom_point(size=8) +
  xlab(paste0("PC1: ",percentVar_pca_mcap[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_pca_mcap[2],"% variance")) +
  scale_color_manual(values = c(control="gray", mid = "darksalmon", high = "darkred")) +
  coord_fixed() +
  ggtitle(label = "M. capitata") +
  theme_bw() + #Set background color
  theme(legend.text = element_text(size=18), 
        #legend.position="none",
        plot.background = element_blank(),
        legend.title = element_text(size=18, face="bold"), 
        axis.text = element_text(size=18), 
        axis.title = element_text(size=18,  face="bold"), 
        plot.title = element_text(size = 25, face = "italic", hjust = 0.5))
mcap_DEGPCAplot
# PCA plot is of differentially expressed genes only
PC.info <- mcap_DEGPCAplot$data
ggsave("Output/Figs/mcap/mcap_DEGs_PCA_20220326.jpeg", mcap_DEGPCAplot, width = 15, height = 25, units = "cm")
ggsave("Output/Figs/mcap/mcap_DEGs_PCA_20220326.pdf", mcap_DEGPCAplot, width = 15, height = 25, units = "cm")
```







