# Title: Sediment stress - plob GOseq
# Author: Jill Ashey
# date: 1/24/2021

### Gene Ontology Analysis of Differentially Expressed Genes in Plob
# Using updated GO terms (terms generated by Blast2GO, Interproscan, and Uniprot )


# Load packages 
library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("forcats")
library("gridExtra")
library("gapminder")

# Obtain names of all expressed ofav genes (poverA = 0.85,5), and all differentially expressed planuala genes (p<0.05)
gcounts_filt_plob <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/plob/plob_counts_filt.csv", header = TRUE)
dim(gcounts_filt_plob) # 15369 rows x 13
for ( col in 1:ncol(gcounts_filt_plob)){
  colnames(gcounts_filt_plob)[col] <-  gsub("X", "", colnames(gcounts_filt_plob)[col])
}
colnames(gcounts_filt_plob)[1] <-"gene_id"
head(gcounts_filt_plob)
dim(gcounts_filt_plob) # 15369 x 13

# Load DEGs
DEG_plob <- read.csv("~/Desktop/plob_unique.sig.list_20210326.csv", header = TRUE)
dim(DEG_plob) # 153 x 13
for ( col in 1:ncol(DEG_plob)){
  colnames(DEG_plob)[col] <-  gsub("X", "", colnames(DEG_plob)[col])
}
colnames(DEG_plob)[1] <-"gene_id"
dim(DEG_plob) # 153 x 13

# Read in length data (calculated directly from transcripts) and merge with filtered counts 
length_Plob <- read.csv("~/Desktop/length.mRNA_Plob.csv")
length_Plob$gene_id <- gsub(".m1", "", length_Plob$gene_id)
length_Plob$gene_id <- gsub("model", "TU", length_Plob$gene_id)
length_merge <- merge(length_Plob, gcounts_filt_plob, by = "gene_id")




#### Build GOSEQ vector 
#GOseq requires a vector of all genes, all differentially expressed genes, and gene lengths
# Make DEG/gene vector 
DEG <- filter(length_merge, gene_id%in%DEG_plob$gene_id) #make vector of differentially expressed genes
dim(DEG) # 153 rows 
DEG_names <- as.vector(DEG$gene_id)
gene_vector=as.integer(length_merge$gene_id%in%DEG_names)
names(gene_vector)=length_merge$gene_id

# Make ID vector 
IDvector <- length_merge$gene_id

# Make length vector
lengthVector <- length_merge$length

# Calculate Probability Weighting Function
DEG.pwf<-nullp(gene_vector, IDvector, bias.data=lengthVector) #weight vector by length of gene
# plot looks weird...does this mean that there is not much length bias in the dataset?





#### Prepare GO term dataframe 
annot <- read.csv("Desktop/PutnamLab/Repositories/FunctionalAnnotation/FunctionalAnnotation/final_Annotations/plob_FullAnnot.csv", header = TRUE)
annot$gene_id <- gsub("model", "TU", annot$gene_id)
GO.annot <- select(annot, gene_id, GO.IDs)
GO.annot$GO.IDs <- gsub(",", ";", GO.annot$GO.IDs)
splitted <- strsplit(as.character(GO.annot$GO.ID), ";") #split into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.annot$gene_id, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row
colnames(GO.terms) <- c("gene_id", "GO.ID")
GO.terms$GO.ID <- gsub("F:", "", GO.terms$GO.ID)
GO.terms$GO.ID <- gsub("P:", "", GO.terms$GO.ID)
GO.terms$GO.ID <- gsub("C:", "", GO.terms$GO.ID)
GO.terms$GO.ID <- gsub(" ", "", GO.terms$GO.ID)
GO.terms[GO.terms == "NA"] <- NA

head(GO.terms)
tail(GO.terms)
GO.terms$GO.ID<- as.character(GO.terms$GO.ID)
GO.terms$GO.ID <- replace_na(GO.terms$GO.ID, "unknown")
GO.terms$GO.ID <- as.factor(GO.terms$GO.ID)
GO.terms$gene_id <- as.factor(GO.terms$gene_id)
GO.terms <- unique(GO.terms)

dim(GO.terms) # 52861 x 2
head(GO.terms, 10)
tail(GO.terms, 10)


## old GO terms (not my annots)
annot_GO <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/GOseq/plob/plob_GO_20210125.csv", header=TRUE)
annot_GO <- select(annot_GO, -X)
annot_GO$gene_id <- gsub(".m1", "", annot_GO$gene_id)
annot_GO$gene_id <- gsub("model", "TU", annot_GO$gene_id)
annot_GO$GO.ID <- gsub(",", ";", annot_GO$GO.ID)
splitted <- strsplit(as.character(annot_GO$GO.ID), ";") #split into multiple GO ids
GO.terms_OG <- data.frame(v1 = rep.int(annot_GO$gene_id, sapply(splitted, length)), v2 = unlist(splitted)) #list all genes with each of their GO terms in a single row
colnames(GO.terms_OG) <- c("gene_id", "GO.ID")
GO.terms_OG[GO.terms_OG == "NA"] <- NA

head(GO.terms_OG)
tail(GO.terms_OG)
GO.terms_OG$GO.ID<- as.character(GO.terms_OG$GO.ID)
GO.terms_OG$GO.ID <- replace_na(GO.terms_OG$GO.ID, "unknown")
GO.terms_OG$GO.ID <- as.factor(GO.terms_OG$GO.ID)
GO.terms_OG$gene_id <- as.factor(GO.terms_OG$gene_id)
GO.terms_OG <- unique(GO.terms_OG)

dim(GO.terms_OG) # 52861 x 2
head(GO.terms_OG, 10)
tail(GO.terms_OG, 10)






##Find enriched GO terms, "selection-unbiased testing for category enrichment amongst significantly expressed genes for RNA-seq data"
GO.wall<-goseq(DEG.pwf, IDvector, gene2cat=GO.terms_OG, method="Wallenius", use_genes_without_cat=TRUE)
# Using manually entered categories.
# Calculating the p-values...
# 'select()' returned 1:1 mapping between keys and columns
write.csv(GO.wall_FullAnnot, file = "~/Desktop/mcav_GO_ALL_FullAnnot_20210222.csv")



















#Import merged annotated gtf file
map <- read.csv("~/Desktop/GFFs/Plob.merged.annotated.gtf", header=FALSE, sep="\t")
colnames(map) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "attr")
map <- subset(map, id=="transcript") # select only transcripts
dim(map) # 31126 x 9
map$gene_id <- gsub(";.*", "", map$attr)
map$gene_id <- gsub("transcript_id", "", map$gene_id)
map$gene_id <- gsub("-.*", "", map$gene_id)
# Remove any duplicates
map <- map %>% mutate_all(na_if, " ")
map <- select(map, c(scaffold, gene.start, gene.stop, gene_id))
map <- unique(map) 
dim(map) # 31126 x 4

# Import reference annotation file 
ref <- read.table("~/Desktop/GFFs/Plut.GFFannotation.fixed_transcript.gff", sep = "\t", header = FALSE)
colnames(ref) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "attr")
ref <- subset(ref, id == "gene") 
ref$gene_id <- gsub(";.*", "", ref$attr)
ref$gene_id <- gsub("ID=", "", ref$gene_id)
ref <- select(ref, c(scaffold, gene.start, gene.stop, gene_id))
ref <- ref %>% mutate(ref, length = gene.stop - gene.start)
dim(ref) # 31126 x 5

# Build a dataframe that links the gene IDs of expressed genes (poverA = 0.85,5), the gene ids of those genes (from the gene map), and the gene lengths (from the annotation file)
plob_filt.map <- merge(gcounts_filt_plob, map, by = "gene_id", all.x = TRUE)
dim(plob_filt.map) # should be same # of rows as gcounts_filt_plob - it is! 15369

#Find gene positions in ref corresponding to expressed genes 
plob_filt.map_ref <- merge(plob_filt.map, ref, by = "gene_id", all.x= TRUE)
plob_filt.map_ref <- select(plob_filt.map_ref, -c(scaffold.x, gene.start.x, gene.stop.x))

#### Build GOSEQ vector 
#GOseq requires a vector of all genes and all differentially expressed genes. 
#Make gene vector
DEG <- filter(plob_filt.map_ref, gene_id%in%DEG_plob$gene_id) #make vector of differentially expressed genes
dim(DEG) #should be same # of rows as DEG_plob - it is! 154
DEG_names <- as.vector(DEG$gene_id)

# Make vector of all expressed genes (poverA = 0.85,5) with
# non-differentially expressed genes as 0 and differentially expressed genes as 1
gene_vector=as.integer(plob_filt.map_ref$gene_id%in%DEG_names)
names(gene_vector)=plob_filt.map_ref$gene_id
head(gene_vector)

#Make ID vector
ID_vector <- plob_filt.map_ref$gene_id
head(ID_vector)

#Make length vector
length_vector <- plob_filt.map_ref$length
head(length_vector)

#Calculate Probability Weighting Function
DEG.pwf<-nullp(gene_vector, ID_vector, bias.data=length_vector) #weight vector by length of gene
# NAs in nullp: https://rdrr.io/bioc/goseq/man/nullp.html
# plot looks strange ???

### Prepare GO term dataframe 
# Import GO terms
annot_GO <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/GOSeq/plob/plob_GO_20210125.csv", header=TRUE)
annot_GO <- select(annot_GO, -X)
annot_GO$gene_id <- gsub("model", "TU", annot_GO$gene_id) # replace model with TU to match DEG names
annot_GO$GO.ID <- gsub(",", ";", annot_GO$GO.ID)
#annot_GO$gene_id <- gsub("-.*", "", annot_GO$gene_id)

# Split GO terms so there is only 1 GO term per row
GO.terms <- separate_rows(annot_GO, GO.ID, sep = ";")
GO.terms <- unique(GO.terms)
colnames(GO.terms) <- c("gene_id", "GO.ID")
GO.terms <- GO.terms[!grepl("NA", GO.terms$GO.ID),]
GO.terms$gene_id <- gsub(".m1", "", GO.terms$gene_id)

# Merge GO terms with all expressed acerv genes 
GO.terms <- merge(GO.terms, plob_filt.map_ref, by.x="gene_id")
dim(GO.terms) # 22690 x 18

# Clean up GO.terms sd
GO.terms <- select(GO.terms, gene_id, GO.ID)
GO.terms$GO.ID <- replace_na(GO.terms$GO.ID, "unknown")
GO.terms$gene_id <- gsub(" ", "", GO.terms$gene_id)
head(GO.terms, 10)
tail(GO.terms, 10)

### Perform GOseq
# Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
GO.wall<-goseq(DEG.pwf, ID_vector, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)
# Using manually entered categories.
# Calculating the p-values...
# 'select()' returned 1:1 mapping between keys and columns
write.csv(GO.wall, file = "~/Desktop/plob_GO_ALL_20210207.csv")



#### Explore enriched GO terms
class(GO.wall)
dim(GO.wall) # 1966 x 7
head(GO.wall)
tail(GO.wall)
#Find overrepresented (expressed more?) enriched GO terms that are statistically significant at cutoff (p<0.05)
enriched.GO.05 <- GO.wall$category[GO.wall$over_represented_pvalue<0.05]
enriched.GO.05 <- data.frame(enriched.GO.05)
dim(enriched.GO.05) # 17
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")
enriched.GO.05 <- enriched.GO.05[order(-enriched.GO.05$numDEInCat),]
enriched.GO.05$term <- as.factor(enriched.GO.05$term)
head(enriched.GO.05)
#Subset enriched GO terms by category and save as csv
MF <- subset(enriched.GO.05, ontology=="MF")
MF <- MF[order(-MF$numDEInCat),]
CC <- subset(enriched.GO.05, ontology=="CC") # no CC significantly enriched go terms
CC <- CC[order(-CC$numDEInCat),]
BP <- subset(enriched.GO.05, ontology=="BP")
BP <- BP[order(-BP$numDEInCat),]
write.csv(MF, file = "~/Desktop/mcav_MF_Sig_Enriched_GO.05_20210207.csv")
write.csv(CC, file = "~/Desktop/mcav_CC_Sig_Enriched_GO.05_20210207.csv")
write.csv(BP, file = "~/Desktop/mcav_BP_Sig_Enriched_GO.05_20210207.csv")
write.csv(enriched.GO.05, file = "~/Desktop/mcav_Sig_Enriched_GO.05_ALL_20210207.csv")

# Merge enriched GO terms with gene ids (from GO.terms df)


# Merge enriched GO terms with gene ids (from GO.terms df)
colnames(GO.terms) <- c("gene_id","category")
GO.terms_gene_ids <- merge(enriched.GO.05, GO.terms, by = "category", all.x = TRUE) # this combines gene ids with GO info from goseq analysis by GO term
GO.terms_gene_ids <- unique(GO.terms_gene_ids)
GO.terms_gene_ids.plob_genes.map_unique.ref <- merge(GO.terms_gene_ids, plob_filt.map_ref, by = "gene_id") # this combines GO info with counts and length data by gene id 
#GO.terms_gene_ids.pdam_genes.map_unique.ref <- select(GO.terms_gene_ids.pdam_genes.map_unique.ref, c(gene_id, start, stop, length, category, over_represented_pvalue, under_represented_pvalue, numDEInCat, numInCat, term, ontology ))
GO.terms_gene_ids.plob_genes.map_unique.ref <- na.omit(GO.terms_gene_ids.plob_genes.map_unique.ref)
write.csv(GO.terms_gene_ids.plob_genes.map_unique.ref, file = "~/Desktop/plob_GOterms_gene_ids_20210207.csv")

# Merge GO.terms_gene_ids.pdam_genes.map_unique.ref and treatment comparison list 
DEG_treatment <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/plob/plob_DEGs.all_treatment.csv", header = TRUE)
colnames(DEG_treatment)[1] <-"gene_id"
ByTreatment_GO.terms <- merge(DEG_treatment, GO.terms_gene_ids.pdam_genes.map_unique.ref, by = "gene_id", all.x = T) # this combines GO info, counts, and lengths with DEG treatment info by gene id 
ByTreatment_GO.terms <- na.omit(ByTreatment_GO.terms)
ByTreatment_GO.terms <- unique(ByTreatment_GO.terms)
dim(ByTreatment_GO.terms) # 48 x 40
length(unique(ByTreatment_GO.terms$gene_id)) # 37 unique gene ids














colnames(GO.terms) <- c("gene_id","category")
GO.terms_gene_ids <- merge(enriched.GO.05, GO.terms, by = "category", all.x = TRUE)
GO.terms_gene_ids <- unique(GO.terms_gene_ids)
GO.terms_gene_ids.plob_genes.map_unique.ref <- merge(plob_filt.map_ref, GO.terms_gene_ids, by = "gene_id", all.x = TRUE)
GO.terms_gene_ids.plob_genes.map_unique.ref <- select(GO.terms_gene_ids.plob_genes.map_unique.ref, c(gene_id, scaffold.y, gene.start.y, gene.stop.y, length, category, over_represented_pvalue, under_represented_pvalue, numDEInCat, numInCat, term, ontology ))
colnames(GO.terms_gene_ids.plob_genes.map_unique.ref)[colnames(GO.terms_gene_ids.plob_genes.map_unique.ref) == 'scaffold.y'] <- 'scaffold'
colnames(GO.terms_gene_ids.plob_genes.map_unique.ref)[colnames(GO.terms_gene_ids.plob_genes.map_unique.ref) == 'gene.start.y'] <- 'gene.start'
colnames(GO.terms_gene_ids.plob_genes.map_unique.ref)[colnames(GO.terms_gene_ids.plob_genes.map_unique.ref) == 'gene.stop.y'] <- 'gene.stop'
GO.terms_gene_ids.plob_genes.map_unique.ref <- na.omit(GO.terms_gene_ids.plob_genes.map_unique.ref)
GO.terms_gene_ids.plob_genes.map_unique.ref <- unique(GO.terms_gene_ids.plob_genes.map_unique.ref)

# Merge GO.terms_gene_ids.acerv_genes.map_unique.ref and treatment comparison list 
DEG_treatment <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/plob/plob_DEGs.all_treatment.csv", header = TRUE)
colnames(DEG_treatment)[1] <-"gene_id"
ByTreatment_GO.terms <- merge(GO.terms_gene_ids.plob_genes.map_unique.ref, DEG_treatment, by = "gene_id", all.x = TRUE)
ByTreatment_GO.terms <- filter(ByTreatment_GO.terms, over_represented_pvalue > 0)
# now I have a df with DEGs gene names, treatment comparisons, GO terms, and term info
MF_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="MF")
MF_ByTreatment <- MF_ByTreatment[order(-MF_ByTreatment$numDEInCat),]
CC_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="CC")
CC_ByTreatment <- CC_ByTreatment[order(-CC_ByTreatment$numDEInCat),]
BP_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="BP")
BP_ByTreatment <- BP_ByTreatment[order(-BP$numDEInCat),]
write.csv(MF_ByTreatment, file = "~/Desktop/plob_MF_Sig_Enriched_ByTreatment_GO.05_20210125.csv")
write.csv(CC_ByTreatment, file = "~/Desktop/plob_CC_Sig_Enriched_ByTreatment_GO.05_20210125.csv")
write.csv(BP_ByTreatment, file = "~/Desktop/plob_BP_Sig_Enriched_ByTreatment_GO.05_20210125.csv")
write.csv(ByTreatment_GO.terms, file = "~/Desktop/plob_ByTreatment_GO.terms_20210125.csv")

# Plot terms by number of differentially expressed functions
MFplot <- MF %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none"
  ) +
  xlab("") +
  ylab("") +
  ggtitle("Molecular Function") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
MFplot

CCplot <- CC %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none"
  ) +
  xlab("") +
  ylab("") +
  ggtitle("Cellular Component") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
CCplot

BPplot <- BP %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none") +
  xlab("") +
  ylab("") +
  ggtitle("Biological Process") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
BPplot

# Save MF, CC, and BP plot in grid layout
GOplot <- grid.arrange(MFplot, CCplot, BPplot, ncol=3, clip="off")
ggsave("~/Desktop/plob_GOplot_05_20210125.pdf", GOplot, width = 21, height = 21, units = c("in"))

# Combining MF, CC, and BP into one plot and order by pvalue
GOplot2 <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,55) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot2
ggsave("~/Desktop/plob_GOplot2_05_20210125.pdf", GOplot2, width = 28, height = 28, units = c("in"))

# Combining into one plot and order by pvalue 
GOplot_pvalue <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=over_represented_pvalue) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=over_represented_pvalue), color="grey") +
  geom_text(aes(label = numDEInCat), hjust = -1, vjust = 0.5, size = 3) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,0.05) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("p-value") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot_pvalue
ggsave("~/Desktop/plob_GOplot_pvalue_05_20210125.pdf", GOplot_pvalue, width = 28, height = 28, units = c("in"))










