# Title: Sediment stress - mcav GOseq
# Author: Jill Ashey
# date: 1/1/2021

### Gene Ontology Analysis of Differentially Expressed Genes in Mcav
# Using updated GO terms (terms generated by Blast2GO, Interproscan, and Uniprot )


# Load packages 
library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("forcats")
library("gridExtra")
library("gapminder")


# Obtain names of all expressed ofav genes (poverA = 0.85,5), and all differentially expressed planuala genes (p<0.05)
gcounts_filt_mcav <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/mcav_counts_filtered.csv", header = TRUE)
dim(gcounts_filt_mcav) # 12873 rows x 16
for ( col in 1:ncol(gcounts_filt_mcav)){
  colnames(gcounts_filt_mcav)[col] <-  gsub("X", "", colnames(gcounts_filt_mcav)[col])
}
colnames(gcounts_filt_mcav)[1] <-"gene_id"
head(gcounts_filt_mcav)

# Load DEGs
DEG_mcav <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/mcav_unique.sig.list.csv", header = TRUE)
dim(DEG_mcav) # 62 x 16
for ( col in 1:ncol(DEG_mcav)){
  colnames(DEG_mcav)[col] <-  gsub("X", "", colnames(DEG_mcav)[col])
}
colnames(DEG_mcav)[1] <-"gene_id"

#Import merged annotated gtf file
map <- read.csv("~/Desktop/GFFs/Mcav.merged.annotated.gtf", header=FALSE, sep="\t")
colnames(map) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "attr")
map <- subset(map, id=="transcript") # select only transcripts
dim(map) # 25605 x 4
map$gene_id <- gsub(";.*", "", map$attr)
map$gene_id <- gsub("transcript_id", "", map$gene_id)
map$gene_id <- gsub("-.*", "", map$gene_id)
# Remove any duplicates
map <- map %>% mutate_all(na_if, " ")
map <- select(map, c(scaffold, gene.start, gene.stop, gene_id))
map <- unique(map) 
dim(map) # 25605 rows

# Import reference annotation file 
ref <- read.table("~/Desktop/GFFs/Mcav.gff.annotations.fixed_transcript.gff3", sep = "\t", header = FALSE)
colnames(ref) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "attr")
ref <- subset(ref, id == "gene") 
ref$gene_id <- gsub(";.*", "", ref$attr)
ref$gene_id <- gsub("ID=", "", ref$gene_id)
ref <- select(ref, c(scaffold, gene.start, gene.stop, gene_id))
ref <- ref %>% mutate(ref, length = gene.stop - gene.start)
dim(ref) # 25142 x 5

# Build a dataframe that links the gene IDs of expressed genes (poverA = 0.85,5), the gene ids of those genes (from the gene map), and the gene lengths (from the annotation file)
mcav_filt.map <- merge(gcounts_filt_mcav, map, by = "gene_id", all.x = TRUE)
dim(mcav_filt.map) # should be same # of rows as gcounts_filt_mcav - it is! 12873

#Find gene positions in ref corresponding to expressed genes 
mcav_filt.map_ref <- merge(mcav_filt.map, ref, by = "gene_id", all.x= TRUE)
mcav_filt.map_ref <- select(mcav_filt.map_ref, -c(scaffold.x, gene.start.x, gene.stop.x))

#### Build GOSEQ vector 
#GOseq requires a vector of all genes and all differentially expressed genes. 
#Make gene vector
DEG <- filter(mcav_filt.map_ref, gene_id%in%DEG_mcav$gene_id) #make vector of differentially expressed genes
dim(DEG) #should be 62
DEG_names <- as.vector(DEG$gene_id)

# Make vector of all expressed genes (poverA = 0.85,5) with
# non-differentially expressed genes as 0 and differentially expressed genes as 1
gene_vector=as.integer(mcav_filt.map_ref$gene_id%in%DEG_names)
names(gene_vector)=mcav_filt.map_ref$gene_id
head(gene_vector)

#Make ID vector
ID_vector <- mcav_filt.map_ref$gene_id
head(ID_vector)

#Make length vector
length_vector <- mcav_filt.map_ref$length
head(length_vector)

#Calculate Probability Weighting Function
DEG.pwf<-nullp(gene_vector, ID_vector, bias.data=length_vector) #weight vector by length of gene
# plot looks strange 

### Prepare GO term dataframe 
# Import GO terms
annot_GO <- read.csv("~/Desktop/mcav_GO_20210101.csv", header=TRUE)
annot_GO <- select(annot_GO, -X)
annot_GO$gene_id <- gsub("-.*", "", annot_GO$gene_id)

# Split GO terms so there is only 1 GO term per row
GO.terms <- separate_rows(annot_GO, GO.ID, sep = ";")
colnames(GO.terms) <- c("gene_id", "GO.ID")
GO.terms <- GO.terms[!grepl("NA", GO.terms$GO.ID),]

# Merge GO terms with all expressed acerv genes 
GO.terms <- merge(mcav_filt.map_ref, GO.terms, by.x="gene_id")
dim(GO.terms)

# Clean up GO.terms sd
GO.terms <- select(GO.terms, gene_id, GO.ID)
GO.terms$GO.ID <- replace_na(GO.terms$GO.ID, "unknown")
head(GO.terms, 10)
tail(GO.terms, 10)

### Perform GOseq
# Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
GO.wall<-goseq(DEG.pwf, ID_vector, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)
# Using manually entered categories.
# Calculating the p-values...
# 'select()' returned 1:1 mapping between keys and columns
# Warning message:
#   In goseq(DEG.pwf, ID_vector, gene2cat = GO.terms, method = "Wallenius",  :
#              Missing length data for 56% of genes.  Accuarcy of GO test will be reduced.
# need to look into this error 
write.csv(GO.wall, file = "~/Desktop/mcav_GO_ALL_20210101.csv")

#### Explore enriched GO terms
class(GO.wall)
dim(GO.wall) # 1570 x 7
head(GO.wall)
tail(GO.wall)
#Find overrepresented (expressed more?) enriched GO terms that are statistically significant at cutoff (p<0.05)
enriched.GO.05 <- GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05 <- data.frame(enriched.GO.05)
dim(enriched.GO.05) # 33
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")
enriched.GO.05 <- enriched.GO.05[order(-enriched.GO.05$numDEInCat),]
enriched.GO.05$term <- as.factor(enriched.GO.05$term)
head(enriched.GO.05)
#Subset enriched GO terms by category and save as csv
MF <- subset(enriched.GO.05, ontology=="MF")
MF <- MF[order(-MF$numDEInCat),]
CC <- subset(enriched.GO.05, ontology=="CC") # no CC significantly enriched go terms
CC <- CC[order(-CC$numDEInCat),]
BP <- subset(enriched.GO.05, ontology=="BP")
BP <- BP[order(-BP$numDEInCat),]
write.csv(MF, file = "~/Desktop/mcav_MF_Sig_Enriched_GO.05_20210101.csv")
write.csv(CC, file = "~/Desktop/mcav_CC_Sig_Enriched_GO.05_20210101.csv")
write.csv(BP, file = "~/Desktop/mcav_BP_Sig_Enriched_GO.05_20210101.csv")
write.csv(enriched.GO.05, file = "~/Desktop/mcav_Sig_Enriched_GO.05_ALL_20210101.csv")

# Merge enriched GO terms with gene ids (from GO.terms df)
colnames(GO.terms) <- c("gene_id","category")
GO.terms_gene_ids <- merge(enriched.GO.05, GO.terms, by = "category", all.x = TRUE)
GO.terms_gene_ids <- unique(GO.terms_gene_ids)
GO.terms_gene_ids.mcav_genes.map_unique.ref <- merge(mcav_filt.map_ref, GO.terms_gene_ids, by = "gene_id", all.x = TRUE)
GO.terms_gene_ids.mcav_genes.map_unique.ref <- select(GO.terms_gene_ids.mcav_genes.map_unique.ref, c(gene_id, scaffold.y, gene.start.y, gene.stop.y, length, category, over_represented_pvalue, under_represented_pvalue, numDEInCat, numInCat, term, ontology ))
colnames(GO.terms_gene_ids.mcav_genes.map_unique.ref)[colnames(GO.terms_gene_ids.mcav_genes.map_unique.ref) == 'scaffold.y'] <- 'scaffold'
colnames(GO.terms_gene_ids.mcav_genes.map_unique.ref)[colnames(GO.terms_gene_ids.mcav_genes.map_unique.ref) == 'gene.start.y'] <- 'gene.start'
colnames(GO.terms_gene_ids.mcav_genes.map_unique.ref)[colnames(GO.terms_gene_ids.mcav_genes.map_unique.ref) == 'gene.stop.y'] <- 'gene.stop'
GO.terms_gene_ids.mcav_genes.map_unique.ref <- na.omit(GO.terms_gene_ids.mcav_genes.map_unique.ref)
GO.terms_gene_ids.mcav_genes.map_unique.ref <- unique(GO.terms_gene_ids.mcav_genes.map_unique.ref)

# Merge GO.terms_gene_ids.acerv_genes.map_unique.ref and treatment comparison list 
DEG_treatment <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/mcav_DEGs.all_treatment.csv", header = TRUE)
DEG_treatment <- select(DEG_treatment, -X)
colnames(DEG_treatment)[1] <-"gene_id"
ByTreatment_GO.terms <- merge(GO.terms_gene_ids.mcav_genes.map_unique.ref, DEG_treatment, by = "gene_id", all.x = TRUE)
ByTreatment_GO.terms <- na.omit(ByTreatment_GO.terms)
# now I have a df with DEGs gene names, treatment comparisons, GO terms, and term info
MF_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="MF")
MF_ByTreatment <- MF_ByTreatment[order(-MF_ByTreatment$numDEInCat),]
CC_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="CC")
CC_ByTreatment <- CC_ByTreatment[order(-CC_ByTreatment$numDEInCat),]
BP_ByTreatment <- subset(ByTreatment_GO.terms, ontology=="BP")
BP_ByTreatment <- BP_ByTreatment[order(-BP$numDEInCat),]
write.csv(ByTreatment_GO.terms, file = "~/Desktop/mcav_ByTreatment_GO.terms_20210101.csv")
write.csv(MF_ByTreatment, file = "~/Desktop/mcav_MF_Sig_Enriched_ByTreatment_GO.05_20210101.csv")
write.csv(CC_ByTreatment, file = "~/Desktop/mcav_CC_Sig_Enriched_ByTreatment_GO.05_20210101.csv")
write.csv(BP_ByTreatment, file = "~/Desktop/mcav_BP_Sig_Enriched_ByTreatment_GO.05_20210101.csv")

# Plot terms by number of differentially expressed functions
MFplot <- MF %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none"
  ) +
  xlab("") +
  ylab("") +
  ggtitle("Molecular Function") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
MFplot

CCplot <- CC %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none"
  ) +
  xlab("") +
  ylab("") +
  ggtitle("Cellular Component") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
CCplot

BPplot <- BP %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_point(size=3, color="#69b3a2") +
  coord_flip() +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="none") +
  xlab("") +
  ylab("") +
  ggtitle("Biological Process") + #add a main title
  theme(plot.title = element_text(face = 'bold', 
                                  size = 12, 
                                  hjust = 0)) +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())#Set the plot background
BPplot

# Save MF, CC, and BP plot in grid layout
GOplot <- grid.arrange(MFplot, CCplot, BPplot, ncol=3, clip="off")
ggsave("~/Desktop/mcav_GOplot_05_20210101.pdf", GOplot, width = 21, height = 21, units = c("in"))

# Combining MF, CC, and BP into one plot and order by pvalue
GOplot2 <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,25) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot2
ggsave("~/Desktop/mcav_GOplot2_05_20210101.pdf", GOplot2, width = 28, height = 28, units = c("in"))

# Combining into one plot and order by pvalue 
GOplot_pvalue <- enriched.GO.05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=over_represented_pvalue) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=over_represented_pvalue), color="grey") +
  geom_text(aes(label = numDEInCat), hjust = -1, vjust = 0.5, size = 3) +
  geom_point(size=3, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,0.05) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("p-value") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes
GOplot_pvalue
ggsave("~/Desktop/mcav_GOplot_pvalue_05_20210101.pdf", GOplot_pvalue, width = 28, height = 28, units = c("in"))














