---
title: "OrthoFinder"
author: "jillashey"
date: "1/28/2022"
output: html_document
---



20220920 note to self - try this link https://sfirke.github.io/janitor/reference/get_dupes.htmlhttps://sfirke.github.io/janitor/reference/get_dupes.html to identify duplicates in dataset 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(VennDiagram)
library(arsenal) # comparedf function
#library(UpSetR)
library(ComplexUpset)

## NOTE: cannot use UpSetR and ComplexUpset at the same time
```

Why did we run OrthoFinder? 

ran OrthoFinder on bluewaves, got lots of output. One file ('Output/OrthoFinder/Orthogroups.tsv') has the Orthogroups listed with all the genes found in that orthogroup for each species. I'm going to first remove any orthogroups that have blank cells (ie no protein from a species in that orthogroup). Then I'll read in my DEG info and filter the protein names from each species. With that info, I'll make a Venn diagram showing shared orthogroups among species 


20221125
Null Hypothesis: There are no differences in the genes that are differentially expressed across species within the orthologous gene sets

Step 1 - load Otrhogroups.tsv 
Step 2 - filter to those orthogroups found only in all species 
Step 3 - expand table to show each gene as a row and the gene ID and orthogroup ID as the 2 columns
Step 4 - merge the table with the DEG table by gene name and repeat this for each species
Step 5 - merge all DEGs within orthogroups tables for all 6 species into one table that will be used to generate the Upset Plot
Color code by FL or HI
Symbol code by morphology
Step 6 - figure out stats tests for location and for morphology

*Step 1*: Load Orthogroup data 
```{r}
# Read in orthofinder data 
ortho <- read.table(file = 'Output/OrthoFinder/Orthogroups.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
#ortho <- na.omit(ortho)
colnames(ortho) <- c("Orthogroup", "Acervicornis", "Ofaveolata", "Mcavernosa", "Mcapitata", "Pacuta", "Plobata")
length(unique(ortho$Orthogroup)) # 21688 total orthogroups
```

*Step 2*: Filter to those orthogroups found only in all species 
```{r}
# Remove blanks from dataset, so all that remained were orthogroups shared by all species. 
ortho <- na.omit(ortho)
length(unique(ortho$Orthogroup)) # 9216 'filtered' orthogroups
```

*Step 3*: Expand table to show each gene as a row and the gene ID and orthogroup ID as the 2 columns
```{r}
# Split by species (probably easier/cleaner way to do this)
acerv_ortho <- select(ortho, c("Orthogroup", "Acervicornis"))
mcav_ortho <- select(ortho, c("Orthogroup", "Mcavernosa"))
ofav_ortho <- select(ortho, c("Orthogroup", "Ofaveolata"))
mcap_ortho <- select(ortho, c("Orthogroup", "Mcapitata"))
pacuta_ortho <- select(ortho, c("Orthogroup", "Pacuta"))
plob_ortho <- select(ortho, c("Orthogroup", "Plobata"))

# Now split strings of gene names into cols w/ corresponding orthogroup
acerv_ortho <- acerv_ortho %>% mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
acerv_ortho <- na.omit(acerv_ortho)
acerv_ortho$Acervicornis <- gsub(" ", "", acerv_ortho$Acervicornis)

mcav_ortho <- mcav_ortho %>% mutate(Mcavernosa = strsplit(as.character(Mcavernosa), ",")) %>% 
    unnest(Mcavernosa)
mcav_ortho <- na.omit(mcav_ortho)
mcav_ortho$Mcavernosa <- gsub(" ", "", mcav_ortho$Mcavernosa)

ofav_ortho <- ofav_ortho %>% mutate(Ofaveolata = strsplit(as.character(Ofaveolata), ",")) %>% 
    unnest(Ofaveolata)
ofav_ortho <- na.omit(ofav_ortho)
ofav_ortho$Ofaveolata <- gsub(" ", "", ofav_ortho$Ofaveolata)

mcap_ortho <- mcap_ortho %>% mutate(Mcapitata = strsplit(as.character(Mcapitata), ",")) %>% 
    unnest(Mcapitata)
mcap_ortho <- na.omit(mcap_ortho)
mcap_ortho$Mcapitata <- gsub(" ", "", mcap_ortho$Mcapitata)

pacuta_ortho <- pacuta_ortho %>% mutate(Pacuta = strsplit(as.character(Pacuta), ",")) %>% 
    unnest(Pacuta)
pacuta_ortho <- na.omit(pacuta_ortho)
pacuta_ortho$Pacuta <- gsub(" ", "", pacuta_ortho$Pacuta)

plob_ortho <- plob_ortho %>% mutate(Plobata = strsplit(as.character(Plobata), ",")) %>% 
    unnest(Plobata)
plob_ortho <- na.omit(plob_ortho)
plob_ortho$Plobata <- gsub(" ", "", plob_ortho$Plobata)
```

Figure out how to make upset plots for these so that it does frequency counts of each orthogroup for each species and shows shared ones? 



*Step 4*: Merge the table with the DEG table by gene name and repeat this for each species
### Acerv 
```{r}
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# name gene id col in acerv_ortho
colnames(acerv_ortho)[2] <- "gene_id"
acerv_ortho$gene_id <- gsub("model", "TU", acerv_ortho$gene_id)
acerv_ortho$gene_id <- gsub(" ", "", acerv_ortho$gene_id)

# Merge dfs by gene_id
acerv <- merge(acerv_ortho, acerv_DEG, by = "gene_id")
length(unique(acerv$gene_id)) # 119 DEGs found in orthogroups 
length(unique(acerv$Orthogroup)) # 102 orthogroups
write.csv(unique(acerv$Orthogroup), "Output/OrthoFinder/acerv_uniq_ortho_20221125.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(acerv, "Output/OrthoFinder/acerv_DEGs_ortho_20221125.csv")
```

### Mcav 
```{r}
mcav_DEG <- read.csv("Output/DESeq2/mcav/mcav_unique.sig.list_20210208.csv", header = T)[,1]
mcav_DEG <- as.data.frame(mcav_DEG)
colnames(mcav_DEG)[1] <- "gene_id"

# name gene id col in acerv_ortho
colnames(mcav_ortho)[2] <- "gene_id"
mcav_ortho$gene_id <- gsub("-RA", "", mcav_ortho$gene_id)
mcav_ortho$gene_id <- gsub(" ", "", mcav_ortho$gene_id)

# Merge dfs by gene_id
mcav <- merge(mcav_ortho, mcav_DEG, by = "gene_id")
length(unique(mcav$gene_id)) # 31 DEGs found in orthogroups 
length(unique(mcav$Orthogroup)) # 28 orthogroups 
write.csv(unique(mcav$Orthogroup), "Output/OrthoFinder/mcav_uniq_ortho_20221125.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(mcav, "Output/OrthoFinder/mcav_DEGs_ortho_20221125.csv")
```

### Ofav 
```{r}
ofav_DEG <- read.csv("Output/DESeq2/ofav/ofav_unique.sig.list_20220111.csv", header = T)[,1]
ofav_DEG <- as.data.frame(ofav_DEG)
colnames(ofav_DEG)[1] <- "gene_id"

# ofav is so annoying bc it has its prot seqs labeled with XP_xxx but when running STAR, the LOC term was used for gene id. Now I have to read in original gff file and merge original gff with ofav_ortho df to get LOC term

# read in gff 
ofav <- read.csv("~/Desktop/GFFs/ofav/GCF_002042975.1_ofav_dov_v1_genomic.gff", header=FALSE, sep="\t", skip=6) # read in gff from NCBI
colnames(ofav) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene") # name cols
ofav <- na.omit(ofav)
ofav <- subset(ofav, id == "CDS")
ofav$prot_id <- gsub(".*cds-","", ofav$gene) # remove everything before -cds
ofav$prot_id <- gsub(";.*","", ofav$prot_id) # remove everything after ;
ofav$gene_id <- gsub(".*LOC","", ofav$gene) # remove everything before LOC
ofav$gene_id <- gsub(";.*","", ofav$gene_id) # remove everything after ;
ofav$gene_id <- gsub("iso.*","", ofav$gene_id) # remove everything after iso
ofav$gene_id <- paste0("LOC", ofav$gene_id)
#write.csv("~/Desktop/GFFs/ofav/ofav_gff_LOCTerm.gff") # writing gff file so that it always has LOC term already isolated
# select cols with XP- and LOC info
ofav_sub <- ofav[,c("prot_id", "gene_id")]
ofav_sub<- unique(ofav_sub)

# temporarily change names of ofav_ortho to match ofav_sub
colnames(ofav_ortho)[2] <- "prot_id"
ofav_ortho$prot_id <- gsub(" ", "", ofav_ortho$prot_id)

# Merge gff info / ortho results and remove prot_id
merge <- merge(ofav_ortho, ofav_sub, by = "prot_id")
merge <- select(merge, c(Orthogroup, gene_id))
merge <- unique(merge)

# Merge dfs by gene_id
ofav <- merge(merge, ofav_DEG, by = "gene_id")
length(unique(ofav$gene_id)) # 2 DEGs found in orthogroups 
length(unique(ofav$Orthogroup)) # 2 orthogroups 
write.csv(unique(ofav$Orthogroup), "Output/OrthoFinder/ofav_uniq_ortho_20221125.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(ofav, "Output/OrthoFinder/ofav_DEGs_ortho_20221125.csv")
```

### Mcap
```{r}
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# name gene id col in mcap_ortho
colnames(mcap_ortho)[2] <- "gene_id"
mcap_ortho$gene_id <- gsub(".t1", "", mcap_ortho$gene_id)
mcap_ortho$gene_id <- gsub(" ", "", mcap_ortho$gene_id)

# Merge dfs by gene_id
mcap <- merge(mcap_ortho, mcap_DEG, by = "gene_id")
length(unique(mcap$gene_id)) # 87 DEGs found in orthogroups 
length(unique(mcap$Orthogroup)) # 80 orthogroups 
write.csv(unique(mcap$Orthogroup), "Output/OrthoFinder/mcap_uniq_ortho_20221125.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(mcap, "Output/OrthoFinder/mcap_DEGs_ortho_20221125.csv")
```

### Pacuta 
```{r}
pacuta_DEG <- read.csv("Output/DESeq2/pacuta/pacuta_unique.sig.list_20220116.csv", header = T)[,1]
pacuta_DEG <- as.data.frame(pacuta_DEG)
colnames(pacuta_DEG)[1] <- "gene_id"

# name gene id col in acerv_ortho
colnames(pacuta_ortho)[2] <- "gene_id"
pacuta_ortho$gene_id <- gsub(" ", "", pacuta_ortho$gene_id)

# Merge dfs by gene_id
pacuta <- merge(pacuta_ortho, pacuta_DEG, by = "gene_id")
length(unique(pacuta$gene_id)) # 123 DEGs found in orthogroups 
length(unique(pacuta$Orthogroup)) # 100 orthogroups 
write.csv(unique(pacuta$Orthogroup), "Output/OrthoFinder/pacuta_uniq_ortho_20221125.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(pacuta, "Output/OrthoFinder/pacuta_DEGs_ortho_20221125.csv")
```

### Plob 
```{r}
plob_DEG <- read.csv("Output/DESeq2/plob/plob_unique.sig.list_20210326.csv", header = T)[,1]
plob_DEG <- as.data.frame(plob_DEG)
colnames(plob_DEG)[1] <- "gene_id"

# name gene id col in acerv_ortho
colnames(plob_ortho)[2] <- "gene_id"
plob_ortho$gene_id <- gsub(".m1", "", plob_ortho$gene_id)
plob_ortho$gene_id <- gsub("model", "TU", plob_ortho$gene_id)
plob_ortho$gene_id <- gsub(" ", "", plob_ortho$gene_id)

# Merge dfs by gene_id
plob <- merge(plob_ortho, plob_DEG, by = "gene_id")
length(unique(plob$gene_id)) # 66 DEGs found in orthogroups 
length(unique(plob$Orthogroup)) # 56 orthogroups 
write.csv(unique(plob$Orthogroup), "Output/OrthoFinder/plob_uniq_ortho_20221125.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(plob, "Output/OrthoFinder/plob_DEGs_ortho_20221125.csv") 
```

*Step 5*: Merge all DEGs within orthogroups tables for all 6 species into one table that will be used to generate the Upset Plot

This [website](https://statisticsglobe.com/r-dplyr-join-inner-left-right-full-semi-anti) has nice explanations of the different dplyr joining functions 
```{r}
# Replace NAs with 0 and other values to 1 - may need to do this first to avoid duplicates bc some genes have multiple orthogroups or vice versa 
acerv <- unique(acerv)
acerv$gene_id <- ifelse(is.na(acerv$gene_id),0,1)
acerv <- unique(acerv)

mcav <- unique(mcav)
mcav$gene_id <- ifelse(is.na(mcav$gene_id),0,1)
mcav <- unique(mcav)

ofav <- unique(ofav)
ofav$gene_id <- ifelse(is.na(ofav$gene_id),0,1)
ofav <- unique(ofav)

mcap <- unique(mcap)
mcap$gene_id <- ifelse(is.na(mcap$gene_id),0,1)
mcap <- unique(mcap)

pacuta <- unique(pacuta)
pacuta$gene_id <- ifelse(is.na(pacuta$gene_id),0,1)
pacuta <- unique(pacuta)

plob <- unique(plob)
plob$gene_id <- ifelse(is.na(plob$gene_id),0,1)
plob <- unique(plob)


# Merge all orthogroup tables
full1 <- full_join(acerv, mcav, by = "Orthogroup")
full2 <- full_join(full1, ofav, by = "Orthogroup")
full3 <- full_join(full2, mcap, by = "Orthogroup")
full4 <- full_join(full3, pacuta, by = "Orthogroup")
full5 <- full_join(full4, plob, by = "Orthogroup")

# Reorganize table & rename columns 
full <- full5[,c(2,1,3,4,5,6,7)] # moved orthogroup column to be first column 
colnames(full) <- c("Orthogroup", "Acervicornis", "Mcavernosa", "Ofaveolata", "Mcapitata", "Pacuta", "Plobata")
full <- unique(full) # remove duplicate rows

# Replace NAs with 0 and other values to 1 
full$Acervicornis <- ifelse(is.na(full$Acervicornis),0,1)
full$Mcavernosa <- ifelse(is.na(full$Mcavernosa),0,1)
full$Ofaveolata <- ifelse(is.na(full$Ofaveolata),0,1)
full$Mcapitata <- ifelse(is.na(full$Mcapitata),0,1)
full$Pacuta <- ifelse(is.na(full$Pacuta),0,1)
full$Plobata <- ifelse(is.na(full$Plobata),0,1)
```

Make upset plot 

Trying the code from this stackover flow post: https://stackoverflow.com/questions/67094573/upsetr-sets-bar-interacting-different-color-and-its-sets-intersections

Vignette for ComplexUpset [here](https://krassowski.github.io/complex-upset/articles/Examples_R.html)
PDF: https://cran.r-project.org/web/packages/ComplexUpset/ComplexUpset.pdf
```{r}
library(ComplexUpset)

## change column/species names so that they will look nice in the plot (i.e. change Acervicornis to A. cervicornis and make italic in plot). I guess I could do it in the above step
colnames(full) <- c("Orthogroup", "A. cervicornis", "M. cavernosa", "O. faveolata", "M. capitata", "P. acuta", "P. lobata")

# plot
upset_ortho <- upset(
  full,
  c("A. cervicornis", "M. cavernosa", "O. faveolata", "M. capitata", "P. acuta", "P. lobata"),
  queries=list(
    upset_query(set='A. cervicornis', fill='chartreuse3'),
    upset_query(set='M. cavernosa', fill='chartreuse3'),
    upset_query(set='O. faveolata', fill='chartreuse3'),
    upset_query(set='M. capitata', fill='chocolate2'),
    upset_query(set='P. acuta', fill='chocolate2'),
    upset_query(set='P. lobata', fill='chocolate2')
  ),
  base_annotations=list(
    'Intersection size'=(     # this creates the bar plot showing intersection size (top right corner)
        intersection_size(
        bar_number_threshold=0.85,  # show all numbers on top of bars
        width=0.9,   # reduce width of the bars
        text = list(size=10), # change text size over bars
        text_colors = c(on_background = "black", on_bar = "white"), # change text color for text over bars
      )
      + scale_y_continuous(expand=expansion(mult=c(0, 0.05))) # add some space on the top of the bars
      + theme(
          panel.grid.major=element_blank(), # hide grid lines
          panel.grid.minor=element_blank(), # hide grid lines
          axis.line=element_line(colour='black', size = 2), # show axis lines
          axis.title.y = element_text(size = 35, colour = "black"), 
          axis.text.y = element_text(size = 30, colour = "black")
      )
    )
  ),
  stripes=upset_stripes(    # this changes the color of the stripes in the matrix (underneath the bar plot, bottom right corner)
    geom=geom_segment(size=35),  # make the stripes larger  ---  REMEMBER to check color order for all species
    colors=c("#E4E408", # Ofav - massive 
             "#E4E408",  # Mcav - massive 
             "#E4E408",  # Plob - massive 
             "#A0A0A0", # Mcap - intermediate 
             "#0066CC", # Pacuta - branching
             "#0066CC")  # Acerv - branching
  ),
    #0066CC = blue; #E4E408 = yellow; #A0A0A0 = gray
  matrix=intersection_matrix( # this changes the circles and connecting lines in the matrix
      geom=geom_point(
        shape='circle filled', # to prevent connectors from getting the color, use `fill` instead of `color`, together with `shape='circle filled'
        size=10,
        stroke=0.6
      ), 
      segment=geom_segment(size=2)
  ),
  set_sizes=( # this changes the set size bar plot (in the bottom left corner)
    upset_set_size(geom=geom_bar(width=0.5, colour = "black"))
    + theme(
      axis.line.x=element_line(colour='black', size = 2),
      axis.line.y = element_line(colour = 'black'),
      axis.ticks.x=element_line(),
      axis.text.x = element_text(size = 25, colour = "black"),
      axis.title.x = element_text(size = 30, colour = "black"),
    )
  ),
  themes = upset_modify_themes(
       list(
           'intersections_matrix' = theme(text = element_text(size = 50, face = "italic", colour = "black")), # text to the left of the matrix
          'overall_sizes' = theme(axis.text.x = element_text(angle = 0)) # text underneath set size plot
       )),
  sort_sets = 'descending', # sort sets in descending order
  sort_intersections = 'descending' # sort intersections in descending order
)
upset_ortho 

ggsave("Output/Figs/upset/upset_ortho_20221205.pdf", upset_ortho, width = 40, height = 20, units = c("in"))
ggsave("Output/Figs/upset/upset_ortho_20221205.png", upset_ortho, width = 40, height = 20, units = c("in"))
```
Will add legend in powerpoint real quick


This is different code than what I used in the GO upset [plots](https://github.com/JillAshey/SedimentStress/blob/master/RAnalysis/GOslim.Rmd). As I move forward, I'll assess which version of the code is better/easier to work with. 



20221129
Statistical tests for upset plots - This [post](https://krassowski.github.io/complex-upset/articles/Examples_R.html#running-statistical-tests) has lots of good examples for using Complex Upset package. It has a statistical test section as well (section 2). I'm going to see if I can run a statistical test with my data following the code from the post.
```{r}
## First do w/ movie dataset
movies <- read.csv( system.file("extdata", "movies.csv", package = "UpSetR"), header=TRUE, sep=";" )
head(movies)

genres <-colnames(movies)[3:19]
genres

#movies[genres] = movies[genres] == 1
#t(head(movies[genres], 3))

# stat test 
blah <- upset_test(movies, genres)


## with my data 
species <- colnames(full)[2:7]

upset_test(full, species)
```









20221020
Plotted Venn Diagram using this website: https://bioinformatics.psb.ugent.be/webtools/Venn/. can't figure out how to do it in R

How to plot using the above website^^^^. It will not be the most beautiful thing but thats okay
- In excel, from the xxxx_uniq_ortho file, remove all columns except the one that has orthogroups. Delete the column name as well 
- Go to the website. It will have places to upload files and name the files. Upload each file and write the species name for the name of the file in the website. 
- Scroll down to OUTPUT control. Select Venn Shape as Non-symmetric and Fill as Colored
- Click submit. may take like 30 secs
- It will provide the list names, number of elements in each list, and number of unique elements
- Download the image result and text results
- The image will not look great. Since I'm comparing 6 species, it looks pretty rough. But it will show the # of orthogroups shared between lists
- The text result will give the text version of the image. It shows the lists that overlap and which specific element is a part of that overlap. Yay! 
- QC BELOW

20221020
Doing ortho stuff again. This time, I did not omit NAs from the data set when I read it in (lines 23-33). 
This is the QC step to make sure the website didn't mess anything up. Intersect 2 of the lists and compare the intersection results in R and the results from the website. Everything SHOULD match!
```{r}

## Acerv & Pacuta
## remove the species col
test_pa <- pacuta[,-c(1)]
test_ac <- acerv[,-c(1)]
intersect(test_ac, test_pa)
# [1] "OG0000414" "OG0011284" "OG0000989" "OG0000127" "OG0012176" "OG0000487" "OG0000469" "OG0003466" "OG0007132"

## Acerv & Plob
## remove the species col
test_pl <- plob[,-c(1)]
test_ac <- acerv[,-c(1)]
intersect(test_ac, test_pl)
#[1] "OG0000414" "OG0011284" "OG0000989" "OG0000127" "OG0012176" "OG0000487" "OG0000469" "OG0003466" "OG0007132"
#[1] "OG0000000" "OG0000414" "OG0000308" "OG0000221" "OG0000487" "OG0000469" "OG0000008" "OG0000491"

## Acerv & Ofav
## remove the species col
test_of <- ofav[,-c(1)]
test_ac <- acerv[,-c(1)]
intersect(test_ac, test_of)
#[1] "OG0000000"
```



And then make one df with all species info for orthogroups?
```{r}
acerv$Species <- "A.cervicornis"
mcav$Species <- "M.cavernosa"
ofav$Species <- "O.faveolata"
mcap$Species <- "M.capitata"
pacuta$Species <- "P.acuta"
plob$Species <- "P.lobata"

test <- rbind(acerv, mcav, ofav, mcap, pacuta, plob)
length(unique(test$Orthogroup))

# Write to csv 
write.csv(test, "Output/OrthoFinder/AllSpecies_DEGs_ortho_20221020.csv") 
```

20221021
I'm going to try to merge the DEGs & GO terms with the orthogroups. Let's try with acerv first 

### Acerv
```{r}
# using acerv 

# rename gene id in acerv so that there is 'model' instead of 'TU' in the name 
acerv$gene_id <- gsub("TU", "model", acerv$gene_id)

# read in DEG & GO data
acerv_go <- read.csv("Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
acerv_all <- merge(acerv_go, acerv, by = "gene_id", all = T)

acerv_all <- acerv_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(acerv_all$gene_id)) # 83 - not all genes had an orthogroup
length(unique(acerv_all$Orthogroup)) # 66 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(test, "Output/OrthoFinder/acerv_DEGs_GO_ortho_20221020.csv") 
```

Why not just merge the orthogroups with "acerv_sub_ByTreatment_GO.terms_20211104.csv" first instead of "acerv_sub_unique.sig.list_20210219.csv"?
- The GO terms df only has the gene ids for those genes that received GO annotations, so it left out many of the other DEGs in the dataset that were not annotated

Now do with the rest of species 

### Mcav 
```{r}
# read in DEG & GO data
mcav_go <- read.csv("Output/GOSeq/mcav/mcav_ByTreatment_GO.terms_20211109.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
mcav_all <- merge(mcav_go, mcav, by = "gene_id", all = T)

mcav_all <- mcav_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(mcav_all$gene_id)) # 35 - not all genes had an orthogroup
length(unique(mcav_all$Orthogroup)) # 28 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(mcav_all, "Output/OrthoFinder/mcav_DEGs_GO_ortho_20221020.csv")    
```

### Ofav 
```{r}
# read in DEG & GO data
ofav_go <- read.csv("Output/GOSeq/ofav/ofav_ByTreatment_GO.terms_20220111.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
ofav_all <- merge(ofav_go, ofav, by = "gene_id", all = T)

ofav_all <- ofav_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(ofav_all$gene_id)) # 4 - not all genes had an orthogroup
length(unique(ofav_all$Orthogroup)) # 2 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(ofav_all, "Output/OrthoFinder/ofav_DEGs_GO_ortho_20221020.csv") 
```

### Mcap
```{r}
# read in DEG & GO data
mcap_go <- read.csv("Output/GOSeq/mcap/mcap_ByTreatment_GO.terms_20220417.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
mcap_all <- merge(mcap_go, mcap, by = "gene_id", all = T)

mcap_all <- mcap_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(mcap_all$gene_id)) # 85 - not all genes had an orthogroup
length(unique(mcap_all$Orthogroup)) # 73 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(mcap_all, "Output/OrthoFinder/mcap_DEGs_GO_ortho_20221020.csv")    
```

### Pacuta 
```{r}
# read in DEG & GO data
pact_go <- read.csv("Output/GOSeq/pacuta/pacuta_ByTreatment_GO.terms_20220121.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
pact_all <- merge(pact_go, pacuta, by = "gene_id", all = T)

pact_all <- pact_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(pact_all$gene_id)) # 147 - not all genes had an orthogroup
length(unique(pact_all$Orthogroup)) # 100 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(pact_all, "Output/OrthoFinder/pacuta_DEGs_GO_ortho_20221020.csv")    
```

### Plob 
```{r}
# read in DEG & GO data
plob_go <- read.csv("Output/GOSeq/plob/plob_ByTreatment_GO.terms_20211106.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
plob_all <- merge(plob_go, plob, by = "gene_id", all = T)

plob_all <- plob_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(plob_all$gene_id)) # 83 - not all genes had an orthogroup
length(unique(plob_all$Orthogroup)) # 49 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(plob_all, "Output/OrthoFinder/plob_DEGs_GO_ortho_20221020.csv")    
```


20221021
Still need to make changes to orthogroup info in most of the figures/tables/supp info

20221115

Trying to somehow merge the DEGs and the comparison files

Acerv v. Mcav - for this section, I read in the orthogroups comparison file for Acerv and Mcav. Then I isolated the Acerv gene ids and orthogroups from this file. I read in the Acerv DEG file and merged it with the Acerv gene id/orthogroup dataframe. I did the same with the Mcav info. Then I merged the Acerv orthogroup/DEG info and Mcav orthogroup/DEG info into a single df to get the orthogroups that were shared between the DEGs of both Mcav and Acerv. Then I isolated the shared Acerv/DEG info into a separate df and renamed the gene id column to "gene_id" instead of gene_id.x. I read in the Acerv GO term file and I merged the GO term file with the shared Acerv orthogroup/DEG info to get the functional info for Acerv's shared orthogroup w/ Mcav. I repeated the above process for Mcav (this time subsetting Mcav columns, reading in Mcav data, etc). 
```{r}
# Read in comparison file for Acerv & Mcav
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Acerv_assembly_v1.0.protein__v__Mcavernosa.maker.proteins.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Acervicornis", "Mcavernosa")


############# ACERV #############

# Isolate Acerv genes and orthogroups
acerv_ortho <- compare[,1:2]
acerv_ortho <- acerv_ortho %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(acerv_ortho) <- c("Orthogroup", "gene_id")
acerv_ortho$gene_id <- gsub("model", "TU", acerv_ortho$gene_id)
acerv_ortho$gene_id <- gsub(" ", "", acerv_ortho$gene_id)
length(unique(acerv_ortho$Orthogroup)) # 11413 orthogroups
length(unique(acerv_ortho$gene_id)) # 15638 genes

# Read in Acerv DEGs
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# Merge Acerv shared orthogroup & DEG info
acerv_DEG_ortho <- merge(acerv_DEG, acerv_ortho, by = "gene_id")
length(unique(acerv_DEG_ortho$gene_id)) # 112 DEGs
length(unique(acerv_DEG_ortho$Orthogroup)) # 107 orthogroups ?
# okay what this result is saying to me is that there are 112 DEGs of Acerv that fit into the Acerv orthogroup df. There are 107 orthogroups (shared between Acerv and Mcav) that have Acerv DEGs

# Read in GO term file for Acerv and merge GO terms w/ Acerv orthogroup and DEG info
#acerv_DEG_ortho <- acerv_DEG_ortho[,c(1,2)] # separate orthogroups and gene id for acerv
#colnames(acerv_DEG_ortho) <- c("Orthogroups", "gene_id") # renamed col names
acerv_go <- read.csv("Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv", header = T)[,-1]
acerv_go$gene_id <- gsub("model", "TU", acerv_go$gene_id)

# Merge GO terms w/ Acerv orthogroup and DEG info
acerv_DEG_ortho_go <- merge(acerv_go, acerv_DEG_ortho, by = "gene_id")
length(unique(acerv_DEG_ortho_go$gene_id)) # 46 DEGs
length(unique(acerv_DEG_ortho_go$Orthogroup)) # 46 orthogroups





############# MCAV #############

# Isolate Mcav genes and orthogroups
mcav_ortho <- compare[,c(1,3)]
mcav_ortho <- mcav_ortho %>% 
    mutate(Mcavernosa = strsplit(as.character(Mcavernosa), ",")) %>% 
    unnest(Mcavernosa)
colnames(mcav_ortho) <- c("Orthogroup", "gene_id")
mcav_ortho$gene_id <- gsub("-RA", "", mcav_ortho$gene_id)
mcav_ortho$gene_id <- gsub(" ", "", mcav_ortho$gene_id)
length(unique(mcav_ortho$Orthogroup)) # 11413 shared orthogroups
length(unique(mcav_ortho$gene_id)) # 15280 genes

# Read in Mcav DEGs
mcav_DEG <- read.csv("Output/DESeq2/mcav/mcav_unique.sig.list_20210208.csv", header = T)[,1]
mcav_DEG <- as.data.frame(mcav_DEG)
colnames(mcav_DEG)[1] <- "gene_id"

# Merge Mcav orthogroup & DEG info
mcav_DEG_ortho <- merge(mcav_DEG, mcav_ortho, by = "gene_id")
length(unique(mcav_DEG_ortho$gene_id)) # 30 DEGs
length(unique(mcav_DEG_ortho$Orthogroup)) # 28 shared orthogroups

# Read in GO term file for Mcav 
#mcav_DEG_ortho <- mcav_DEG_ortho[,c(1,3)] # separate orthogroups and gene id for mcav
#colnames(mcav_DEG_ortho) <- c("Orthogroups", "gene_id") # renamed col names
mcav_go <- read.csv("Output/GOSeq/mcav/mcav_ByTreatment_GO.terms_20211109.csv", header = T)[,-1] # read in GO terms 

# Merge GO terms w/ Mcav orthogroup and DEG info
mcav_DEG_ortho_go <- merge(mcav_go, mcav_DEG_ortho, by = "gene_id")
length(unique(mcav_DEG_ortho_go$gene_id)) # 16 DEGs
length(unique(mcav_DEG_ortho_go$Orthogroup)) # 15 orthogroups



############# SHARED #############

# Merge dfs of both species by Orthogroup. The dfs contain orthogroups, DEGs and functional annotation
acerv_DEG_ortho_go$Species <- "Acervicornis"
mcav_DEG_ortho_go$Species <- "Mcavernosa"
DEG_ortho_go <- merge(acerv_DEG_ortho_go, mcav_DEG_ortho_go, by = "Orthogroup") 
# there are no annotated DEGs that share orthogroups by species 

# merge DEG and orthogroup dfs for both species - this will have no functional annotation, which increases the number of DEGs/orthogroups
DEG_ortho <- merge(acerv_DEG_ortho, mcav_DEG_ortho, by = "Orthogroup")
length(unique(DEG_ortho$Orthogroup)) # 2 orthogroups shared between DEGs between these species
length(unique(DEG_ortho$gene_id.x)) # 2 Acerv DEGs in the shared orthogroups 
length(unique(DEG_ortho$gene_id.y)) # 4 Mcav DEGs in the shared orthogroups 
```

Acerv v. Mcap
```{r}
# Read in comparison file for Acerv & Mcap
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Acerv_assembly_v1.0.protein__v__Montipora_capitata_HIv2.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Acervicornis", "Mcapitata")


############# ACERV #############

# Isolate Acerv genes and orthogroups
acerv_ortho <- compare[,1:2]
acerv_ortho <- acerv_ortho %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(acerv_ortho) <- c("Orthogroup", "gene_id")
acerv_ortho$gene_id <- gsub("model", "TU", acerv_ortho$gene_id)
acerv_ortho$gene_id <- gsub(" ", "", acerv_ortho$gene_id)
length(unique(acerv_ortho$Orthogroup)) # 15008 orthogroups
length(unique(acerv_ortho$gene_id)) # 23524 genes

# Read in Acerv DEGs
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# Merge Acerv shared orthogroup & DEG info
acerv_DEG_ortho <- merge(acerv_DEG, acerv_ortho, by = "gene_id")
length(unique(acerv_DEG_ortho$gene_id)) # 159 DEGs
length(unique(acerv_DEG_ortho$Orthogroup)) # 139 orthogroups 

# Read in GO term file for Acerv and merge GO terms w/ Acerv orthogroup and DEG info
#acerv_DEG_ortho <- acerv_DEG_ortho[,c(1,2)] # separate orthogroups and gene id for acerv
#colnames(acerv_DEG_ortho) <- c("Orthogroups", "gene_id") # renamed col names
acerv_go <- read.csv("Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv", header = T)[,-1]
acerv_go$gene_id <- gsub("model", "TU", acerv_go$gene_id)

# Merge GO terms w/ Acerv orthogroup and DEG info
acerv_DEG_ortho_go <- merge(acerv_go, acerv_DEG_ortho, by = "gene_id")
length(unique(acerv_DEG_ortho_go$gene_id)) # 66 DEGs
length(unique(acerv_DEG_ortho_go$Orthogroup)) # 60 orthogroups
length(unique(acerv_DEG_ortho_go$term)) # 284 GO terms




############# MCAP #############

# Isolate Mcap genes and orthogroups
mcap_ortho <- compare[, c(1,3)]
mcap_ortho <- mcap_ortho %>% 
    mutate(Mcapitata = strsplit(as.character(Mcapitata), ",")) %>% 
    unnest(Mcapitata)
colnames(mcap_ortho) <- c("Orthogroup", "gene_id")
mcap_ortho$gene_id <- gsub(".t1", "", mcap_ortho$gene_id)
mcap_ortho$gene_id <- gsub(" ", "", mcap_ortho$gene_id)
length(unique(mcap_ortho$Orthogroup)) # 15008 orthogroups
length(unique(mcap_ortho$gene_id)) # 27686 genes

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_DEG_ortho <- merge(mcap_DEG, mcap_ortho, by = "gene_id")
length(unique(mcap_DEG_ortho$gene_id)) # 101 DEGs
length(unique(mcap_DEG_ortho$Orthogroup)) # 97 orthogroups 

# Read in GO term file for Mcap and merge GO terms w/ Mcap orthogroup and DEG info
mcap_go <- read.csv("Output/GOSeq/mcap/mcap_ByTreatment_GO.terms_20220417.csv", header = T)[,-1]

# Merge GO terms w/ Mcap orthogroup and DEG info
mcap_DEG_ortho_go <- merge(mcap_go, mcap_DEG_ortho, by = "gene_id")
length(unique(mcap_DEG_ortho_go$gene_id)) # 69 DEGs
length(unique(mcap_DEG_ortho_go$Orthogroup)) # 66 orthogroups
length(unique(mcap_DEG_ortho_go$term)) # 240 GO terms




############# SHARED #############

# Merge dfs of both species by Orthogroup. The dfs contain orthogroups, DEGs and functional annotation
acerv_DEG_ortho_go$Species <- "Acervicornis"
mcap_DEG_ortho_go$Species <- "Mcapitata"
DEG_ortho_go <- merge(acerv_DEG_ortho_go, mcap_DEG_ortho_go, by = "Orthogroup") 
length(unique(DEG_ortho_go$Orthogroup)) # 2 orthogroups shared between DEGs of this species 
length(unique(DEG_ortho_go$gene_id.x)) # 2 Acerv DEGs in the shared orthogroups 
length(unique(DEG_ortho_go$gene_id.y)) # 3 Mcap DEGs in the shared orthogroups 
length(unique(DEG_ortho_go$term.x)) # 5 Acerv GO terms in the shared orthogroups 
length(unique(DEG_ortho_go$term.y)) # 19 Mcap GO terms in the shared orthogroups 


# merge DEG and orthogroup dfs for both species - this will have no functional annotation, which increases the number of DEGs/orthogroups
DEG_ortho <- merge(acerv_DEG_ortho, mcap_DEG_ortho, by = "Orthogroup")
length(unique(DEG_ortho$Orthogroup)) # 8 orthogroups shared between DEGs between these species
length(unique(DEG_ortho$gene_id.x)) # 10 Acerv DEGs in the shared orthogroups
length(unique(DEG_ortho$gene_id.y)) # 10 Mcap DEGs in the shared orthogroups
```

Acerv v. Ofav
```{r}
# Read in comparison file for Acerv & Ofav
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Acerv_assembly_v1.0.protein__v__GCF_002042975.1_ofav_dov_v1_protein.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Acervicornis", "Ofaveolata")

############# ACERV #############

# Isolate Acerv genes and orthogroups
acerv_ortho <- compare[,1:2]
acerv_ortho <- acerv_ortho %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(acerv_ortho) <- c("Orthogroup", "gene_id")
acerv_ortho$gene_id <- gsub("model", "TU", acerv_ortho$gene_id)
acerv_ortho$gene_id <- gsub(" ", "", acerv_ortho$gene_id)
length(unique(acerv_ortho$Orthogroup)) # 12604 orthogroups
length(unique(acerv_ortho$gene_id)) # 17762 genes

# Read in Acerv DEGs
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# Merge Acerv shared orthogroup & DEG info
acerv_DEG_ortho <- merge(acerv_DEG, acerv_ortho, by = "gene_id")
length(unique(acerv_DEG_ortho$gene_id)) # 138 DEGs
length(unique(acerv_DEG_ortho$Orthogroup)) # 123 orthogroups 

# Read in GO term file for Acerv and merge GO terms w/ Acerv orthogroup and DEG info
acerv_go <- read.csv("Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv", header = T)[,-1]
acerv_go$gene_id <- gsub("model", "TU", acerv_go$gene_id)

# Merge GO terms w/ Acerv orthogroup and DEG info
acerv_DEG_ortho_go <- merge(acerv_go, acerv_DEG_ortho, by = "gene_id")
length(unique(acerv_DEG_ortho_go$gene_id)) # 59 DEGs
length(unique(acerv_DEG_ortho_go$Orthogroup)) # 54 orthogroups
length(unique(acerv_DEG_ortho_go$term)) # 283 GO terms




############# OFAV #############

# Isolate Ofav genes and orthogroups
ofav_ortho <- compare[,c(1,3)]
ofav_ortho <- ofav_ortho %>% 
    mutate(Ofaveolata = strsplit(as.character(Ofaveolata), ",")) %>% 
    unnest(Ofaveolata)
colnames(ofav_ortho) <- c("Orthogroup", "gene_id")
ofav_ortho$gene_id <- gsub(" ", "", ofav_ortho$gene_id)
length(unique(ofav_ortho$Orthogroup)) # 12604 orthogroups
length(unique(ofav_ortho$gene_id)) # 21127 genes

# SPECIFIC TO OFAV - ofav is so annoying bc it has its prot seqs labeled with XP_xxx but when running STAR, the LOC term was used for gene id. Now I have to read in original gff file and merge original gff with ofav_ortho df to get LOC term
ofav <- read.csv("~/Desktop/GFFs/ofav/GCF_002042975.1_ofav_dov_v1_genomic.gff", header=FALSE, sep="\t", skip=6) # read in gff from NCBI
colnames(ofav) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene") # name cols
ofav <- na.omit(ofav)
ofav <- subset(ofav, id == "CDS")
ofav$prot_id <- gsub(".*cds-","", ofav$gene) # remove everything before -cds
ofav$prot_id <- gsub(";.*","", ofav$prot_id) # remove everything after ;
ofav$gene_id <- gsub(".*LOC","", ofav$gene) # remove everything before LOC
ofav$gene_id <- gsub(";.*","", ofav$gene_id) # remove everything after ;
ofav$gene_id <- gsub("iso.*","", ofav$gene_id) # remove everything after iso
ofav$gene_id <- paste0("LOC", ofav$gene_id)
write.csv(ofav, "~/Desktop/GFFs/ofav/ofav_gff_LOCTerm.csv") # writing gff file so that it always has LOC term already isolated
# select cols with XP- and LOC info
ofav_sub <- ofav[,c("prot_id", "gene_id")]
ofav_sub<- unique(ofav_sub)
# temporarily change names of ofav_ortho to match ofav_sub
colnames(ofav_ortho)[2] <- "prot_id"
ofav_ortho$prot_id <- gsub(" ", "", ofav_ortho$prot_id)
# Merge gff info / ortho results and remove prot_id
ofav_ortho_LOC <- merge(ofav_ortho, ofav_sub, by = "prot_id")
ofav_ortho_LOC <- select(ofav_ortho_LOC, c(Orthogroup, gene_id))

# Read in Ofav DEGs
ofav_DEG <- read.csv("Output/DESeq2/ofav/ofav_unique.sig.list_20220111.csv", header = T)[,1]
ofav_DEG <- as.data.frame(ofav_DEG)
colnames(ofav_DEG)[1] <- "gene_id"

# Merge Ofav shared orthogroup & DEG info
ofav_DEG_ortho <- merge(ofav_DEG, ofav_ortho_LOC, by = "gene_id")
length(unique(ofav_DEG_ortho$gene_id)) # 1 DEGs
length(unique(ofav_DEG_ortho$Orthogroup)) # 1 orthogroups 

# Read in GO term file for Ofav and merge GO terms w/ Ofav orthogroup and DEG info
ofav_go <- read.csv("Output/GOSeq/ofav/ofav_ByTreatment_GO.terms_20220111.csv", header = T)[,-1]

# Merge GO terms w/ Ofav orthogroup and DEG info
ofav_DEG_ortho_go <- merge(ofav_go, ofav_DEG_ortho, by = "gene_id")
## No associated GO terms 







############# SHARED #############

# Merge dfs of both species by Orthogroup. The dfs contain orthogroups, DEGs and functional annotation
## Not relevant, as Ofav did not have any GO terms associated w/ DEGs in its orthogroups

# merge DEG and orthogroup dfs for both species - this will have no functional annotation, which increases the number of DEGs/orthogroups
DEG_ortho <- merge(acerv_DEG_ortho, ofav_DEG_ortho, by = "Orthogroup")
## Nothing shared betweeen the 2 species? Doesn't make sense, they should be sharing OG0000000 (according to Venn diagram maker) 

## I am confused. According to this spreadsheet (Tables S11 & S12; https://docs.google.com/spreadsheets/d/1sfCekpJ9ErJElHVwRCdkQn420je_HvOOaPZCVpKxfwU/edit#gid=632142905), Acerv & Ofav should share OG0000000 as an orthogroup. The associated gene id for Ofav is LOC11004532, which is a DEG for Ofav and has functional annotation info. But when I look for LOC110045321 under the Ofav gff, the associated protein IDs are XP_020606575.1, XP_020606576.1, & XP_020606577.1. None of these correspond to the Ofav protein IDs for OG0000000 (XP_020624579.1, XP_020624580.1, XP_020624578.1, XP_020615354.1, XP_020615355.1, XP_020615398.1, XP_020603602.1, XP_020603603.1, XP_020603601.1, XP_020603604.1). 

## I may have to do the Ofav ones manually? Aka use the spreadsheet to find the gene id and its associated GO terms. Idk I think something is getting lost in translation when I change the XP_xxxx terms to the LOC terms. 

## According to the spreadsheet (Tables S11 & S12; https://docs.google.com/spreadsheets/d/1sfCekpJ9ErJElHVwRCdkQn420je_HvOOaPZCVpKxfwU/edit#gid=632142905), Acerv & Ofav should share OG0000000 as an orthogroup. The associated gene id for Ofav is LOC11004532; the associated gene ids for Acerv are Acerv_evm.TU.Segkk1078_pilon.63, Acerv_evm.TU.Segkk1851_pilon.3, Acerv_evm.TU.Segkk20_pilon.1, Acerv_evm.TU.Segkk2725_pilon.8, Acerv_evm.TU.Segkk3087_pilon.2, Acerv_evm.TU.Segkk3096_pilon.7, Acerv_evm.TU.Segkk3387_pilon.12, Acerv_evm.TU.Segkk3618_pilon.7 & Acerv_evm.TU.Segkk3996_pilon.1. 

## GO terms for LOC11004532: 
# 


```

















Acerv v. Pacuta
```{r}
# Read in comparison file for Acerv & Pacuta
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Acerv_assembly_v1.0.protein__v__Pocillopora_acuta_HIv1.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Acervicornis", "Pacuta")

############# ACERV #############

# Isolate Acerv genes and orthogroups
acerv_ortho <- compare[,1:2]
acerv_ortho <- acerv_ortho %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(acerv_ortho) <- c("Orthogroup", "gene_id")
acerv_ortho$gene_id <- gsub("model", "TU", acerv_ortho$gene_id)
acerv_ortho$gene_id <- gsub(" ", "", acerv_ortho$gene_id)
length(unique(acerv_ortho$Orthogroup)) # 13209 orthogroups
length(unique(acerv_ortho$gene_id)) # 18779 genes

# Read in Acerv DEGs
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# Merge Acerv shared orthogroup & DEG info
acerv_DEG_ortho <- merge(acerv_DEG, acerv_ortho, by = "gene_id")
length(unique(acerv_DEG_ortho$gene_id)) # 127 DEGs
length(unique(acerv_DEG_ortho$Orthogroup)) # 118 orthogroups 

# Read in GO term file for Acerv and merge GO terms w/ Acerv orthogroup and DEG info
acerv_go <- read.csv("Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv", header = T)[,-1]
acerv_go$gene_id <- gsub("model", "TU", acerv_go$gene_id)

# Merge GO terms w/ Acerv orthogroup and DEG info
acerv_DEG_ortho_go <- merge(acerv_go, acerv_DEG_ortho, by = "gene_id")
length(unique(acerv_DEG_ortho_go$gene_id)) # 49 DEGs
length(unique(acerv_DEG_ortho_go$Orthogroup)) # 48 orthogroups
length(unique(acerv_DEG_ortho_go$term)) # 244 GO terms




############# PACUTA #############

# Isolate Pacuta genes and orthogroups
pacuta_ortho <- compare[, c(1,3)]
pacuta_ortho <- pacuta_ortho %>% 
    mutate(Pacuta = strsplit(as.character(Pacuta), ",")) %>% 
    unnest(Pacuta)
colnames(pacuta_ortho) <- c("Orthogroup", "gene_id")
pacuta_ortho$gene_id <- gsub(" ", "", pacuta_ortho$gene_id)
length(unique(pacuta_ortho$Orthogroup)) # 13209 orthogroups
length(unique(pacuta_ortho$gene_id)) # 23303 genes

# Read in Pacuta DEGs
pacuta_DEG <- read.csv("Output/DESeq2/pacuta/pacuta_unique.sig.list_20220116.csv", header = T)[,1]
pacuta_DEG <- as.data.frame(pacuta_DEG)
colnames(pacuta_DEG)[1] <- "gene_id"

# Merge Pacuta orthogroup & DEG info
pacuta_DEG_ortho <- merge(pacuta_DEG, pacuta_ortho, by = "gene_id")
length(unique(pacuta_DEG_ortho$gene_id)) # 145 DEGs
length(unique(pacuta_DEG_ortho$Orthogroup)) # 118 orthogroups 

# Read in GO term file for Pacuta and merge GO terms w/ Pacuta orthogroup and DEG info
pacuta_go <- read.csv("Output/GOSeq/pacuta/pacuta_ByTreatment_GO.terms_20220121.csv", header = T)[,-1]

# Merge GO terms w/ Pacuta orthogroup and DEG info
pacuta_DEG_ortho_go <- merge(pacuta_go, pacuta_DEG_ortho, by = "gene_id")
length(unique(pacuta_DEG_ortho_go$gene_id)) # 104 DEGs
length(unique(pacuta_DEG_ortho_go$Orthogroup)) # 84 orthogroups
length(unique(pacuta_DEG_ortho_go$term)) # 369 GO terms




############# SHARED #############

# Merge dfs of both species by Orthogroup. The dfs contain orthogroups, DEGs and functional annotation
acerv_DEG_ortho_go$Species <- "Acervicornis"
pacuta_DEG_ortho_go$Species <- "Pacuta"
DEG_ortho_go <- merge(acerv_DEG_ortho_go, pacuta_DEG_ortho_go, by = "Orthogroup") 
length(unique(DEG_ortho_go$Orthogroup)) # 1 orthogroups shared between DEGs of this species 
length(unique(DEG_ortho_go$gene_id.x)) # 1 Acerv DEGs in the shared orthogroups 
length(unique(DEG_ortho_go$gene_id.y)) # 1 Pacuta DEGs in the shared orthogroups 
length(unique(DEG_ortho_go$term.x)) # 3 Acerv GO terms in the shared orthogroups 
length(unique(DEG_ortho_go$term.y)) # 1 Pacuta GO term in the shared orthogroups 


# merge DEG and orthogroup dfs for both species - this will have no functional annotation, which increases the number of DEGs/orthogroups
DEG_ortho <- merge(acerv_DEG_ortho, pacuta_DEG_ortho, by = "Orthogroup")
length(unique(DEG_ortho$Orthogroup)) # 9 orthogroups shared between DEGs between these species
length(unique(DEG_ortho$gene_id.x)) # 11 Acerv DEGs in the shared orthogroups
length(unique(DEG_ortho$gene_id.y)) # 13 Pacuta DEGs in the shared orthogroups
```

Acerv v. Plob
```{r}
# Read in comparison file for Acerv & Plob
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Acerv_assembly_v1.0.protein__v__plut2v1.1.proteins.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Acervicornis", "Plob")

# Isolate Acerv genes and orthogroups
acerv_test <- compare[,1:2]
acerv_test <- acerv_test %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(acerv_test) <- c("Orthogroup", "gene_id")
acerv_test$gene_id <- gsub("model", "TU", acerv_test$gene_id)
acerv_test$gene_id <- gsub(" ", "", acerv_test$gene_id)

# Read in Acerv DEGs
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# Merge Acerv orthogroup & DEG info
acerv_merge <- merge(acerv_DEG, acerv_test, by = "gene_id")


# Isolate Plob genes and orthogroups
plob_test <- compare[, c(1,3)]
plob_test <- plob_test %>% 
    mutate(Plob = strsplit(as.character(Plob), ",")) %>% 
    unnest(Plob)
colnames(plob_test) <- c("Orthogroup", "gene_id")
plob_test$gene_id <- gsub(".m1", "", plob_test$gene_id)
plob_test$gene_id <- gsub(" ", "", plob_test$gene_id)

# Read in Plob DEGs
plob_DEG <- read.csv("Output/DESeq2/plob/plob_unique.sig.list_20210326.csv", header = T)[,1]
plob_DEG <- as.data.frame(plob_DEG)
colnames(plob_DEG)[1] <- "gene_id"

# Merge Plob orthogroup & DEG info
plob_merge <- merge(plob_DEG, plob_test, by = "gene_id")

# Merge Acerv orthogroup/DEG info and Plob orthogroup/DEG info
all <- merge(acerv_merge, plob_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 6 orthogroups shared between Acerv & Plob
length(unique(all$gene_id.x)) # 14 DEGs from Acerv in orthogroup comparison between Acerv & Plob
length(unique(all$gene_id.y)) # 7 DEGs from Pacuta in orthogroup comparison between Acerv & Plob

# save as csv? ^^ 
```

Mcav v. Mcap
```{r}
# Read in comparison file for Mcav & Mcap
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Mcavernosa.maker.proteins__v__Montipora_capitata_HIv2.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Mcavernosa", "Mcapitata")

# Isolate Mcav genes and orthogroups
mcav_test <- compare[, c(1,2)]
mcav_test <- mcav_test %>% 
    mutate(Mcavernosa = strsplit(as.character(Mcavernosa), ",")) %>% 
    unnest(Mcavernosa)
colnames(mcav_test) <- c("Orthogroup", "gene_id")
mcav_test$gene_id <- gsub("-RA", "", mcav_test$gene_id)
mcav_test$gene_id <- gsub(" ", "", mcav_test$gene_id)

# Read in Mcav DEGs
mcav_DEG <- read.csv("Output/DESeq2/mcav/mcav_unique.sig.list_20210208.csv", header = T)[,1]
mcav_DEG <- as.data.frame(mcav_DEG)
colnames(mcav_DEG)[1] <- "gene_id"

# Merge Mcav orthogroup & DEG info
mcav_merge <- merge(mcav_DEG, mcav_test, by = "gene_id")



# Isolate Mcap genes and orthogroups
mcap_test <- compare[,1:3]
mcap_test <- mcap_test %>% 
    mutate(Mcapitata = strsplit(as.character(Mcapitata), ",")) %>% 
    unnest(Mcapitata)
colnames(mcap_test) <- c("Orthogroup", "gene_id")
mcap_test$gene_id <- gsub(".t1", "", mcap_test$gene_id)
mcap_test$gene_id <- gsub(" ", "", mcap_test$gene_id)

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_merge <- merge(mcap_DEG, mcap_test, by = "gene_id")

# Merge Mcav orthogroup/DEG info and Mcap orthogroup/DEG info
all <- merge(mcav_merge, mcap_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 0 orthogroups shared between Mcav & Mcap
length(unique(all$gene_id.x)) # 0 DEGs from Acerv in orthogroup comparison between Mcav & Mcap
length(unique(all$gene_id.y)) # 0 DEGs from Pacuta in orthogroup comparison between Mcav & Mcap

# save as csv? ^^ 
```

Mcav v. Ofav
DO THIS!!!!!!!!!!!!!!!!!!!!!!!!******************

Mcav v. Pacuta
```{r}
# Read in comparison file for Mcav & Pacuta















compare <- read.table(file = 'Output/OrthoFinder/species_compare/Mcavernosa.maker.proteins__v__Montipora_capitata_HIv2.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Mcavernosa", "Mcapitata")

# Isolate Mcav genes and orthogroups
mcav_test <- compare[, c(1,2)]
mcav_test <- mcav_test %>% 
    mutate(Mcavernosa = strsplit(as.character(Mcavernosa), ",")) %>% 
    unnest(Mcavernosa)
colnames(mcav_test) <- c("Orthogroup", "gene_id")
mcav_test$gene_id <- gsub("-RA", "", mcav_test$gene_id)
mcav_test$gene_id <- gsub(" ", "", mcav_test$gene_id)

# Read in Mcav DEGs
mcav_DEG <- read.csv("Output/DESeq2/mcav/mcav_unique.sig.list_20210208.csv", header = T)[,1]
mcav_DEG <- as.data.frame(mcav_DEG)
colnames(mcav_DEG)[1] <- "gene_id"

# Merge Mcav orthogroup & DEG info
mcav_merge <- merge(mcav_DEG, mcav_test, by = "gene_id")



# Isolate Mcap genes and orthogroups
mcap_test <- compare[,1:3]
mcap_test <- mcap_test %>% 
    mutate(Mcapitata = strsplit(as.character(Mcapitata), ",")) %>% 
    unnest(Mcapitata)
colnames(mcap_test) <- c("Orthogroup", "gene_id")
mcap_test$gene_id <- gsub(".t1", "", mcap_test$gene_id)
mcap_test$gene_id <- gsub(" ", "", mcap_test$gene_id)

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_merge <- merge(mcap_DEG, mcap_test, by = "gene_id")

# Merge Mcav orthogroup/DEG info and Mcap orthogroup/DEG info
all <- merge(mcav_merge, mcap_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 0 orthogroups shared between Mcav & Mcap
length(unique(all$gene_id.x)) # 0 DEGs from Acerv in orthogroup comparison between Mcav & Mcap
length(unique(all$gene_id.y)) # 0 DEGs from Pacuta in orthogroup comparison between Mcav & Mcap

# save as csv? ^^ 
```

Mcav v. Plob

Ofav v. Mcap
DO THIS!!!!!!!!!!!!!!!!!!!!!!!!******************

Ofav v. Pacuta
DO THIS!!!!!!!!!!!!!!!!!!!!!!!!******************

Ofav v. Plob
DO THIS!!!!!!!!!!!!!!!!!!!!!!!!******************

Mcap v. Pacuta 
```{r}
# Read in comparison file for Mcap & Pacuta
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Montipora_capitata_HIv2.genes.pep__v__Pocillopora_acuta_HIv1.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Mcap", "Pacuta")

# Isolate Mcap genes and orthogroups
mcap_test <- compare[,1:2]
mcap_test <- mcap_test %>% 
    mutate(Mcap = strsplit(as.character(Mcap), ",")) %>% 
    unnest(Mcap)
colnames(mcap_test) <- c("Orthogroup", "gene_id")
mcap_test$gene_id <- gsub(".t1", "", mcap_test$gene_id)
mcap_test$gene_id <- gsub(" ", "", mcap_test$gene_id)

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_merge <- merge(mcap_DEG, mcap_test, by = "gene_id")


# Isolate Pacuta genes and orthogroups
pacuta_test <- compare[, c(1,3)]
pacuta_test <- pacuta_test %>% 
    mutate(Pacuta = strsplit(as.character(Pacuta), ",")) %>% 
    unnest(Pacuta)
colnames(pacuta_test) <- c("Orthogroup", "gene_id")
pacuta_test$gene_id <- gsub(" ", "", pacuta_test$gene_id)

# Read in Pacuta DEGs
pacuta_DEG <- read.csv("Output/DESeq2/pacuta/pacuta_unique.sig.list_20220116.csv", header = T)[,1]
pacuta_DEG <- as.data.frame(pacuta_DEG)
colnames(pacuta_DEG)[1] <- "gene_id"

# Merge Pacuta orthogroup & DEG info
pacuta_merge <- merge(pacuta_DEG, pacuta_test, by = "gene_id")

# Merge Mcap orthogroup/DEG info and Pacuta orthogroup/DEG info
all <- merge(mcap_merge, pacuta_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 7 orthogroups shared between Mcap & Pacuta
length(unique(all$gene_id.x)) # 8 DEGs from Pacuta in orthogroup comparison between Mcap & Pacuta
length(unique(all$gene_id.y)) # 9 DEGs from Plob in orthogroup comparison between Mcap & Pacuta

# save as csv? ^^ 
```

Mcap v. Plob 
```{r}
# Read in comparison file for Mcap & Plob
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Montipora_capitata_HIv2.genes.pep__v__plut2v1.1.proteins.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Mcap", "Plob")

# Isolate Mcap genes and orthogroups
mcap_test <- compare[,1:2]
mcap_test <- mcap_test %>% 
    mutate(Mcap = strsplit(as.character(Mcap), ",")) %>% 
    unnest(Mcap)
colnames(mcap_test) <- c("Orthogroup", "gene_id")
mcap_test$gene_id <- gsub(".t1", "", mcap_test$gene_id)
mcap_test$gene_id <- gsub(" ", "", mcap_test$gene_id)

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_merge <- merge(mcap_DEG, mcap_test, by = "gene_id")


# Isolate Plob genes and orthogroups
plob_test <- compare[, c(1,3)]
plob_test <- plob_test %>% 
    mutate(Plob = strsplit(as.character(Plob), ",")) %>% 
    unnest(Plob)
colnames(plob_test) <- c("Orthogroup", "gene_id")
plob_test$gene_id <- gsub(".m1", "", plob_test$gene_id)
plob_test$gene_id <- gsub(" ", "", plob_test$gene_id)

# Read in Plob DEGs
plob_DEG <- read.csv("Output/DESeq2/plob/plob_unique.sig.list_20210326.csv", header = T)[,1]
plob_DEG <- as.data.frame(plob_DEG)
colnames(plob_DEG)[1] <- "gene_id"

# Merge Plob orthogroup & DEG info
plob_merge <- merge(plob_DEG, plob_test, by = "gene_id")

# Merge Mcap orthogroup/DEG info and Plob orthogroup/DEG info
all <- merge(mcap_merge, plob_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 8 orthogroups shared between Pacuta & Plob
length(unique(all$gene_id.x)) # 10 DEGs from Pacuta in orthogroup comparison between Pacuta & Plob
length(unique(all$gene_id.y)) # 10 DEGs from Plob in orthogroup comparison between Pacuta & Plob

# save as csv? ^^ 
```

Pacuta v. Plob 
```{r}
# Read in comparison file for Pacuta & Plob
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Pocillopora_acuta_HIv1.genes.pep__v__plut2v1.1.proteins.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Pacuta", "Plob")

# Isolate Pacuta genes and orthogroups
pacuta_test <- compare[,1:2]
pacuta_test <- pacuta_test %>% 
    mutate(Pacuta = strsplit(as.character(Pacuta), ",")) %>% 
    unnest(Pacuta)
colnames(pacuta_test) <- c("Orthogroup", "gene_id")
pacuta_test$gene_id <- gsub(" ", "", pacuta_test$gene_id)

# Read in Pacuta DEGs
pacuta_DEG <- read.csv("Output/DESeq2/pacuta/pacuta_unique.sig.list_20220116.csv", header = T)[,1]
pacuta_DEG <- as.data.frame(pacuta_DEG)
colnames(pacuta_DEG)[1] <- "gene_id"

# Merge Pacuta orthogroup & DEG info
pacuta_merge <- merge(pacuta_DEG, pacuta_test, by = "gene_id")


# Isolate Plob genes and orthogroups
plob_test <- compare[, c(1,3)]
plob_test <- plob_test %>% 
    mutate(Plob = strsplit(as.character(Plob), ",")) %>% 
    unnest(Plob)
colnames(plob_test) <- c("Orthogroup", "gene_id")
plob_test$gene_id <- gsub(".m1", "", plob_test$gene_id)
plob_test$gene_id <- gsub(" ", "", plob_test$gene_id)

# Read in Plob DEGs
plob_DEG <- read.csv("Output/DESeq2/plob/plob_unique.sig.list_20210326.csv", header = T)[,1]
plob_DEG <- as.data.frame(plob_DEG)
colnames(plob_DEG)[1] <- "gene_id"

# Merge Plob orthogroup & DEG info
plob_merge <- merge(plob_DEG, plob_test, by = "gene_id")

# Merge Pacuta orthogroup/DEG info and Plob orthogroup/DEG info
all <- merge(pacuta_merge, plob_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 6 orthogroups shared between Pacuta & Plob
length(unique(all$gene_id.x)) # 8 DEGs from Pacuta in orthogroup comparison between Pacuta & Plob
length(unique(all$gene_id.y)) # 6 DEGs from Plob in orthogroup comparison between Pacuta & Plob

# save as csv? ^^ 
```



