---
title: "OrthoFinder"
author: "jillashey"
date: "1/28/2022"
output: html_document
---



20220920 note to self - try this link https://sfirke.github.io/janitor/reference/get_dupes.htmlhttps://sfirke.github.io/janitor/reference/get_dupes.html to identify duplicates in dataset 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(VennDiagram)
library(arsenal) # comparedf function
```

Why did we run OrthoFinder? 

ran OrthoFinder on bluewaves, got lots of output. One file ('Output/OrthoFinder/Orthogroups.tsv') has the Orthogroups listed with all the genes found in that orthogroup for each species. I'm going to first remove any orthogroups that have blank cells (ie no protein from a species in that orthogroup). Then I'll read in my DEG info and filter the protein names from each species. With that info, I'll make a Venn diagram showing shared orthogroups among species 

```{r}
# Read in orthofinder data 
ortho <- read.table(file = 'Output/OrthoFinder/Orthogroups.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
#ortho <- na.omit(ortho)
colnames(ortho) <- c("Orthogroup", "Acervicornis", "Ofaveolata", "Mcavernosa", "Mcapitata", "Pacuta", "Plobata")
length(unique(ortho$Orthogroup)) # 21688 total orthogroups

# Remove blanks from dataset, so all that remained were orthogroups shared by all species. 
#ortho <- na.omit(ortho)
#length(unique(ortho$Orthogroup)) # 9216 'filtered' orthogroups
```

```{r}
# Split by species (probably easier/cleaner way to do this)
acerv_ortho <- select(ortho, c("Orthogroup", "Acervicornis"))
mcav_ortho <- select(ortho, c("Orthogroup", "Mcavernosa"))
ofav_ortho <- select(ortho, c("Orthogroup", "Ofaveolata"))
mcap_ortho <- select(ortho, c("Orthogroup", "Mcapitata"))
pacuta_ortho <- select(ortho, c("Orthogroup", "Pacuta"))
plob_ortho <- select(ortho, c("Orthogroup", "Plobata"))

# Now split strings of gene names into cols w/ corresponding orthogroup
acerv_ortho <- acerv_ortho %>% mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
acerv_ortho <- na.omit(acerv_ortho)
acerv_ortho$Acervicornis <- gsub(" ", "", acerv_ortho$Acervicornis)

mcav_ortho <- mcav_ortho %>% mutate(Mcavernosa = strsplit(as.character(Mcavernosa), ",")) %>% 
    unnest(Mcavernosa)
mcav_ortho <- na.omit(mcav_ortho)
mcav_ortho$Mcavernosa <- gsub(" ", "", mcav_ortho$Mcavernosa)

ofav_ortho <- ofav_ortho %>% mutate(Ofaveolata = strsplit(as.character(Ofaveolata), ",")) %>% 
    unnest(Ofaveolata)
ofav_ortho <- na.omit(ofav_ortho)
ofav_ortho$Ofaveolata <- gsub(" ", "", ofav_ortho$Ofaveolata)

mcap_ortho <- mcap_ortho %>% mutate(Mcapitata = strsplit(as.character(Mcapitata), ",")) %>% 
    unnest(Mcapitata)
mcap_ortho <- na.omit(mcap_ortho)
mcap_ortho$Mcapitata <- gsub(" ", "", mcap_ortho$Mcapitata)

pacuta_ortho <- pacuta_ortho %>% mutate(Pacuta = strsplit(as.character(Pacuta), ",")) %>% 
    unnest(Pacuta)
pacuta_ortho <- na.omit(pacuta_ortho)
pacuta_ortho$Pacuta <- gsub(" ", "", pacuta_ortho$Pacuta)

plob_ortho <- plob_ortho %>% mutate(Plobata = strsplit(as.character(Plobata), ",")) %>% 
    unnest(Plobata)
plob_ortho <- na.omit(plob_ortho)
plob_ortho$Plobata <- gsub(" ", "", plob_ortho$Plobata)

```

Figure out how to make upset plots for these so that it does frequency counts of each orthogroup for each species and shows shared ones? 



Read DEGs in and merge w/ orthogroup by gene for each species 
### Acerv 
```{r}
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# name gene id col in acerv_ortho
colnames(acerv_ortho)[2] <- "gene_id"
acerv_ortho$gene_id <- gsub("model", "TU", acerv_ortho$gene_id)
acerv_ortho$gene_id <- gsub(" ", "", acerv_ortho$gene_id)

# Merge dfs by gene_id
acerv <- merge(acerv_ortho, acerv_DEG, by = "gene_id")
length(unique(acerv$gene_id)) # 179 DEGs found in orthogroups 
length(unique(acerv$Orthogroup)) # 159 orthogroups
write.csv(unique(acerv$Orthogroup), "Output/OrthoFinder/acerv_uniq_ortho_20221020.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(acerv, "Output/OrthoFinder/acerv_DEGs_ortho_20221020.csv")
```

### Mcav 
```{r}
mcav_DEG <- read.csv("Output/DESeq2/mcav/mcav_unique.sig.list_20210208.csv", header = T)[,1]
mcav_DEG <- as.data.frame(mcav_DEG)
colnames(mcav_DEG)[1] <- "gene_id"

# name gene id col in acerv_ortho
colnames(mcav_ortho)[2] <- "gene_id"
mcav_ortho$gene_id <- gsub("-RA", "", mcav_ortho$gene_id)
mcav_ortho$gene_id <- gsub(" ", "", mcav_ortho$gene_id)

# Merge dfs by gene_id
mcav <- merge(mcav_ortho, mcav_DEG, by = "gene_id")
length(unique(mcav$gene_id)) # 53 DEGs found in orthogroups 
length(unique(mcav$Orthogroup)) # 48 orthogroups 
write.csv(unique(mcav$Orthogroup), "Output/OrthoFinder/mcav_uniq_ortho_20221020.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(mcav, "Output/OrthoFinder/mcav_DEGs_ortho_20221020.csv")
```

### Ofav 
```{r}
ofav_DEG <- read.csv("Output/DESeq2/ofav/ofav_unique.sig.list_20220111.csv", header = T)[,1]
ofav_DEG <- as.data.frame(ofav_DEG)
colnames(ofav_DEG)[1] <- "gene_id"

# ofav is so annoying bc it has its prot seqs labeled with XP_xxx but when running STAR, the LOC term was used for gene id. Now I have to read in original gff file and merge original gff with ofav_ortho df to get LOC term

# read in gff 
ofav <- read.csv("~/Desktop/GFFs/ofav/GCF_002042975.1_ofav_dov_v1_genomic.gff", header=FALSE, sep="\t", skip=6) # read in gff from NCBI
colnames(ofav) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene") # name cols
ofav <- na.omit(ofav)
ofav <- subset(ofav, id == "CDS")
ofav$prot_id <- gsub(".*cds-","", ofav$gene) # remove everything before -cds
ofav$prot_id <- gsub(";.*","", ofav$prot_id) # remove everything after ;
ofav$gene_id <- gsub(".*LOC","", ofav$gene) # remove everything before LOC
ofav$gene_id <- gsub(";.*","", ofav$gene_id) # remove everything after ;
ofav$gene_id <- gsub("iso.*","", ofav$gene_id) # remove everything after iso
ofav$gene_id <- paste0("LOC", ofav$gene_id)
write.csv("~/Desktop/GFFs/ofav/ofav_gff_LOCTerm.gff") # writing gff file so that it always has LOC term already isolated
# select cols with XP- and LOC info
ofav_sub <- ofav[,c("prot_id", "gene_id")]
ofav_sub<- unique(ofav_sub)

# temporarily change names of ofav_ortho to match ofav_sub
colnames(ofav_ortho)[2] <- "prot_id"
ofav_ortho$prot_id <- gsub(" ", "", ofav_ortho$prot_id)

# Merge gff info / ortho results and remove prot_id
merge <- merge(ofav_ortho, ofav_sub, by = "prot_id")
merge <- select(merge, c(Orthogroup, gene_id))
merge <- unique(merge)

# Merge dfs by gene_id
ofav <- merge(merge, ofav_DEG, by = "gene_id")
length(unique(ofav$gene_id)) # 2 DEGs found in orthogroups 
length(unique(ofav$Orthogroup)) # 2 orthogroups 
write.csv(unique(ofav$Orthogroup), "Output/OrthoFinder/ofav_uniq_ortho_20221020.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(ofav, "Output/OrthoFinder/ofav_DEGs_ortho_20221020.csv")
```

### Mcap
```{r}
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# name gene id col in mcap_ortho
colnames(mcap_ortho)[2] <- "gene_id"
mcap_ortho$gene_id <- gsub(".t1", "", mcap_ortho$gene_id)
mcap_ortho$gene_id <- gsub(" ", "", mcap_ortho$gene_id)

# Merge dfs by gene_id
mcap <- merge(mcap_ortho, mcap_DEG, by = "gene_id")
length(unique(mcap$gene_id)) # 118 DEGs found in orthogroups 
length(unique(mcap$Orthogroup)) # 110 orthogroups 
write.csv(unique(mcap$Orthogroup), "Output/OrthoFinder/mcap_uniq_ortho_20221020.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(mcap, "Output/OrthoFinder/mcap_DEGs_ortho_20221020.csv")
```

### Pacuta 
```{r}
pacuta_DEG <- read.csv("Output/DESeq2/pacuta/pacuta_unique.sig.list_20220116.csv", header = T)[,1]
pacuta_DEG <- as.data.frame(pacuta_DEG)
colnames(pacuta_DEG)[1] <- "gene_id"

# name gene id col in acerv_ortho
colnames(pacuta_ortho)[2] <- "gene_id"
pacuta_ortho$gene_id <- gsub(" ", "", pacuta_ortho$gene_id)

# Merge dfs by gene_id
pacuta <- merge(pacuta_ortho, pacuta_DEG, by = "gene_id")
length(unique(pacuta$gene_id)) # 199 DEGs found in orthogroups 
length(unique(pacuta$Orthogroup)) # 159 orthogroups 
write.csv(unique(pacuta$Orthogroup), "Output/OrthoFinder/pacuta_uniq_ortho_20221020.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(pacuta, "Output/OrthoFinder/pacuta_DEGs_ortho_20221020.csv")
```

### Plob 
```{r}
plob_DEG <- read.csv("Output/DESeq2/plob/plob_unique.sig.list_20210326.csv", header = T)[,1]
plob_DEG <- as.data.frame(plob_DEG)
colnames(plob_DEG)[1] <- "gene_id"

# name gene id col in acerv_ortho
colnames(plob_ortho)[2] <- "gene_id"
plob_ortho$gene_id <- gsub(".m1", "", plob_ortho$gene_id)
plob_ortho$gene_id <- gsub("model", "TU", plob_ortho$gene_id)
plob_ortho$gene_id <- gsub(" ", "", plob_ortho$gene_id)

# Merge dfs by gene_id
plob <- merge(plob_ortho, plob_DEG, by = "gene_id")
length(unique(plob$gene_id)) # 122 DEGs found in orthogroups 
length(unique(plob$Orthogroup)) # 101 orthogroups 
write.csv(unique(plob$Orthogroup), "Output/OrthoFinder/plob_uniq_ortho_20221020.csv") ## In Excel, I take the unique ortho files, remove excess rows/cols so its just a list of orthogroups. 

# Write to csv 
write.csv(plob, "Output/OrthoFinder/plob_DEGs_ortho_20221020.csv") 
```

Plotted Venn Diagram using this website: https://bioinformatics.psb.ugent.be/webtools/Venn/. can't figure out how to do it in R

How to plot using the above website^^^^. It will not be the most beautiful thing but thats okay
- In excel, from the xxxx_uniq_ortho file, remove all columns except the one that has orthogroups. Delete the column name as well 
- Go to the website. It will have places to upload files and name the files. Upload each file and write the species name for the name of the file in the website. 
- Scroll down to OUTPUT control. Select Venn Shape as Non-symmetric and Fill as Colored
- Click submit. may take like 30 secs
- It will provide the list names, number of elements in each list, and number of unique elements
- Download the image result and text results
- The image will not look great. Since I'm comparing 6 species, it looks pretty rough. But it will show the # of orthogroups shared between lists
- The text result will give the text version of the image. It shows the lists that overlap and which specific element is a part of that overlap. Yay! 
- QC BELOW

20221020
Doing ortho stuff again. This time, I did not omit NAs from the data set when I read it in (lines 23-33). 
This is the QC step to make sure the website didn't mess anything up. Intersect 2 of the lists and compare the intersection results in R and the results from the website. Everything SHOULD match!
```{r}

## Acerv & Pacuta
## remove the species col
test_pa <- pacuta[,-c(1)]
test_ac <- acerv[,-c(1)]
intersect(test_ac, test_pa)
# [1] "OG0000414" "OG0011284" "OG0000989" "OG0000127" "OG0012176" "OG0000487" "OG0000469" "OG0003466" "OG0007132"

## Acerv & Plob
## remove the species col
test_pl <- plob[,-c(1)]
test_ac <- acerv[,-c(1)]
intersect(test_ac, test_pl)
#[1] "OG0000414" "OG0011284" "OG0000989" "OG0000127" "OG0012176" "OG0000487" "OG0000469" "OG0003466" "OG0007132"
#[1] "OG0000000" "OG0000414" "OG0000308" "OG0000221" "OG0000487" "OG0000469" "OG0000008" "OG0000491"

## Acerv & Ofav
## remove the species col
test_of <- ofav[,-c(1)]
test_ac <- acerv[,-c(1)]
intersect(test_ac, test_of)
#[1] "OG0000000"
```



And then make one df with all species info for orthogroups?
```{r}
acerv$Species <- "A.cervicornis"
mcav$Species <- "M.cavernosa"
ofav$Species <- "O.faveolata"
mcap$Species <- "M.capitata"
pacuta$Species <- "P.acuta"
plob$Species <- "P.lobata"

test <- rbind(acerv, mcav, ofav, mcap, pacuta, plob)
length(unique(test$Orthogroup))

# Write to csv 
write.csv(test, "Output/OrthoFinder/AllSpecies_DEGs_ortho_20221020.csv") 
```

20221021
I'm going to try to merge the DEGs & GO terms with the orthogroups. Let's try with acerv first 

### Acerv
```{r}
# using acerv 

# rename gene id in acerv so that there is 'model' instead of 'TU' in the name 
acerv$gene_id <- gsub("TU", "model", acerv$gene_id)

# read in DEG & GO data
acerv_go <- read.csv("Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
acerv_all <- merge(acerv_go, acerv, by = "gene_id", all = T)

acerv_all <- acerv_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(acerv_all$gene_id)) # 83 - not all genes had an orthogroup
length(unique(acerv_all$Orthogroup)) # 66 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(test, "Output/OrthoFinder/acerv_DEGs_GO_ortho_20221020.csv") 
```

Why not just merge the orthogroups with "acerv_sub_ByTreatment_GO.terms_20211104.csv" first instead of "acerv_sub_unique.sig.list_20210219.csv"?
- The GO terms df only has the gene ids for those genes that received GO annotations, so it left out many of the other DEGs in the dataset that were not annotated

Now do with the rest of species 

### Mcav 
```{r}
# read in DEG & GO data
mcav_go <- read.csv("Output/GOSeq/mcav/mcav_ByTreatment_GO.terms_20211109.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
mcav_all <- merge(mcav_go, mcav, by = "gene_id", all = T)

mcav_all <- mcav_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(mcav_all$gene_id)) # 35 - not all genes had an orthogroup
length(unique(mcav_all$Orthogroup)) # 28 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(mcav_all, "Output/OrthoFinder/mcav_DEGs_GO_ortho_20221020.csv")    
```

### Ofav 
```{r}
# read in DEG & GO data
ofav_go <- read.csv("Output/GOSeq/ofav/ofav_ByTreatment_GO.terms_20220111.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
ofav_all <- merge(ofav_go, ofav, by = "gene_id", all = T)

ofav_all <- ofav_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(ofav_all$gene_id)) # 4 - not all genes had an orthogroup
length(unique(ofav_all$Orthogroup)) # 2 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(ofav_all, "Output/OrthoFinder/ofav_DEGs_GO_ortho_20221020.csv") 
```

### Mcap
```{r}
# read in DEG & GO data
mcap_go <- read.csv("Output/GOSeq/mcap/mcap_ByTreatment_GO.terms_20220417.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
mcap_all <- merge(mcap_go, mcap, by = "gene_id", all = T)

mcap_all <- mcap_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(mcap_all$gene_id)) # 85 - not all genes had an orthogroup
length(unique(mcap_all$Orthogroup)) # 73 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(mcap_all, "Output/OrthoFinder/mcap_DEGs_GO_ortho_20221020.csv")    
```

### Pacuta 
```{r}
# read in DEG & GO data
pact_go <- read.csv("Output/GOSeq/pacuta/pacuta_ByTreatment_GO.terms_20220121.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
pact_all <- merge(pact_go, pacuta, by = "gene_id", all = T)

pact_all <- pact_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(pact_all$gene_id)) # 147 - not all genes had an orthogroup
length(unique(pact_all$Orthogroup)) # 100 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(pact_all, "Output/OrthoFinder/pacuta_DEGs_GO_ortho_20221020.csv")    
```

### Plob 
```{r}
# read in DEG & GO data
plob_go <- read.csv("Output/GOSeq/plob/plob_ByTreatment_GO.terms_20211106.csv", header = T)[,-1]

# Merge by gene id with orthogroup data 
plob_all <- merge(plob_go, plob, by = "gene_id", all = T)

plob_all <- plob_all %>% drop_na(category) # dropping NAs if they don't have a GO term associated 

# looking at #s of genes and orthogroups left 
length(unique(plob_all$gene_id)) # 83 - not all genes had an orthogroup
length(unique(plob_all$Orthogroup)) # 49 - not all orthogroups were assigned a gene 

##This file now has it all! 

# Write to csv
write.csv(plob_all, "Output/OrthoFinder/plob_DEGs_GO_ortho_20221020.csv")    
```


20221021
Still need to make changes to orthogroup info in most of the figures/tables/supp info

20221115

Trying to somehow merge the DEGs and the comparison files???

Acerv v. Mcav - for this section, I read in the orthogroups comparison file for Acerv and Mcav. Then I isolated the Acerv gene ids and orthogroups from this file. I read in the Acerv DEG file and merged it with the Acerv gene id/orthogroup dataframe. I did the same with the Mcav info. Then I merged the Acerv orthogroup/DEG info and Mcav orthogroup/DEG info into a single df to get the orthogroups that were shared between the DEGs of both Mcav and Acerv. Then I isolated the shared Acerv/DEG info into a separate df and renamed the gene id column to "gene_id" instead of gene_id.x. I read in the Acerv GO term file and I merged the GO term file with the shared Acerv orthogroup/DEG info to get the functional info for Acerv's shared orthogroup w/ Mcav. I repeated the above process for Mcav (this time subsetting Mcav columns, reading in Mcav data, etc). 
```{r}
# Read in comparison file for Acerv & Mcav
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Acerv_assembly_v1.0.protein__v__Mcavernosa.maker.proteins.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Acervicornis", "Mcavernosa")


############# Acerv #############

# Isolate Acerv genes and orthogroups
acerv_ortho <- compare[,1:2]
acerv_ortho <- acerv_ortho %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(acerv_ortho) <- c("Orthogroup", "gene_id")
acerv_ortho$gene_id <- gsub("model", "TU", acerv_ortho$gene_id)
acerv_ortho$gene_id <- gsub(" ", "", acerv_ortho$gene_id)
length(unique(acerv_ortho$Orthogroup)) # 11413 orthogroups
length(unique(acerv_ortho$gene_id)) # 15638 genes

# Read in Acerv DEGs
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# Merge Acerv shared orthogroup & DEG info
acerv_DEG_ortho <- merge(acerv_DEG, acerv_ortho, by = "gene_id")
length(unique(acerv_DEG_ortho$gene_id)) # 2 DEGs
length(unique(acerv_DEG_ortho$Orthogroups)) # 1 orthogroup
length(unique(acerv_DEG_ortho$category)) # 7 GO terms

# Read in GO term file for Acerv and merge GO terms w/ Acerv orthogroup and DEG info
acerv_DEG_ortho <- acerv_DEG_ortho[,c(1,2)] # separate orthogroups and gene id for acerv
colnames(acerv_DEG_ortho) <- c("Orthogroups", "gene_id") # renamed col names
acerv_DEG_ortho_GO <- read.csv("Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv", header = T)[,-1]
acerv_DEG_ortho_GO$gene_id <- gsub("model", "TU", acerv_DEG_ortho_GO$gene_id)

# Merge df w/ DEGs & GO terms with the orthogroup df
acerv_all <- merge(acerv_DEG_ortho_GO, acerv_ortho, by = "gene_id")


############# Mcav #############
# Isolate Mcav genes and orthogroups
mcav_ortho <- compare[,1:3]
mcav_ortho <- mcav_ortho %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(mcav_ortho) <- c("Orthogroup", "gene_id")
mcav_ortho$gene_id <- gsub("-RA", "", mcav_ortho$gene_id)
mcav_ortho$gene_id <- gsub(" ", "", mcav_ortho$gene_id)
length(unique(mcav_ortho$Orthogroup)) # xxxxx orthogroups
length(unique(mcav_ortho$gene_id)) # xxxx genes

# Read in Mcav DEGs
mcav_DEG <- read.csv("Output/DESeq2/mcav/mcav_unique.sig.list_20210208.csv", header = T)[,1]
mcav_DEG <- as.data.frame(mcav_DEG)
colnames(mcav_DEG)[1] <- "gene_id"

# Merge Mcav orthogroup & DEG info
mcav_DEG_ortho <- merge(mcav_DEG, mcav_ortho, by = "gene_id")
length(unique(mcav_DEG_ortho$gene_id)) # 2 DEGs
length(unique(mcav_DEG_ortho$Orthogroups)) # 1 orthogroup
length(unique(mcav_DEG_ortho$category)) # 7 GO terms

# Read in GO term file for Mcav and merge GO terms w/ Mcav orthogroup and DEG info
mcav_DEG_ortho <- mcav_DEG_ortho[,c(1,3)] # separate orthogroups and gene id for mcav
colnames(mcav_DEG_ortho) <- c("Orthogroups", "gene_id") # renamed col names
acerv_go <- read.csv("Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv", header = T)[,-1]
acerv_go$gene_id <- gsub("model", "TU", acerv_go$gene_id)























# Isolate Mcav genes and orthogroups
mcav_test <- compare[, c(1,3)]
mcav_test <- mcav_test %>% 
    mutate(Mcavernosa = strsplit(as.character(Mcavernosa), ",")) %>% 
    unnest(Mcavernosa)
colnames(mcav_test) <- c("Orthogroup", "gene_id")
mcav_test$gene_id <- gsub("-RA", "", mcav_test$gene_id)
mcav_test$gene_id <- gsub(" ", "", mcav_test$gene_id)

# Read in Mcav DEGs
mcav_DEG <- read.csv("Output/DESeq2/mcav/mcav_unique.sig.list_20210208.csv", header = T)[,1]
mcav_DEG <- as.data.frame(mcav_DEG)
colnames(mcav_DEG)[1] <- "gene_id"

# Merge Mcav orthogroup & DEG info
blah2 <- merge(mcav_DEG, mcav_test, by = "gene_id")

# Merge Acerv orthogroup/DEG info and Mcav orthogroup/DEG info
all <- merge(blah, blah2, by = "Orthogroup")
length(unique(all$Orthogroup)) # 2 orthogroups shared between Acerv & Mcav
length(unique(all$gene_id.x)) # 2 DEGs from Acerv in orthogroup comparison between Acerv & Mcav
length(unique(all$gene_id.y)) # 4 DEGs from Mcav in orthogroup comparison between Acerv & Mcav


# merge GO terms 
## Acerv
acerv_all <- all[,c(1,2)] # separate orthogroups and gene id for acerv
colnames(acerv_all) <- c("Orthogroups", "gene_id") # renamed col names
acerv_go <- read.csv("Output/GOSeq/acerv/acerv_sub_ByTreatment_GO.terms_20211104.csv", header = T)[,-1]
acerv_go$gene_id <- gsub("model", "TU", acerv_go$gene_id)

blah2 <- merge(acerv_all, acerv_go, by = "gene_id")



mcav_all <- all[,c(1,3)] # separate orthogroups and gene id for mcav
colnames(mcav_all) <- c("Orthogroups", "gene_id") # renamed col names
mcav_go <- read.csv("Output/GOSeq/mcav/mcav_ByTreatment_GO.terms_20211109.csv", header = T)[,-1] # read in GO terms 
blah <- merge(mcav_all, mcav_go, by = "gene_id")



## Mcav
mcav_all <- all[,c(1,3)] # separate orthogroups and gene id for mcav
colnames(mcav_all) <- c("Orthogroups", "gene_id") # renamed col names
mcav_go <- read.csv("Output/GOSeq/mcav/mcav_ByTreatment_GO.terms_20211109.csv", header = T)[,-1] # read in GO terms 
blah <- merge(mcav_all, mcav_go, by = "gene_id")




```

Acerv v. Mcap
```{r}
# Read in comparison file for Acerv & Mcap
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Acerv_assembly_v1.0.protein__v__Montipora_capitata_HIv2.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Acervicornis", "Mcapitata")

# Isolate Acerv genes and orthogroups
acerv_test <- compare[,1:2]
acerv_test <- acerv_test %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(acerv_test) <- c("Orthogroup", "gene_id")
acerv_test$gene_id <- gsub("model", "TU", acerv_test$gene_id)
acerv_test$gene_id <- gsub(" ", "", acerv_test$gene_id)

# Read in Acerv DEGs
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# Merge Acerv orthogroup & DEG info
acerv_merge <- merge(acerv_DEG, acerv_test, by = "gene_id")


# Isolate Mcap genes and orthogroups
mcap_test <- compare[, c(1,3)]
mcap_test <- mcap_test %>% 
    mutate(Mcapitata = strsplit(as.character(Mcapitata), ",")) %>% 
    unnest(Mcapitata)
colnames(mcap_test) <- c("Orthogroup", "gene_id")
mcap_test$gene_id <- gsub(".t1", "", mcap_test$gene_id)
mcap_test$gene_id <- gsub(" ", "", mcap_test$gene_id)

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_merge <- merge(mcap_DEG, mcap_test, by = "gene_id")

# Merge Acerv orthogroup/DEG info and Mcap orthogroup/DEG info
all <- merge(acerv_merge, mcap_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 8 orthogroups shared between Acerv & Mcap
length(unique(all$gene_id.x)) # 10 DEGs from Acerv in orthogroup comparison between Acerv & Mcap
length(unique(all$gene_id.y)) # 10 DEGs from Mcap in orthogroup comparison between Acerv & Mcap

# save as csv? ^^ 
```

Acerv v. Ofav
DO THIS!!!!!!!!!!!!!!!!!!!!!!!!******************

Acerv v. Pacuta
```{r}
# Read in comparison file for Acerv & Pacuta
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Acerv_assembly_v1.0.protein__v__Pocillopora_acuta_HIv1.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Acervicornis", "Pacuta")

# Isolate Acerv genes and orthogroups
acerv_test <- compare[,1:2]
acerv_test <- acerv_test %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(acerv_test) <- c("Orthogroup", "gene_id")
acerv_test$gene_id <- gsub("model", "TU", acerv_test$gene_id)
acerv_test$gene_id <- gsub(" ", "", acerv_test$gene_id)

# Read in Acerv DEGs
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# Merge Acerv orthogroup & DEG info
acerv_merge <- merge(acerv_DEG, acerv_test, by = "gene_id")


# Isolate Pacuta genes and orthogroups
pacuta_test <- compare[, c(1,3)]
pacuta_test <- pacuta_test %>% 
    mutate(Pacuta = strsplit(as.character(Pacuta), ",")) %>% 
    unnest(Pacuta)
colnames(pacuta_test) <- c("Orthogroup", "gene_id")
pacuta_test$gene_id <- gsub(" ", "", pacuta_test$gene_id)

# Read in Pacuta DEGs
pacuta_DEG <- read.csv("Output/DESeq2/pacuta/pacuta_unique.sig.list_20220116.csv", header = T)[,1]
pacuta_DEG <- as.data.frame(pacuta_DEG)
colnames(pacuta_DEG)[1] <- "gene_id"

# Merge Pacuta orthogroup & DEG info
pacuta_merge <- merge(pacuta_DEG, pacuta_test, by = "gene_id")

# Merge Acerv orthogroup/DEG info and Pacuta orthogroup/DEG info
all <- merge(acerv_merge, pacuta, by = "Orthogroup")
length(unique(all$Orthogroup)) # 9 orthogroups shared between Acerv & Pacuta
length(unique(all$gene_id.x)) # 11 DEGs from Acerv in orthogroup comparison between Acerv & Pacuta
length(unique(all$gene_id.y)) # 14 DEGs from Pacuta in orthogroup comparison between Acerv & Pacuta

# save as csv? ^^ 
```

Acerv v. Plob
```{r}
# Read in comparison file for Acerv & Plob
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Acerv_assembly_v1.0.protein__v__plut2v1.1.proteins.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Acervicornis", "Plob")

# Isolate Acerv genes and orthogroups
acerv_test <- compare[,1:2]
acerv_test <- acerv_test %>% 
    mutate(Acervicornis = strsplit(as.character(Acervicornis), ",")) %>% 
    unnest(Acervicornis)
colnames(acerv_test) <- c("Orthogroup", "gene_id")
acerv_test$gene_id <- gsub("model", "TU", acerv_test$gene_id)
acerv_test$gene_id <- gsub(" ", "", acerv_test$gene_id)

# Read in Acerv DEGs
acerv_DEG <- read.csv("Output/DESeq2/acerv/acerv_sub_unique.sig.list_20210219.csv", header = T)[,1]
acerv_DEG <- as.data.frame(acerv_DEG)
colnames(acerv_DEG) <- "gene_id"

# Merge Acerv orthogroup & DEG info
acerv_merge <- merge(acerv_DEG, acerv_test, by = "gene_id")


# Isolate Plob genes and orthogroups
plob_test <- compare[, c(1,3)]
plob_test <- plob_test %>% 
    mutate(Plob = strsplit(as.character(Plob), ",")) %>% 
    unnest(Plob)
colnames(plob_test) <- c("Orthogroup", "gene_id")
plob_test$gene_id <- gsub(".m1", "", plob_test$gene_id)
plob_test$gene_id <- gsub(" ", "", plob_test$gene_id)

# Read in Plob DEGs
plob_DEG <- read.csv("Output/DESeq2/plob/plob_unique.sig.list_20210326.csv", header = T)[,1]
plob_DEG <- as.data.frame(plob_DEG)
colnames(plob_DEG)[1] <- "gene_id"

# Merge Plob orthogroup & DEG info
plob_merge <- merge(plob_DEG, plob_test, by = "gene_id")

# Merge Acerv orthogroup/DEG info and Plob orthogroup/DEG info
all <- merge(acerv_merge, plob_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 6 orthogroups shared between Acerv & Plob
length(unique(all$gene_id.x)) # 14 DEGs from Acerv in orthogroup comparison between Acerv & Plob
length(unique(all$gene_id.y)) # 7 DEGs from Pacuta in orthogroup comparison between Acerv & Plob

# save as csv? ^^ 
```

Mcav v. Mcap
```{r}
# Read in comparison file for Mcav & Mcap
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Mcavernosa.maker.proteins__v__Montipora_capitata_HIv2.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Mcavernosa", "Mcapitata")

# Isolate Mcav genes and orthogroups
mcav_test <- compare[, c(1,2)]
mcav_test <- mcav_test %>% 
    mutate(Mcavernosa = strsplit(as.character(Mcavernosa), ",")) %>% 
    unnest(Mcavernosa)
colnames(mcav_test) <- c("Orthogroup", "gene_id")
mcav_test$gene_id <- gsub("-RA", "", mcav_test$gene_id)
mcav_test$gene_id <- gsub(" ", "", mcav_test$gene_id)

# Read in Mcav DEGs
mcav_DEG <- read.csv("Output/DESeq2/mcav/mcav_unique.sig.list_20210208.csv", header = T)[,1]
mcav_DEG <- as.data.frame(mcav_DEG)
colnames(mcav_DEG)[1] <- "gene_id"

# Merge Mcav orthogroup & DEG info
mcav_merge <- merge(mcav_DEG, mcav_test, by = "gene_id")



# Isolate Mcap genes and orthogroups
mcap_test <- compare[,1:3]
mcap_test <- mcap_test %>% 
    mutate(Mcapitata = strsplit(as.character(Mcapitata), ",")) %>% 
    unnest(Mcapitata)
colnames(mcap_test) <- c("Orthogroup", "gene_id")
mcap_test$gene_id <- gsub(".t1", "", mcap_test$gene_id)
mcap_test$gene_id <- gsub(" ", "", mcap_test$gene_id)

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_merge <- merge(mcap_DEG, mcap_test, by = "gene_id")

# Merge Mcav orthogroup/DEG info and Mcap orthogroup/DEG info
all <- merge(mcav_merge, mcap_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 0 orthogroups shared between Mcav & Mcap
length(unique(all$gene_id.x)) # 0 DEGs from Acerv in orthogroup comparison between Mcav & Mcap
length(unique(all$gene_id.y)) # 0 DEGs from Pacuta in orthogroup comparison between Mcav & Mcap

# save as csv? ^^ 
```

Mcav v. Ofav
DO THIS!!!!!!!!!!!!!!!!!!!!!!!!******************

Mcav v. Pacuta
```{r}
# Read in comparison file for Mcav & Pacuta















compare <- read.table(file = 'Output/OrthoFinder/species_compare/Mcavernosa.maker.proteins__v__Montipora_capitata_HIv2.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Mcavernosa", "Mcapitata")

# Isolate Mcav genes and orthogroups
mcav_test <- compare[, c(1,2)]
mcav_test <- mcav_test %>% 
    mutate(Mcavernosa = strsplit(as.character(Mcavernosa), ",")) %>% 
    unnest(Mcavernosa)
colnames(mcav_test) <- c("Orthogroup", "gene_id")
mcav_test$gene_id <- gsub("-RA", "", mcav_test$gene_id)
mcav_test$gene_id <- gsub(" ", "", mcav_test$gene_id)

# Read in Mcav DEGs
mcav_DEG <- read.csv("Output/DESeq2/mcav/mcav_unique.sig.list_20210208.csv", header = T)[,1]
mcav_DEG <- as.data.frame(mcav_DEG)
colnames(mcav_DEG)[1] <- "gene_id"

# Merge Mcav orthogroup & DEG info
mcav_merge <- merge(mcav_DEG, mcav_test, by = "gene_id")



# Isolate Mcap genes and orthogroups
mcap_test <- compare[,1:3]
mcap_test <- mcap_test %>% 
    mutate(Mcapitata = strsplit(as.character(Mcapitata), ",")) %>% 
    unnest(Mcapitata)
colnames(mcap_test) <- c("Orthogroup", "gene_id")
mcap_test$gene_id <- gsub(".t1", "", mcap_test$gene_id)
mcap_test$gene_id <- gsub(" ", "", mcap_test$gene_id)

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_merge <- merge(mcap_DEG, mcap_test, by = "gene_id")

# Merge Mcav orthogroup/DEG info and Mcap orthogroup/DEG info
all <- merge(mcav_merge, mcap_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 0 orthogroups shared between Mcav & Mcap
length(unique(all$gene_id.x)) # 0 DEGs from Acerv in orthogroup comparison between Mcav & Mcap
length(unique(all$gene_id.y)) # 0 DEGs from Pacuta in orthogroup comparison between Mcav & Mcap

# save as csv? ^^ 
```

Mcav v. Plob

Ofav v. Mcap
DO THIS!!!!!!!!!!!!!!!!!!!!!!!!******************

Ofav v. Pacuta
DO THIS!!!!!!!!!!!!!!!!!!!!!!!!******************

Ofav v. Plob
DO THIS!!!!!!!!!!!!!!!!!!!!!!!!******************

Mcap v. Pacuta 
```{r}
# Read in comparison file for Mcap & Pacuta
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Montipora_capitata_HIv2.genes.pep__v__Pocillopora_acuta_HIv1.genes.pep.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Mcap", "Pacuta")

# Isolate Mcap genes and orthogroups
mcap_test <- compare[,1:2]
mcap_test <- mcap_test %>% 
    mutate(Mcap = strsplit(as.character(Mcap), ",")) %>% 
    unnest(Mcap)
colnames(mcap_test) <- c("Orthogroup", "gene_id")
mcap_test$gene_id <- gsub(".t1", "", mcap_test$gene_id)
mcap_test$gene_id <- gsub(" ", "", mcap_test$gene_id)

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_merge <- merge(mcap_DEG, mcap_test, by = "gene_id")


# Isolate Pacuta genes and orthogroups
pacuta_test <- compare[, c(1,3)]
pacuta_test <- pacuta_test %>% 
    mutate(Pacuta = strsplit(as.character(Pacuta), ",")) %>% 
    unnest(Pacuta)
colnames(pacuta_test) <- c("Orthogroup", "gene_id")
pacuta_test$gene_id <- gsub(" ", "", pacuta_test$gene_id)

# Read in Pacuta DEGs
pacuta_DEG <- read.csv("Output/DESeq2/pacuta/pacuta_unique.sig.list_20220116.csv", header = T)[,1]
pacuta_DEG <- as.data.frame(pacuta_DEG)
colnames(pacuta_DEG)[1] <- "gene_id"

# Merge Pacuta orthogroup & DEG info
pacuta_merge <- merge(pacuta_DEG, pacuta_test, by = "gene_id")

# Merge Mcap orthogroup/DEG info and Pacuta orthogroup/DEG info
all <- merge(mcap_merge, pacuta_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 7 orthogroups shared between Mcap & Pacuta
length(unique(all$gene_id.x)) # 8 DEGs from Pacuta in orthogroup comparison between Mcap & Pacuta
length(unique(all$gene_id.y)) # 9 DEGs from Plob in orthogroup comparison between Mcap & Pacuta

# save as csv? ^^ 
```

Mcap v. Plob 
```{r}
# Read in comparison file for Mcap & Plob
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Montipora_capitata_HIv2.genes.pep__v__plut2v1.1.proteins.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Mcap", "Plob")

# Isolate Mcap genes and orthogroups
mcap_test <- compare[,1:2]
mcap_test <- mcap_test %>% 
    mutate(Mcap = strsplit(as.character(Mcap), ",")) %>% 
    unnest(Mcap)
colnames(mcap_test) <- c("Orthogroup", "gene_id")
mcap_test$gene_id <- gsub(".t1", "", mcap_test$gene_id)
mcap_test$gene_id <- gsub(" ", "", mcap_test$gene_id)

# Read in Mcap DEGs
mcap_DEG <- read.csv("Output/DESeq2/mcap/mcap_unique.sig.list_20220315.csv", header = T)[,1]
mcap_DEG <- as.data.frame(mcap_DEG)
colnames(mcap_DEG)[1] <- "gene_id"

# Merge Mcap orthogroup & DEG info
mcap_merge <- merge(mcap_DEG, mcap_test, by = "gene_id")


# Isolate Plob genes and orthogroups
plob_test <- compare[, c(1,3)]
plob_test <- plob_test %>% 
    mutate(Plob = strsplit(as.character(Plob), ",")) %>% 
    unnest(Plob)
colnames(plob_test) <- c("Orthogroup", "gene_id")
plob_test$gene_id <- gsub(".m1", "", plob_test$gene_id)
plob_test$gene_id <- gsub(" ", "", plob_test$gene_id)

# Read in Plob DEGs
plob_DEG <- read.csv("Output/DESeq2/plob/plob_unique.sig.list_20210326.csv", header = T)[,1]
plob_DEG <- as.data.frame(plob_DEG)
colnames(plob_DEG)[1] <- "gene_id"

# Merge Plob orthogroup & DEG info
plob_merge <- merge(plob_DEG, plob_test, by = "gene_id")

# Merge Mcap orthogroup/DEG info and Plob orthogroup/DEG info
all <- merge(mcap_merge, plob_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 8 orthogroups shared between Pacuta & Plob
length(unique(all$gene_id.x)) # 10 DEGs from Pacuta in orthogroup comparison between Pacuta & Plob
length(unique(all$gene_id.y)) # 10 DEGs from Plob in orthogroup comparison between Pacuta & Plob

# save as csv? ^^ 
```

Pacuta v. Plob 
```{r}
# Read in comparison file for Pacuta & Plob
compare <- read.table(file = 'Output/OrthoFinder/species_compare/Pocillopora_acuta_HIv1.genes.pep__v__plut2v1.1.proteins.tsv', sep = '\t', header = TRUE, na.strings=c(""," ","NA"))
colnames(compare) <- c("Orthogroup", "Pacuta", "Plob")

# Isolate Pacuta genes and orthogroups
pacuta_test <- compare[,1:2]
pacuta_test <- pacuta_test %>% 
    mutate(Pacuta = strsplit(as.character(Pacuta), ",")) %>% 
    unnest(Pacuta)
colnames(pacuta_test) <- c("Orthogroup", "gene_id")
pacuta_test$gene_id <- gsub(" ", "", pacuta_test$gene_id)

# Read in Pacuta DEGs
pacuta_DEG <- read.csv("Output/DESeq2/pacuta/pacuta_unique.sig.list_20220116.csv", header = T)[,1]
pacuta_DEG <- as.data.frame(pacuta_DEG)
colnames(pacuta_DEG)[1] <- "gene_id"

# Merge Pacuta orthogroup & DEG info
pacuta_merge <- merge(pacuta_DEG, pacuta_test, by = "gene_id")


# Isolate Plob genes and orthogroups
plob_test <- compare[, c(1,3)]
plob_test <- plob_test %>% 
    mutate(Plob = strsplit(as.character(Plob), ",")) %>% 
    unnest(Plob)
colnames(plob_test) <- c("Orthogroup", "gene_id")
plob_test$gene_id <- gsub(".m1", "", plob_test$gene_id)
plob_test$gene_id <- gsub(" ", "", plob_test$gene_id)

# Read in Plob DEGs
plob_DEG <- read.csv("Output/DESeq2/plob/plob_unique.sig.list_20210326.csv", header = T)[,1]
plob_DEG <- as.data.frame(plob_DEG)
colnames(plob_DEG)[1] <- "gene_id"

# Merge Plob orthogroup & DEG info
plob_merge <- merge(plob_DEG, plob_test, by = "gene_id")

# Merge Pacuta orthogroup/DEG info and Plob orthogroup/DEG info
all <- merge(pacuta_merge, plob_merge, by = "Orthogroup")
length(unique(all$Orthogroup)) # 6 orthogroups shared between Pacuta & Plob
length(unique(all$gene_id.x)) # 8 DEGs from Pacuta in orthogroup comparison between Pacuta & Plob
length(unique(all$gene_id.y)) # 6 DEGs from Plob in orthogroup comparison between Pacuta & Plob

# save as csv? ^^ 
```



