---
title: "mcav_DESeq2"
author: "jillashey"
date: "5/9/2022"
output: html_document
---

#### Code for Francois sedimentation data. Mcav only samples analyzed here aligned against Mcav STAR was read aligner and Mcav genomic info can be found [here](https://www.dropbox.com/s/yfqefzntt896xfz/Mcavernosa_genome.tgz) and/or [here](https://matzlab.weebly.com/data--code.html)

## Load packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("ggrepel")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("gridExtra")
library("clusterProfiler")
library("DataCombine")
library("VennDiagram")
```



## Load gene count matrix
```{r}
mcav_counts <- read.csv("Output/DESeq2/mcav/gene_count_mcav_only_matrix.csv", header = TRUE, row.names = "gene_id")
dim(mcav_counts) # 25142 x 15
head(mcav_counts)
for ( col in 1:ncol(mcav_counts)){
  colnames(mcav_counts)[col] <-  sub(".fastq.trim.fq.Aligned.sortedByCoord.out.bam.merge.gtf", "", colnames(mcav_counts)[col])
}
for ( col in 1:ncol(mcav_counts)){
  colnames(mcav_counts)[col] <-  gsub("X", "", colnames(mcav_counts)[col])
}
```

## Load metadata file 
```{r}
metadata <- read.csv("Data/FL_sediment_metadata.csv", header = TRUE)
dim(metadata) # 45 by 15
head(metadata)
# Selecting only the columns I need for analyses 
metadata <- select(metadata, c(Rep, Species, Treatment.in.mg.L.of.sediment, Location, File.Name.fastq))
# Renaming cols
colnames(metadata) <-c("Replicate","Species", "Treatment", "Location", "SampleID")
# Select Mcav species only 
mcav_metadata <- subset(metadata, Species=="Montastraea cavernosa")

# Renaming treatments
mcav_metadata$Treatment <- gsub("Ctl", "control", mcav_metadata$Treatment)
mcav_metadata$Treatment <- gsub("T1", "Treatment1", mcav_metadata$Treatment)
mcav_metadata$Treatment <- gsub("T2", "Treatment2", mcav_metadata$Treatment)
mcav_metadata$Treatment <- gsub("T3", "Treatment3", mcav_metadata$Treatment)
mcav_metadata$Treatment <- gsub("T4", "Treatment4", mcav_metadata$Treatment)
# Removing unwanted text from SampleID
mcav_metadata$SampleID <- gsub(".txt.gz", "", mcav_metadata$SampleID)
mcav_metadata$SampleID <- gsub(";.*", "", mcav_metadata$SampleID)
mcav_metadata$SampleID <- gsub(".fastq.gz", "", mcav_metadata$SampleID)
mcav_metadata$SampleID <- sub("\\.", "", mcav_metadata$SampleID)
# Make row names to be SampleID 
rownames(mcav_metadata) <- mcav_metadata$SampleID
```

## Subset count data for only mcav samples based on SampleID and make sure rows of metadata = cols of count data
```{r}
mcav_ID <- mcav_metadata$SampleID
count_mcav <- select(mcav_counts, all_of(mcav_ID))
all(rownames(mcav_metadata) %in% colnames(count_mcav)) # must come out TRUE
```

## Filter reads by proportion of samples containing cutoff value
```{r}
filt <- filterfun(pOverA(0.85, 5)) # set filter values for P over A; I used 0.85 and 5
tfil <- genefilter(count_mcav, filt) # create filter for counts data 
keep <- count_mcav[tfil,] # identify genes to keep based on filter
gn.keep <- rownames(keep)
mcav_counts_filt <- as.matrix(count_mcav[which(rownames(count_mcav) %in% gn.keep),]) 
#write.csv(mcav_counts_filt, "~/Desktop/mcav_counts_filtered.csv")
storage.mode(mcav_counts_filt) <- "integer" # stores count data as integer 
# Checking to make sure rownames in metadata == colnames in counts data 
all(rownames(mcav_metadata) %in% colnames(mcav_counts_filt)) # must come out TRUE
```

## Set Treatment as a factor
```{r}
mcav_metadata$Treatment <- factor(mcav_metadata$Treatment, levels = c("control", "Treatment1", "Treatment2", "Treatment3", "Treatment4"))
data <- DESeqDataSetFromMatrix(countData = mcav_counts_filt, colData = mcav_metadata, design = ~ Treatment)
```

## Expression visualization
First we are going to log-transform the data using a variance stabilizing transforamtion (vst). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects. To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4. Otherwise, there could be artefacts in our results.
```{r}
# Estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than 4 to use vst
SF.data <- estimateSizeFactors(data) 
SF.data
print(sizeFactors(SF.data)) #view size factors
# size factors all less than 4, can use VST

# apply a variance stabilizing transforamtion to minimize effects of small counts and normalize wrt library size
vst <- vst(data, blind = FALSE) 
head(assay(vst), 3) # view data

# Using vst object, plot heat-map of sample-to-sample distances
sampleDists <- dist(t(assay(vst))) # calculate distance matrix
sampleDistMatrix <- as.matrix(sampleDists) # create distance matrix
rownames(sampleDistMatrix) <- colnames(vst) # assign row names
colnames(sampleDistMatrix) <- NULL # assign col names 
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255) # assign colors 
mcav_heatmap <- pheatmap(sampleDistMatrix, # plot matrix
         clustering_distance_rows = sampleDists, # cluster rows
         clustering_distance_cols = sampleDists, # cluster cols
         col=colors) # set colors
mcav_heatmap

# Using vst object, make PCA plot of samples 
vst_PCAdata <- plotPCA(vst, intgroup = c("Treatment"), returnData=TRUE) # create PCA loadings 
percentVar <- round(100*attr(vst_PCAdata, "percentVar")) #plot PCA of samples with all data
mcav_PCAplot <- ggplot(vst_PCAdata, aes(PC1, PC2, color=Treatment)) + 
   geom_point(size=3) +
   geom_text(aes(label=name),hjust=0, vjust=0) +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  scale_color_manual(values = c(control="gray", Treatment1="darkslategray2", Treatment2="darkslategray3", Treatment3="darkslategray4", Treatment4="darkslategray")) +
   coord_fixed() +
   ggtitle("M. cavernosa - all genes") +
   theme_bw() + #Set background color
   theme(panel.border = element_blank(), # Set border
         #panel.grid.major = element_blank(), #Set major gridlines
         #panel.grid.minor = element_blank(), #Set minor gridlines
         axis.line = element_line(colour = "black"), #Set axes color
         plot.background=element_blank()) #Set the plot background
mcav_PCAplot 

# save plots 
mcav_heatmap_PCA <- grid.arrange(mcav_PCAplot, mcav_heatmap[[4]], nrow=1, ncol = 2, clip="off")
ggsave("Output/Figs/mcav/mcav_heatmap_PCA_AllGenes_20220520.jpeg", mcav_heatmap_PCA, width = 20, height = 25, units = "cm")
ggsave("Output/Figs/mcav/mcav_heatmap_PCA_AllGenes_20220520.pdf", mcav_heatmap_PCA, width = 20, height = 25, units = "cm")
```

## Run DESeq2
```{r}
# Differential gene expression analysis 
DEG.int <- DESeq(data) # run differential expression test by treatment (?) using wald test 
# estimating size factors
# estimating dispersions
# gene-wise dispersion estimates
# mean-dispersion relationship
# final dispersion estimates
# fitting model and testing
DEG.int.res <- results(DEG.int) # save DE results ; why does it say 'Wald test p-value: Treatment Treatment4 vs control' for DEG.int.res? Is it only looking at treatment 4 and control? In DESeq object created above, it says that design is Treatment
resultsNames(DEG.int) # view DE results 
#[1] "Intercept"                       "Treatment_Treatment1_vs_control" "Treatment_Treatment2_vs_control"
#[4] "Treatment_Treatment3_vs_control" "Treatment_Treatment4_vs_control"
# NAs in padj: https://hbctraining.github.io/DGE_workshop/lessons/05_DGE_DESeq2_analysis2.html 
# http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#why-are-some-p-values-set-to-na
```

## Differential expression comparisons by treatment 
### Compare control and T1
```{r}
DEG_control_vs_T1 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment1")) # results of DESeq2 comparing C and T1
DEG_control_vs_T1 
DEG_control_vs_T1 <- as.data.frame(DEG_control_vs_T1) # make results into a df
DEG_control_vs_T1["Treatment_Compare"] <- "CvsT1" # adding treatment comparison column
#write.csv(DEG_control_vs_T1, file = "~/Desktop/mcav_control_vs_T1_all_genes.csv") # maybe include gene counts too?
DEG_control_vs_T1.sig.num <- sum(DEG_control_vs_T1$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_control_vs_T1.sig.num
# 19 DEGs
DEG_control_vs_T1.sig <- subset(DEG_control_vs_T1, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T1.sig["Treatment_Compare"] <- "CvsT1" # adding treatment comparison column
DEG_control_vs_T1.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T1.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T1.sig.list <- as.data.frame(counts(DEG_control_vs_T1.sig.list)) # make list of sig gene counts into a df
DEG_control_vs_T1.sig.list_full <- cbind(DEG_control_vs_T1.sig, DEG_control_vs_T1.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T1.sig.list_full, file = "~/Desktop/mcav_control_vs_T1_DEG_full.csv") # write out csv
DEG_control_vs_T1.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T1.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
# function: y = a/x + b, and a local regression fit was automatically substituted.
# specify fitType='local' or 'mean' to avoid this message next time.
```

### Compare control and T2
```{r}
DEG_control_vs_T2 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment2")) # results of DESeq2 comparing C and T1
DEG_control_vs_T2 
DEG_control_vs_T2 <- as.data.frame(DEG_control_vs_T2) # make results into a df
DEG_control_vs_T2["Treatment_Compare"] <- "CvsT2" # adding treatment comparison column
#write.csv(DEG_control_vs_T2, file = "~/Desktop/mcav_control_vs_T2_all_genes.csv") # maybe include gene counts too?
DEG_control_vs_T2.sig.num <- sum(DEG_control_vs_T2$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_control_vs_T2.sig.num
# 43 DEGs
DEG_control_vs_T2.sig <- subset(DEG_control_vs_T2, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T2.sig["Treatment_Compare"] <- "CvsT2" # adding treatment comparison column
DEG_control_vs_T2.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T2.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T2.sig.list <- as.data.frame(counts(DEG_control_vs_T2.sig.list)) # make list of sig gene counts into a df
DEG_control_vs_T2.sig.list_full <- cbind(DEG_control_vs_T2.sig, DEG_control_vs_T2.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T2.sig.list_full, file = "~/Desktop/mcav_control_vs_T2_DEG_full.csv") # write out csv
DEG_control_vs_T2.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T2.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
# function: y = a/x + b, and a local regression fit was automatically substituted.
# specify fitType='local' or 'mean' to avoid this message next time.
```

### Compare control and T3
```{r}
DEG_control_vs_T3 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment3")) # results of DESeq2 comparing C and T1
DEG_control_vs_T3
DEG_control_vs_T3 <- as.data.frame(DEG_control_vs_T3) # make results into a df
DEG_control_vs_T3["Treatment_Compare"] <- "CvsT3" # adding treatment comparison column
#write.csv(DEG_control_vs_T3, file = "~/Desktop/mcav_control_vs_T3_all_genes.csv") # maybe include gene counts too?
DEG_control_vs_T3.sig.num <- sum(DEG_control_vs_T3$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_control_vs_T3.sig.num
# 26 DEGs
DEG_control_vs_T3.sig <- subset(DEG_control_vs_T3, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T3.sig["Treatment_Compare"] <- "CvsT3" # adding treatment comparison column
DEG_control_vs_T3.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T3.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T3.sig.list <- as.data.frame(counts(DEG_control_vs_T3.sig.list)) # make list of sig gene counts into a df
DEG_control_vs_T3.sig.list_full <- cbind(DEG_control_vs_T3.sig, DEG_control_vs_T3.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T3.sig.list_full, file = "~/Desktop/mcav_control_vs_T3_DEG_full.csv") # write out csv
DEG_control_vs_T3.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T3.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
```

### Compare control and T4
```{r}
DEG_control_vs_T4 <- results(DEG.int, contrast = c("Treatment", "control", "Treatment4")) # results of DESeq2 comparing C and T1
DEG_control_vs_T4
DEG_control_vs_T4 <- as.data.frame(DEG_control_vs_T4) # make results into a df
DEG_control_vs_T4["Treatment_Compare"] <- "CvsT4" # adding treatment comparison column
#write.csv(DEG_control_vs_T4, file = "~/Desktop/mcav_control_vs_T4_all_genes.csv") # maybe include gene counts too?
DEG_control_vs_T4.sig.num <- sum(DEG_control_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_control_vs_T4.sig.num
# 18 DEGs
DEG_control_vs_T4.sig <- subset(DEG_control_vs_T4, padj <0.05) # identify and subset significant pvalues
DEG_control_vs_T4.sig["Treatment_Compare"] <- "CvsT4" # adding treatment comparison column
DEG_control_vs_T4.sig.list <- data[which(rownames(data) %in% rownames(DEG_control_vs_T4.sig)),] # subset list of significant genes from original count data 
DEG_control_vs_T4.sig.list <- as.data.frame(counts(DEG_control_vs_T4.sig.list)) # make list of sig gene counts into a df
DEG_control_vs_T4.sig.list_full <- cbind(DEG_control_vs_T4.sig, DEG_control_vs_T4.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_control_vs_T4.sig.list_full, file = "~/Desktop/mcav_control_vs_T4_DEG_full.csv") # write out csv
DEG_control_vs_T4.vst.sig <- varianceStabilizingTransformation(DEG_control_vs_T4.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
```

### Compare T1 and T2
```{r}
DEG_T1_vs_T2 <- results(DEG.int, contrast = c("Treatment", "Treatment1", "Treatment2")) # results of DESeq2 comparing C and T1
DEG_T1_vs_T2
DEG_T1_vs_T2 <- as.data.frame(DEG_T1_vs_T2) # make results into a df
DEG_T1_vs_T2["Treatment_Compare"] <- "T1vsT2" # adding treatment comparison column
#write.csv(DEG_T1_vs_T2, file = "~/Desktop/mcav_T1_vs_T2_all_genes.csv") # maybe include gene counts too?
DEG_T1_vs_T2.sig.num <- sum(DEG_T1_vs_T2$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T1_vs_T2.sig.num
# 0 DEGs
```

### Compare T1 and T3
```{r}
DEG_T1_vs_T3 <- results(DEG.int, contrast = c("Treatment", "Treatment1", "Treatment3")) # results of DESeq2 comparing C and T1
DEG_T1_vs_T3
DEG_T1_vs_T3 <- as.data.frame(DEG_T1_vs_T3) # make results into a df
DEG_T1_vs_T3["Treatment_Compare"] <- "T1vsT3" # adding treatment comparison column
#write.csv(DEG_T1_vs_T3, file = "~/Desktop/mcav_T1_vs_T3_all_genes.csv") # maybe include gene counts too?
DEG_T1_vs_T3.sig.num <- sum(DEG_T1_vs_T3$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T1_vs_T3.sig.num
# 1 DEGs
DEG_T1_vs_T3.sig <- subset(DEG_T1_vs_T3, padj <0.05) # identify and subset significant pvalues
DEG_T1_vs_T3.sig["Treatment_Compare"] <- "T1vsT3" # adding treatment comparison column
DEG_T1_vs_T3.sig.list <- data[which(rownames(data) %in% rownames(DEG_T1_vs_T3.sig)),] # subset list of significant genes from original count data 
DEG_T1_vs_T3.sig.list <- as.data.frame(counts(DEG_T1_vs_T3.sig.list)) # make list of sig gene counts into a df
DEG_T1_vs_T3.sig.list_full <- cbind(DEG_T1_vs_T3.sig, DEG_T1_vs_T3.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_T1_vs_T3.sig.list_full, file = "~/Desktop/mcav_T1_vs_T3_DEG_full.csv") # write out csv
DEG_T1_vs_T3.vst.sig <- varianceStabilizingTransformation(DEG_T1_vs_T3.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
```

### Compare T1 and T4
```{r}
DEG_T1_vs_T4 <- results(DEG.int, contrast = c("Treatment", "Treatment1", "Treatment4")) # results of DESeq2 comparing C and T1
DEG_T1_vs_T4
DEG_T1_vs_T4 <- as.data.frame(DEG_T1_vs_T4) # make results into a df
DEG_T1_vs_T4["Treatment_Compare"] <- "T1vsT4" # adding treatment comparison column
#write.csv(DEG_T1_vs_T4, file = "~/Desktop/mcav_T1_vs_T4_all_genes.csv") # maybe include gene counts too?
DEG_T1_vs_T4.sig.num <- sum(DEG_T1_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T1_vs_T4.sig.num
# 0 DEGs
```

### Compare T2 and T3
```{r}
DEG_T2_vs_T3 <- results(DEG.int, contrast = c("Treatment", "Treatment2", "Treatment3")) # results of DESeq2 comparing C and T1
DEG_T2_vs_T3
DEG_T2_vs_T3 <- as.data.frame(DEG_T2_vs_T3) # make results into a df
DEG_T2_vs_T3["Treatment_Compare"] <- "T2vsT3" # adding treatment comparison column
#write.csv(DEG_T2_vs_T3, file = "~/Desktop/mcav_T2_vs_T3_all_genes.csv") # maybe include gene counts too?
DEG_T2_vs_T3.sig.num <- sum(DEG_T2_vs_T3$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T2_vs_T3.sig.num
# 1 DEG
DEG_T2_vs_T3.sig <- subset(DEG_T2_vs_T3, padj <0.05) # identify and subset significant pvalues
DEG_T2_vs_T3.sig["Treatment_Compare"] <- "T2vsT3" # adding treatment comparison column
DEG_T2_vs_T3.sig.list <- data[which(rownames(data) %in% rownames(DEG_T2_vs_T3.sig)),] # subset list of significant genes from original count data 
DEG_T2_vs_T3.sig.list <- as.data.frame(counts(DEG_T2_vs_T3.sig.list)) # make list of sig gene counts into a df
DEG_T2_vs_T3.sig.list_full <- cbind(DEG_T2_vs_T3.sig, DEG_T2_vs_T3.sig.list) # bind results with gene counts for DEGs
#write.csv(DEG_T2_vs_T3.sig.list_full, file = "~/Desktop/mcav_T2_vs_T3_DEG_full.csv") # write out csv
DEG_T2_vs_T3.vst.sig <- varianceStabilizingTransformation(DEG_T3_vs_T3.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
```

### Compare T2 and T4
```{r}
DEG_T2_vs_T4 <- results(DEG.int, contrast = c("Treatment", "Treatment2", "Treatment4")) # results of DESeq2 comparing C and T1
DEG_T2_vs_T4
DEG_T2_vs_T4 <- as.data.frame(DEG_T2_vs_T4) # make results into a df
DEG_T2_vs_T4["Treatment_Compare"] <- "T2vsT4" # adding treatment comparison column
#write.csv(DEG_T2_vs_T4, file = "~/Desktop/mcav_T2_vs_T4_all_genes.csv") # maybe include gene counts too?
DEG_T2_vs_T4.sig.num <- sum(DEG_T2_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T2_vs_T4.sig.num
# 0 DEG
```

### Compare T3 and T4
```{r}
DEG_T3_vs_T4 <- results(DEG.int, contrast = c("Treatment", "Treatment3", "Treatment4")) # results of DESeq2 comparing C and T1
DEG_T3_vs_T4
DEG_T3_vs_T4 <- as.data.frame(DEG_T3_vs_T4) # make results into a df
DEG_T3_vs_T4["Treatment_Compare"] <- "T3vsT4" # adding treatment comparison column
#write.csv(DEG_T3_vs_T4, file = "~/Desktop/mcav_T3_vs_T4_all_genes.csv") # maybe include gene counts too?
DEG_T3_vs_T4.sig.num <- sum(DEG_T3_vs_T4$padj <0.05, na.rm = T) # identify # of significant pvalues with 5%FDR (padj<0.05) 
DEG_T3_vs_T4.sig.num
# 0 DEG
```

## Make full list of differentially expressed genes and treatments 
```{r}
DEG_control_vs_T1.sig.list_full$gene_id <- rownames(DEG_control_vs_T1.sig.list_full)
rownames(DEG_control_vs_T1.sig.list_full) <- NULL
DEG_control_vs_T2.sig.list_full$gene_id <- rownames(DEG_control_vs_T2.sig.list_full)
rownames(DEG_control_vs_T2.sig.list_full) <- NULL
DEG_control_vs_T3.sig.list_full$gene_id <- rownames(DEG_control_vs_T3.sig.list_full)
rownames(DEG_control_vs_T3.sig.list_full) <- NULL
DEG_control_vs_T4.sig.list_full$gene_id <- rownames(DEG_control_vs_T4.sig.list_full)
rownames(DEG_control_vs_T4.sig.list_full) <- NULL
DEG_T1_vs_T3.sig.list_full$gene_id <- rownames(DEG_T1_vs_T3.sig.list_full)
rownames(DEG_T1_vs_T3.sig.list_full) <- NULL
DEG_T2_vs_T3.sig.list_full$gene_id <- rownames(DEG_T2_vs_T3.sig.list_full)
rownames(DEG_T2_vs_T3.sig.list_full) <- NULL

DEGs.all <- rbind(DEG_control_vs_T1.sig.list_full, 
                  DEG_control_vs_T2.sig.list_full,
                  DEG_control_vs_T3.sig.list_full, 
                  DEG_control_vs_T4.sig.list_full,
                  DEG_T1_vs_T3.sig.list_full,
                  DEG_T2_vs_T3.sig.list_full
                  )
```

## Identify unique genes from DEGs from treatments 
```{r}
dim(DEGs.all) # 108 x 23
length(unique(DEGs.all$gene_id)) # 62 unique genes between all treatments 
#write.csv(DEGs.all, file = "~/Desktop/mcav_DEGs.all_treatment_20210208.csv")
```

## Look at size factors and run vst normalization
```{r}
unique.sig.list <- data[which(rownames(data) %in% DEGs.all_mcav$DEGs), ] # subset list of sig transcripts from original count data
#write.csv(counts(unique.sig.list), file = "~/Desktop/mcav_unique.sig.list_20210208.csv")
SFtest <- estimateSizeFactors(unique.sig.list)
print(sizeFactors(SFtest))
unique.vst.sig <- varianceStabilizingTransformation(unique.sig.list, blind = FALSE) # apply a regularized log transformation to minimize effects of small counts and normalize wrt library 
# -- note: fitType='parametric', but the dispersion trend was not well captured by the
# function: y = a/x + b, and a local regression fit was automatically substituted.
# specify fitType='local' or 'mean' to avoid this message next time.
```


# PCA plot of DEGs
```{r}
mcav_DEGPCAdata <- plotPCA(unique.vst.sig, intgroup = c("Treatment"), returnData=TRUE)
percentVar_pca_mcav <- round(100*attr(mcav_DEGPCAdata, "percentVar")) #plot PCA of samples with all data

mcav_DEG_PCA_plot <- ggplot(mcav_DEGPCAdata, aes(PC1, PC2, color=Treatment)) +
  geom_point(size=6) +
  xlab(paste0("PC1: ",percentVar_pca_mcav[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_pca_mcav[2],"% variance")) +
  scale_color_manual(values = c(control="gray", Treatment1="darkslategray2", Treatment2="darkslategray3", Treatment3="darkslategray4", Treatment4="darkslategray")) +
  #coord_fixed() +
  ggtitle(label = "M. cavernosa") +
  theme_bw() + #Set background color
  theme(legend.text = element_text(size=8), 
        #legend.position="none",
        plot.background = element_blank(),
        #legend.title = element_text(size=18, face="bold"), 
        legend.title=element_blank(),
        axis.text = element_text(size=8), 
        axis.title = element_text(size=10,  face="bold", hjust = 0.5), 
        axis.title.y = element_text(vjust=-1.5),
        plot.title = element_text(size = 15, face = "italic", hjust = 0.5))
mcav_DEG_PCA_plot
# PCA plot is of differentially expressed genes only
PC.info <- mcav_DEG_PCA_plot$data
#ggsave("Output/Figs/mcav/mcav_DEGs_PCA_20220326.jpeg", mcav_DEG_PCA_plot, width = 20, height = 15, units = "cm")
#ggsave("Output/Figs/mcav/mcav_DEGs_PCA_20220326.pdf", mcav_DEG_PCA_plot, width = 20, height = 15, units = "cm")
```






### old code 
```{r}
## Heatmap of DEGs
# This heatmap is going to be wild. Grouping columns by treatment, putting gene id on left hand side and GO term on right hand side
# Also taking out legend 

# Need a file with counts, gene names, GO IDs, term, and ontology 
# This file is acerv genes with GO terms associated with them. One GO term per line, so multiple acerv gene names sometimes if genes have multiple GO terms
# This file includes all gene names and GO IDs
mcav_sig <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/GOSeq/mcav_GOterms.unique.csv", header = TRUE)
mcav_sig <- select(mcav_sig, -X)
colnames(mcav_sig)[1] <-"gene"
head(mcav_sig)

# Getting col order to order unique counts data
list(mcav_metadata)
list <- mcav_metadata[order(mcav_metadata$Treatment),] # need to order them so it will group by treatment in plot
list(list$SampleID) # look at sample IDs and use that list to make col.order
col.order <- c("22_ctl2_Mc_TWF_1",
               "28_ctl1_Mc_GBM_1",
               "42_ctl3_Mc_MGR_1",
               "20_T12_Mc_PWC",
               "39_T13_Mc_FJE",
               "61_T11_Mc_RAP",
               "29_T23_Mc_PND",
               "34_T22_Mc_SVS",
               "58_T21_Mc_EAH",
               "21_T33_Mc_EOU",
               "49_T31_Mc_SWQ",
               "55_T32_Mc_TWP",
               "33_T43_Mc_RFV",
               "46_T41_Mc_QYH_1",
               "56_T42_Mc_JAW") 
  





##### Volcano plots 
## Here, the log transformed adjusted p-values are plotted on the y-axis and log2 fold change values on the x-axis (https://hbctraining.github.io/Intro-to-R-with-DGE/lessons/B1_DGE_visualizing_results.html)
# Read in data 
mcav.DEG <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/DESeq2/mcav/mcav_DEGs.all_treatment_20210208.csv")
mcav.DEG <- select(mcav.DEG, -X)
mcav.DEG$diffexpressed <- "NA"
mcav.DEG$diffexpressed[mcav.DEG$log2FoldChange > 0] <- "Up"
mcav.DEG$diffexpressed[mcav.DEG$log2FoldChange < 0] <- "Down"

# Set thresholds
padj.cutoff <- 0.05
lfc.cutoff <- 0.5

threshold <- mcav.DEG$padj < padj.cutoff & abs(mcav.DEG$log2FoldChange) > lfc.cutoff
length(which(threshold)) # this did not reduce anything, as the df only has DEGs in it?

# Add vector to df
mcav.DEG$threshold <- threshold   

# Volcano plot w/ DEGs
mcav.volcano <- ggplot(mcav.DEG) +
  geom_point(aes(x=log2FoldChange, y=-log10(padj), shape=Treatment_Compare, colour=diffexpressed), size = 2) +
  xlab("log2 fold change") + 
  ylab("-log10 adjusted p-value") +
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25))) 
mcav.volcano
ggsave("~/Desktop/mcav_volcano_20210705.pdf", mcav.volcano, width = 25, height = 25)
ggsave("~/Desktop/mcav_volcano_20210705.jpeg", mcav.volcano, width = 25, height = 25)


## trying volcano plot with GO.slim data 
mcav_ByTreatment <- read.csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/GOSeq/mcav/mcav_ByTreatment_GO.terms_20210208.csv")
View(mcav_ByTreatment)
names(mcav_ByTreatment)[names(mcav_ByTreatment) == "category"] <- "GO.IDs"
mcav_ByTreatment$diffexpressed <- "NA"
mcav_ByTreatment$diffexpressed[mcav_ByTreatment$log2FoldChange > 0] <- "Up"
mcav_ByTreatment$diffexpressed[mcav_ByTreatment$log2FoldChange < 0] <- "Down"

# Read in GO slim info
go.slim <- read_csv("~/Desktop/PutnamLab/Repositories/SedimentStress/SedimentStress/Output/GOSeq/GO-GOslim.csv")
colnames(go.slim) <- c("GO.IDs", "GO.Term", "GO.Slim.Term", "Cat") #rename columns
mcav_ByTreatment <- merge(mcav_ByTreatment, go.slim, by="GO.IDs", all = TRUE) # merge pdam info and GOslim
mcav_ByTreatment <- na.omit(mcav_ByTreatment)

# Volcano plot
mcav_go.volcano <- ggplot(mcav_ByTreatment) +
  geom_point(aes(x=log2FoldChange, y=-log10(padj), colour=diffexpressed, shape=GO.Slim.Term), size = 3) +
  xlab("log2 fold change") + 
  ylab("-log10 adjusted p-value") +
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25))) 
mcav_go.volcano
ggsave("~/Desktop/mcav_volcano.GOterms_20210705.pdf", mcav_go.volcano, width = 25, height = 25)
ggsave("~/Desktop/mcav_volcano.GOterms_20210705.jpeg", mcav_go.volcano, width = 25, height = 25)
```

